<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Transforming dataset for myTAI • myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Transforming dataset for myTAI">
<meta name="robots" content="noindex">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">2.3.0.9006</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-stats.html">Statistical testing with myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-transform.html">Transforming dataset for myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/index.html">... and view more!</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Transforming dataset for myTAI</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/vignettes/tai-transform.Rmd" class="external-link"><code>vignettes/tai-transform.Rmd</code></a></small>
      <div class="d-none name"><code>tai-transform.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="the-motivation-for-transformating-of-rna-seq-data">The motivation for transformating of RNA-seq data<a class="anchor" aria-label="anchor" href="#the-motivation-for-transformating-of-rna-seq-data"></a>
</h2>
<p>As noted in the vignette <a href="tai-stats.html"><code>Statistical testing with myTAI</code></a>,
RNA-seq data transformation (e.g. log2 transformation) can affect the
results of the tests and the overall TAI (as well as TDI, TSI and
analogous) profile. There has been a lively debate around this topic
(see <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003476" class="external-link">Piasecka
et al. (2013)</a> and <a href="https://academic.oup.com/gbe/article/10/9/2266/5076813" class="external-link">Liu &amp;
Robinson-Rechavi (2018)</a> for more information).</p>
<p>How can we use this to our advantage?</p>
<p>The <code>myTAI</code> package provides a flexible framework to apply
different transformations to the expression data in a
<code>PhyloExpressionSet</code> object. This allows you to evaluate the
<code>robustness</code> of the results, i.e. whether the observed
patterns are driven primarily by variation in highly expressed genes
(e.g. <code>raw TPM</code> or counts), by variation in lowly expressed
genes (e.g. under <code>log2</code> or vst transformation), or somewhere
in between (e.g. <code>sqrt</code> transformation).</p>
</div>
<div class="section level2">
<h2 id="transforming-rna-seq-data-for-mytai">Transforming RNA-seq data for myTAI<a class="anchor" aria-label="anchor" href="#transforming-rna-seq-data-for-mytai"></a>
</h2>
<p>The <code>myTAI</code> package provides a function
<code><a href="../reference/tf.html">myTAI::tf()</a></code> that allows you to apply any transformation
function to the expression data in a <code>PhyloExpressionSet</code>
object. This function is flexible and can be used with any
transformation function that takes a matrix as input.</p>
<p>Here is a more extreme example of changes to the overall TAI pattern
due to RNA-seq transformations.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drostlab.github.io/myTAI/">myTAI</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_phyex_set"</span><span class="op">)</span></span>
<span><span class="co"># no transformation</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [===================                     ] 47%  (~1s remaining)       Computing: [======================================  ] 93%  (~0s remaining)       Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-2-1.png" alt="plot_signature function output no transform" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># rank transformation</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set</span> <span class="op">|&gt;</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/tf.html">tf</a></span><span class="op">(</span>FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">2</span>, <span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/rank.html" class="external-link">rank</a></span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [====================                    ] 48%  (~1s remaining)       Computing: [======================================  ] 93%  (~0s remaining)       Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-3-1.png" alt="plot_signature function output rank transform" width="576" style="display: block; margin: auto;"></p>
<p>Indeed, <code>myTAIv2</code> is fitted with a single plotting
function (<code><a href="../reference/plot_signature_transformed.html">myTAI::plot_signature_transformed()</a></code>) to evaluate
the robustness of the patterns against multiple transformations,
i.e.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature_transformed.html">plot_signature_transformed</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [===================                     ] 46%  (~1s remaining)       Computing: [====================================    ] 90%  (~0s remaining)       Computing: [========================================] 100% (done)                         </span></span>
<span><span class="co">## Computing: [==================                      ] 43%  (~1s remaining)       Computing: [=================================       ] 82%  (~0s remaining)       Computing: [========================================] 100% (done)                         </span></span>
<span><span class="co">## Computing: [================                        ] 40%  (~1s remaining)       Computing: [==================================      ] 84%  (~0s remaining)       Computing: [========================================] 100% (done)                         </span></span>
<span><span class="co">## Computing: [====================                    ] 48%  (~1s remaining)       Computing: [======================================  ] 95%  (~0s remaining)       Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-4-1.png" alt="plot_signature_transformed function output transform all" width="768" style="display: block; margin: auto;"></p>
<p>See how different some of the transformations are! In general,
though, the earlier stages are younger compared to the mid-stage. Across
most transformations, the late-stage is younger compared to the
mid-stage. This is consistent with the <code>molecular hourglass</code>
model.</p>
<p>In addition, the p-values for various tests (default = flat line
test) can be assessed through the <code><a href="../reference/tf_stability.html">myTAI::tf_stability()</a></code>
function. The test can be changed with the
<code>conservation_test</code> argument,
e.g. <code>conservation_test = stat_flatline_test</code>,
<code>conservation_test = stat_flatline_test</code> and so on.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tf_stability_res</span> <span class="op">&lt;-</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/tf_stability.html">tf_stability</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tf_stability_res</span></span></code></pre></div>
<pre><code><span><span class="co">##         none         sqrt         log2         rank          vst         rlog </span></span>
<span><span class="co">## 7.661681e-01 9.358450e-06 2.690131e-37 4.020056e-93 4.799501e-33 2.751516e-34</span></span></code></pre>
<p>Here, we can see that the p-values for the flat line test are all
below 0.05 for all transformations. This is reassuring as to the
stability of the TAI profile across transformations.</p>
<p>Indeed, within the <code><a href="../reference/plot_signature_transformed.html">myTAI::plot_signature_transformed()</a></code>
function, we can also set the <code>conservation_test</code> parameter
(e.g. <code>conservation_test = stat_flatline_test</code>) alongside the
display for the p-values (<code>show_p_val = TRUE</code>), to get an
all-in-one plot!</p>
<div class="tip">
<p>If you are curious about the exact nature of the transformations used
by <code><a href="../reference/plot_signature_transformed.html">myTAI::plot_signature_transformed()</a></code> and
<code><a href="../reference/tf_stability.html">myTAI::tf_stability()</a></code> you can check the list of functions
that were applied via <code>print(myTAI::COUNT_TRANSFORMS)</code>. Note
that DESeq2-based transforms (‘vst’, ‘rlog’) are only available if
DESeq2 is installed.</p>
<details><summary>
Removing or adding transformations
</summary><p>To remove transformations, e.g. “rlog”, “rank”, “vst” from the list,
e.g.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">COUNT_TRANSFORMS_REMOVED</span> <span class="op">&lt;-</span> <span class="va">COUNT_TRANSFORMS</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">COUNT_TRANSFORMS</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rlog"</span>, <span class="st">"rank"</span>, <span class="st">"vst"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>To add transformations, e.g.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">COUNT_TRANSFORMS_ADDED</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">COUNT_TRANSFORMS</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>new_transform <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then apply to tf_stability or plot_signature functions, e.g.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/tf_stability.html">tf_stability</a></span><span class="op">(</span><span class="va">example_phyex_set</span>, transformations <span class="op">=</span> <span class="va">COUNT_TRANSFORMS_REMOVED</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature_transformed.html">plot_signature_transformed</a></span><span class="op">(</span><span class="va">example_phyex_set</span>, transformations <span class="op">=</span> <span class="va">COUNT_TRANSFORMS_REMOVED</span><span class="op">)</span></span></code></pre></div>
</details><p>For <code>ScPhyloExpressionSet</code>, it is important to remove
certain transformations such as “rlog”, “rank”, “vst” and “log2” as they
are not suitable for single cell data.</p>
</div>
</div>
<div class="section level2">
<h2 id="other-robustness-checks">Other robustness checks<a class="anchor" aria-label="anchor" href="#other-robustness-checks"></a>
</h2>
<p>Alternatively (or in conjunction) to the transformations,
<code>myTAIv2</code> allows us to check how stable the TAI profile is
after progressively removing the top expressed genes. This can be done
using the <code><a href="../reference/plot_signature_gene_quantiles.html">myTAI::plot_signature_gene_quantiles()</a></code> function.
By default, the quantiles are set to <code>1</code>, <code>0.99</code>,
<code>0.95</code>, <code>0.9</code> and <code>0.8</code> (i.e. from
removing no genes <code>quantile = 1</code> to removing the top 20% of
expressed genes <code>quantile = 0.8</code>).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature_gene_quantiles.html">plot_signature_gene_quantiles</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-10-1.png" alt="plot_signature_gene_quantiles function output" width="768" style="display: block; margin: auto;"></p>
<p>From this, we can see that the most of the ‘bulge’ on the Bent
Cotyledon stage is driven by a very few highly expressed genes.
Otherwise, the TAI profile is very stable.</p>
<p>We could also filter genes based on other categories using:<br><code><a href="../reference/genes_top_variance.html">myTAI::genes_top_variance()</a></code> → get high variance genes.
<code><a href="../reference/genes_top_mean.html">myTAI::genes_top_mean()</a></code> and
<code><a href="../reference/genes_top_expr.html">myTAI::genes_top_expr()</a></code> → get highly expressed genes, like
in in <code><a href="../reference/plot_signature_gene_quantiles.html">myTAI::plot_signature_gene_quantiles()</a></code><br><code><a href="../reference/genes_lowly_expressed.html">myTAI::genes_lowly_expressed()</a></code> → get lowly expressed
genes</p>
<details><summary>
See filtering workflow
</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get top 1% variance genes (default parameters)</span></span>
<span><span class="va">genes.top_var</span> <span class="op">&lt;-</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/genes_top_variance.html">genes_top_variance</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span>
<span><span class="co"># get top 1% expressed genes (default parameters)</span></span>
<span><span class="va">genes.top_mean</span> <span class="op">&lt;-</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/genes_top_mean.html">genes_top_mean</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span>
<span><span class="co"># get lowly expressed genes with mean &lt; 1 (default parameters)</span></span>
<span><span class="va">genes.low_expr</span> <span class="op">&lt;-</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/genes_lowly_expressed.html">genes_lowly_expressed</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<p>Once you have a gene set, we can use
<code><a href="../reference/remove_genes.html">myTAI::remove_genes()</a></code> to filter out genes in a given
set:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># removing top 1% variance genes</span></span>
<span><span class="va">example_phyex_set.rm_top_var</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/remove_genes.html">remove_genes</a></span><span class="op">(</span><span class="va">example_phyex_set</span>, <span class="va">genes.top_var</span><span class="op">)</span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set.rm_top_var</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-12-1.png" alt="plot_signature function output removing top variance" width="576" style="display: block; margin: auto;"></p>
<p>We could also use <code><a href="../reference/select_genes.html">myTAI::select_genes()</a></code> to keep the
genes in a given set:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select top 1% expressed genes</span></span>
<span><span class="va">example_phyex_set.select_top_expr</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/select_genes.html">select_genes</a></span><span class="op">(</span><span class="va">example_phyex_set</span>, <span class="va">genes.top_mean</span><span class="op">)</span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set.select_top_expr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-13-1.png" alt="plot_signature function output for top expressed" width="576" style="display: block; margin: auto;"></p>
<p>If you have certain genes of interest, we can also select or remove
individual genes with <code><a href="../reference/select_genes.html">myTAI::select_genes()</a></code> and
<code><a href="../reference/remove_genes.html">myTAI::remove_genes()</a></code>, respectively.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove two genes (AT1G02780 &amp; AT1G03880)</span></span>
<span><span class="va">example_phyex_set.select_goi</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/remove_genes.html">remove_genes</a></span><span class="op">(</span><span class="va">example_phyex_set</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AT1G02780"</span>, <span class="st">"AT1G03880"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set.select_goi</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-transform_files/figure-html/unnamed-chunk-14-1.png" alt="plot_signature function output for gene of interest" width="576" style="display: block; margin: auto;"></p>
</details><p>For a more sophisticated workflow to extract genes that drive the TAI
profile, check out this vignette<br>
→ <a href="tai-breaker.html"><code>Break TAI patterns using gaTAI</code></a></p>
</div>
<div class="section level2">
<h2 id="transforming-gene-age-information-for-mytai">Transforming gene age information for myTAI<a class="anchor" aria-label="anchor" href="#transforming-gene-age-information-for-mytai"></a>
</h2>
<p>Another aspect of the <code>myTAI</code> package is the ability to
transform gene age information itself using the
<code><a href="../reference/tf_PS.html">myTAI::tf_PS()</a></code> function. This workflow is still being
tested and will be documented in due time.</p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The choice of RNA-seq data transformation (including no
transformation) is an important and often overlooked aspect of gene
expression data analysis. These choices may affect the results of the
TAI (and related indices) profiles, as well as the <a href="tai-stats.html">statistical tests</a> and the interpretation of
the data.</p>
<p><code>myTAIv2</code> provides a powerful framework to apply various
RNA-seq data transformations to <code>BulkPhyloExpressionSet</code> and
<code>ScPhyloExpressionSet</code> objects. With this, we can use the
transformed patterns to our advantage, namely, to evaluate the overall
robustness of our patterns. We also explored the stability of the TAI
profile after progressively removing the top expressed genes. Taken
together, these analyses allow us to better understand the expression
levels of genes that drive the TAI profile.</p>
<div class="important">
<p>While we can perturb the TAI patterns <em>in silico</em>, nothing
beats repeating experiments in the lab, with multiple biological
replicates and gold-standard protocols for RNA-seq data generation and
processing.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Ștefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
