---
title: "Bring your datasets into myTAI"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{phylo-expression-object}
  %\VignetteEngine{knitr::rmarkdown}
---

## S7 PhyloExpressionSet object

Starting from `myTAIv2`, we moved on from the traditional `data.frame` to the new [`S7 OOP system`](https://rconsortium.github.io/S7/) to facilitate fast and flexible computation for evolutionary transcriptomics analyses.

To illustrate this, we will use the example data `example_phyex_set`. The `PhyloExpressionSet` object is designed to store gene expression data along with associated metadata such as gene ages, phylostratum information, and other relevant properties.

```{r message = FALSE}
library(myTAI)
data("example_phyex_set")
```

```{r message = FALSE, eval=FALSE}
# inspect the object
class(example_phyex_set)
```

<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
class(example_phyex_set)
```
</details>

```{r message = FALSE, eval=FALSE}
S7::prop_names(example_phyex_set)
```
<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
S7::prop_names(example_phyex_set)
```
</details>

```{r message = FALSE, eval = FALSE}
# let's explore the properties using @
# for example:
example_phyex_set@strata |> head()
example_phyex_set@gene_ids |> head()
example_phyex_set@counts[1:4,1:5]
```

See how rich this `PhyloExpressionSet` object is!

The `PhyloExpressionSet` is designed to ensure efficient interaction with `myTAIv2` functions. For example, if you try the `myTAI::flatline_test` function twice, i.e.

```{r message = FALSE, eval = FALSE}
myTAI::flatline_test(example_phyex_set)
myTAI::flatline_test(example_phyex_set)
``` 

you will notice that the permutations are already pre-computed for the second run so that you don't need to wait a couple of seconds again.

But how do you get your `.tsv` and `.csv` files and convert your `data.frame` and `tibble` to a S7 `PhyloExpressionSet` object?

## Constructing a PhyloExpressionSet

The necessary and sufficient information needed for creating a `PhyloExpressionSet` object is:  
(1) A matrix or data frame of gene expression values (genes x samples).  
(2) A numerical value or factored string associated with each gene, typically gene age ranks or [`phylorank`](articles/phylostratigraphy.html) for `TAI` (but also [`deciled dNdS values`](articles/other_strata.html) for `TDI` or [`deciled tau values`](articles/other_strata.html) for `TSI` etc.).

### Loading raw data

The raw outputs of (1) gene expression quantification and (2) gene age inference can typically be found in the form of a tab-separated `.tsv` or comma-separated `.csv` file format. One useful package to do this is `readr`, using the functions `readr::read_csv()` or `readr::read_tsv()`. For `.csv` files, if `readr::read_csv()` doesn't work, try `readr::read_csv2()` in European locales.

### Mock dataset without replicates

#### Bulk RNA-seq data

In this example, we are using a dataset with only one entry per stage (no replicates).

```{r message = FALSE}
data("example_phyex_set_old")
example_expression <- 
    example_phyex_set_old@counts |> 
    as.data.frame() |> 
    tibble::rownames_to_column(var = "GeneID")
example_phylorank <-
    example_phyex_set_old@strata
```

```{r message = FALSE, eval = FALSE}
# let's explore the example expression dataset
# for example:
example_expression |> head()
```

<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
example_expression |> head()
```
As you can see, there is one column as the gene identifier and the rest of the columns are the expression values for each developmental stage. Instead of developmental stage, one can also use experimental conditions, e.g. `control`, `treatment`, etc.
</details>
 
Now we construct the input data.frame for as_PhyloExpressionSet()

```{r message = FALSE}
example_phyex_set.df <-
    data.frame(phylorank = example_phylorank) |>
    tibble::rownames_to_column(var = "GeneID") |>
    dplyr::select(phylorank, GeneID) |>
    dplyr::left_join(example_expression, by = "GeneID")
```

```{r message = FALSE, eval=FALSE}
example_phyex_set.df |> head()
example_phyex_set.df |> str()
```

<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
example_phyex_set.df |> head()
example_phyex_set.df |> str()
```
</details>

The example `example_phyex_set.df` is now formatted to meet the requirements for the `as_PhyloExpressionSet()` function, where: 

::: {.important}
column 1 contains the phylorank information, which can be either numeric or factors.  
column 2 contains the gene identifier (here called `GeneID`).  
column 3 onwards contain the expression data with the column titles being the developmental stages or the experimental conditions.
:::



::: {.tip}
The `as_PhyloExpressionSet()` function will automatically detect the first column as the phylorank and the second column as the gene identifier. If your data is structured differently, you can reassign the columns accordingly using `dplyr::relocate()`.
:::

Now, we can use the `as_PhyloExpressionSet()` function to convert the correctly formatted input `data.frame` to a `PhyloExpressionSet` object.

```{r warning=FALSE, message=FALSE}
example_phyex_set.remake <- 
    myTAI::as_PhyloExpressionSet(
        example_phyex_set.df,
        name = "reconstituted phyex_set_old")
```

There you have it, a `PhyloExpressionSet` object!

```{r message = FALSE, eval=FALSE}
# inspect the object
class(example_phyex_set.remake)
```

<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
class(example_phyex_set.remake)
```
</details>

```{r message = FALSE, eval=FALSE}
S7::prop_names(example_phyex_set.remake)
```

<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
S7::prop_names(example_phyex_set.remake)
```
</details>

You can further explore the properties of the `PhyloExpressionSet` object using the `@` operator, for example: `example_phyex_set.remake@counts |> head()`.

... and plot your transcriptome age index!

```{r message = FALSE, fig.height=4, fig.width=6, fig.alt="plot_signature_output", dev.args = list(bg = 'transparent'), fig.align='center'}
example_phyex_set.remake |> 
    myTAI::plot_signature()
```


#### What if the dataset contains replicates?

This is exactly the case for the new `example_phyex_set` dataset (not like `example_phyex_set_old`), which contains replicates for each developmental stage. The `as_PhyloExpressionSet()` function can handle this as well, as long as we use the `groups` parameter to specify the grouping of replicates, i.e.

```{r message = FALSE}
data("example_phyex_set") # not data("example_phyex_set_old")
example_expression <- 
    example_phyex_set@counts |> 
    as.data.frame() |> 
    tibble::rownames_to_column(var = "GeneID")
example_phylorank <-
    example_phyex_set@strata

example_phyex_set.df <-
    data.frame(phylorank = example_phylorank) |>
    tibble::rownames_to_column(var = "GeneID") |>
    dplyr::select(phylorank, GeneID) |>
    dplyr::left_join(example_expression, by = "GeneID")
```

```{r message = FALSE, eval=FALSE}
colnames(example_phyex_set.df)
```
<details> <summary>Show output</summary>
```{r message = FALSE, echo=FALSE}
colnames(example_phyex_set.df)
```
</details>

As you can see, this dataset has three replicates per stage.

```{r message = FALSE}
groups_example_phyex_set.df <- c(
            rep("Preglobular",3), 
            rep("Globular",3), 
            rep("Early Heart",3), 
            rep("Late Heart",3), 
            rep("Early Torpedo",3), 
            rep("Late Torpedo",3), 
            rep("Bent Cotyledon",3), 
            rep("Mature Green",3))

# we can add the group information using groups
example_phyex_set.remake <- 
    myTAI::as_PhyloExpressionSet(
        example_phyex_set.df,
        groups = groups_example_phyex_set.df, # adding group information here
        name = "reconstituted phyex_set")
```

Done!

::: {.tip}
Along with the `groups` parameter, you can also specify the `name`, `strata_labels` and other parameters to provide a more descriptions for your `PhyloExpressionSet` object. This is useful for plotting and visualisation purposes. Check this using `?` before the function (i.e. `?myTAI::as_PhyloExpressionSet()`.
:::

#### Single cell RNA-seq data

If you are interested in single cell RNA-seq data, the process is similar, but you will need to ensure that your data is structured correctly for single cell analysis. The `as_PhyloExpressionSet()` function can still be used, but you may need to adjust the input data frame to include cell identifiers and other relevant metadata.

## Summary

In this section, we have learned how to create a `PhyloExpressionSet` object from raw gene expression data and phylorank information. This object is essential for performing evolutionary transcriptomics analyses using the `myTAI` package, and provides a structured way to store and manipulate evolutionary transcriptomics data, making it easier to perform various analyses and visualisations.
