<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Obtaining taxonomy information</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Obtaining taxonomy information</h1>



<p>The <code>taxonomy()</code> function (formerly) implemented in
<code>myTAI</code> relies on the powerful package <a href="https://github.com/ropensci/taxize">taxize</a>. More specifically,
the taxonomic information retrieval has been customized for the
<code>myTAI</code> standard and for organism specific information
retrieval.</p>
<div class="important">
<p>While the previous <code>taxonomy()</code> function has been
<strong>deprecated</strong> since <code>taxize</code> was pulled from
CRAN, users can nevertheless follow the taxonomy pipeline by installing
the <code>taxize</code> package and <strong>copy</strong> the old
taxonomy function.</p>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install taxize from CRAN</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;taxize&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co"># if taxize is not available again</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;ropensci/taxize&quot;</span>)</span></code></pre></div>
<p><strong>Copy</strong> the taxonomy function:</p>
<details>
<summary>
<strong>open for the taxonomy function</strong>
</summary>
<p>Click on the copy icon to copy the function.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co">#&#39; @title Retrieving Taxonomic Information of a Query Organism</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co">#&#39; @description This function takes the scientific name of a query organism</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&#39; and returns selected output formats of taxonomic information for the corresponding organism.</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&#39; @param organism a character string specifying the scientific name of a query organism.</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&#39; @param db a character string specifying the database to query, e.g. \code{db} = \code{&quot;itis&quot;} or \code{&quot;ncbi&quot;}.</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&#39; @param output a character string specifying the taxonomic information that shall be returned. </span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&#39; Implemented are: \code{output} = \code{&quot;classification&quot;}, \code{&quot;taxid&quot;}, or \code{&quot;children&quot;}.</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&#39; @details This function is based on the powerful package \pkg{taxize} and implements</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&#39; the customized retrieval of taxonomic information for a query organism. </span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&#39; The following data bases can be selected to retrieve taxonomic information:</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&#39; \itemize{</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&#39; \item \code{db = &quot;itis&quot;} : Integrated Taxonomic Information Service</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&#39; \item \code{db = &quot;ncbi&quot;} : National Center for Biotechnology Information</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&#39; }</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&#39; @author Hajk-Georg Drost</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&#39; @examples</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&#39; \dontrun{</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&#39; # retrieving the taxonomic hierarchy of &quot;Arabidopsis thaliana&quot;</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co">#&#39; # from NCBI Taxonomy</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#&#39; taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;ncbi&quot;)</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&#39; # the same can be applied to database : &quot;itis&quot;</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;itis&quot;)</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co">#&#39; # retrieving the taxonomic hierarchy of &quot;Arabidopsis&quot;</span></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis&quot;,db = &quot;ncbi&quot;) # analogous : db = &quot;ncbi&quot; or &quot;itis&quot;</span></span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a><span class="co">#&#39; # or just &quot;Arabidopsis&quot;</span></span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis&quot;,db = &quot;ncbi&quot;)</span></span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a><span class="co">#&#39; # retrieving the taxonomy id of the query organism and in the correspondning database</span></span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a><span class="co">#&#39; # taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;ncbi&quot;, output = &quot;taxid&quot;)</span></span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a><span class="co">#&#39; # the same can be applied to databases : &quot;ncbi&quot; and &quot;itis&quot;</span></span>
<span id="cb2-40"><a href="#cb2-40" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;ncbi&quot;, output = &quot;taxid&quot;)</span></span>
<span id="cb2-41"><a href="#cb2-41" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;itis&quot;, output = &quot;taxid&quot;)</span></span>
<span id="cb2-42"><a href="#cb2-42" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-43"><a href="#cb2-43" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-44"><a href="#cb2-44" tabindex="-1"></a><span class="co">#&#39; # retrieve children taxa of the query organism stored in the correspondning database</span></span>
<span id="cb2-45"><a href="#cb2-45" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis&quot;,db = &quot;ncbi&quot;, output = &quot;children&quot;)</span></span>
<span id="cb2-46"><a href="#cb2-46" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-47"><a href="#cb2-47" tabindex="-1"></a><span class="co">#&#39; # the same can be applied to databases : &quot;ncbi&quot; and &quot;itis&quot;</span></span>
<span id="cb2-48"><a href="#cb2-48" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;ncbi&quot;, output = &quot;children&quot;)</span></span>
<span id="cb2-49"><a href="#cb2-49" tabindex="-1"></a><span class="co">#&#39;  taxonomy(&quot;Arabidopsis thaliana&quot;,db = &quot;itis&quot;, output = &quot;children&quot;)</span></span>
<span id="cb2-50"><a href="#cb2-50" tabindex="-1"></a><span class="co">#&#39;  </span></span>
<span id="cb2-51"><a href="#cb2-51" tabindex="-1"></a><span class="co">#&#39; }</span></span>
<span id="cb2-52"><a href="#cb2-52" tabindex="-1"></a><span class="co">#&#39; @references</span></span>
<span id="cb2-53"><a href="#cb2-53" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-54"><a href="#cb2-54" tabindex="-1"></a><span class="co">#&#39; Scott Chamberlain and Eduard Szocs (2013). taxize - taxonomic search and retrieval in R. F1000Research,</span></span>
<span id="cb2-55"><a href="#cb2-55" tabindex="-1"></a><span class="co">#&#39; 2:191. URL: http://f1000research.com/articles/2-191/v2.</span></span>
<span id="cb2-56"><a href="#cb2-56" tabindex="-1"></a><span class="co">#&#39; </span></span>
<span id="cb2-57"><a href="#cb2-57" tabindex="-1"></a><span class="co">#&#39; Scott Chamberlain, Eduard Szocs, Carl Boettiger, Karthik Ram, Ignasi Bartomeus, and John Baumgartner</span></span>
<span id="cb2-58"><a href="#cb2-58" tabindex="-1"></a><span class="co">#&#39; (2014) taxize: Taxonomic information from around the web. R package version 0.3.0.</span></span>
<span id="cb2-59"><a href="#cb2-59" tabindex="-1"></a><span class="co">#&#39; https://github.com/ropensci/taxize</span></span>
<span id="cb2-60"><a href="#cb2-60" tabindex="-1"></a><span class="co">#&#39; @export</span></span>
<span id="cb2-61"><a href="#cb2-61" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" tabindex="-1"></a>taxonomy <span class="ot">&lt;-</span> <span class="cf">function</span>(organism, <span class="at">db =</span> <span class="st">&quot;ncbi&quot;</span>, <span class="at">output =</span> <span class="st">&quot;classification&quot;</span>){</span>
<span id="cb2-63"><a href="#cb2-63" tabindex="-1"></a>        </span>
<span id="cb2-64"><a href="#cb2-64" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.element</span>(output,<span class="fu">c</span>(<span class="st">&quot;classification&quot;</span>,<span class="st">&quot;taxid&quot;</span>,<span class="st">&quot;children&quot;</span>)))</span>
<span id="cb2-65"><a href="#cb2-65" tabindex="-1"></a>                <span class="fu">stop</span> (<span class="st">&quot;The output &#39;&quot;</span>,output,<span class="st">&quot;&#39; is not supported by this function.&quot;</span>)</span>
<span id="cb2-66"><a href="#cb2-66" tabindex="-1"></a>        </span>
<span id="cb2-67"><a href="#cb2-67" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.element</span>(db,<span class="fu">c</span>(<span class="st">&quot;ncbi&quot;</span>,<span class="st">&quot;itis&quot;</span>)))</span>
<span id="cb2-68"><a href="#cb2-68" tabindex="-1"></a>                <span class="fu">stop</span> (<span class="st">&quot;Database &#39;&quot;</span>,db,<span class="st">&quot;&#39; is not supported by this function.&quot;</span>)</span>
<span id="cb2-69"><a href="#cb2-69" tabindex="-1"></a>        </span>
<span id="cb2-70"><a href="#cb2-70" tabindex="-1"></a>        name <span class="ot">&lt;-</span> id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-71"><a href="#cb2-71" tabindex="-1"></a></span>
<span id="cb2-72"><a href="#cb2-72" tabindex="-1"></a>        <span class="cf">if</span> (db <span class="sc">==</span> <span class="st">&quot;ncbi&quot;</span>)</span>
<span id="cb2-73"><a href="#cb2-73" tabindex="-1"></a>                tax_hierarchy <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(taxize<span class="sc">::</span><span class="fu">classification</span>(taxize<span class="sc">::</span><span class="fu">get_uid</span>(organism), <span class="at">db =</span> <span class="st">&quot;ncbi&quot;</span>)[[<span class="dv">1</span>]])</span>
<span id="cb2-74"><a href="#cb2-74" tabindex="-1"></a>        </span>
<span id="cb2-75"><a href="#cb2-75" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span> (db <span class="sc">==</span> <span class="st">&quot;itis&quot;</span>)    </span>
<span id="cb2-76"><a href="#cb2-76" tabindex="-1"></a>                tax_hierarchy <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(taxize<span class="sc">::</span><span class="fu">classification</span>(taxize<span class="sc">::</span><span class="fu">get_tsn</span>(organism), <span class="at">db =</span> <span class="st">&quot;itis&quot;</span>)[[<span class="dv">1</span>]])</span>
<span id="cb2-77"><a href="#cb2-77" tabindex="-1"></a>        </span>
<span id="cb2-78"><a href="#cb2-78" tabindex="-1"></a>        <span class="co"># tryCatch({colnames(tax_hierarchy) &lt;- c(&quot;name&quot;,&quot;rank&quot;,&quot;id&quot;)},stop(&quot;The connection to &quot;,db,&quot; did not work properly. Please check your internet connection or maybe the API did change.&quot;, call. = FALSE))</span></span>
<span id="cb2-79"><a href="#cb2-79" tabindex="-1"></a>        </span>
<span id="cb2-80"><a href="#cb2-80" tabindex="-1"></a>        <span class="cf">if</span>(output <span class="sc">==</span> <span class="st">&quot;classification&quot;</span>){</span>
<span id="cb2-81"><a href="#cb2-81" tabindex="-1"></a>                </span>
<span id="cb2-82"><a href="#cb2-82" tabindex="-1"></a>                <span class="fu">return</span>(tax_hierarchy)</span>
<span id="cb2-83"><a href="#cb2-83" tabindex="-1"></a>        }</span>
<span id="cb2-84"><a href="#cb2-84" tabindex="-1"></a>        </span>
<span id="cb2-85"><a href="#cb2-85" tabindex="-1"></a>        <span class="cf">if</span>(output <span class="sc">==</span> <span class="st">&quot;taxid&quot;</span>){</span>
<span id="cb2-86"><a href="#cb2-86" tabindex="-1"></a>                </span>
<span id="cb2-87"><a href="#cb2-87" tabindex="-1"></a>                        <span class="fu">return</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">filter</span>(tax_hierarchy, name <span class="sc">==</span> organism),id))</span>
<span id="cb2-88"><a href="#cb2-88" tabindex="-1"></a>        }</span>
<span id="cb2-89"><a href="#cb2-89" tabindex="-1"></a>        </span>
<span id="cb2-90"><a href="#cb2-90" tabindex="-1"></a>        <span class="cf">if</span>(output <span class="sc">==</span> <span class="st">&quot;children&quot;</span>){</span>
<span id="cb2-91"><a href="#cb2-91" tabindex="-1"></a>                </span>
<span id="cb2-92"><a href="#cb2-92" tabindex="-1"></a>                <span class="fu">return</span>(<span class="fu">as.data.frame</span>(taxize<span class="sc">::</span><span class="fu">children</span>(organism, <span class="at">db =</span> db)[[<span class="dv">1</span>]]))</span>
<span id="cb2-93"><a href="#cb2-93" tabindex="-1"></a>        } </span>
<span id="cb2-94"><a href="#cb2-94" tabindex="-1"></a>}</span></code></pre></div>
</details>
<p>The <code>taxonomy()</code> function can be used to classify genomes
according to phylogenetic classification into Phylostrata
(Phylostratigraphy) or to retrieve species specific taxonomic
information when performing Divergence Stratigraphy.</p>
<p>For larger taxonomy queries it may be useful to create an NCBI
Account and set up an <a href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/">ENTREZ
API KEY</a>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># install.packages(c(&quot;taxize&quot;, &quot;usethis&quot;))</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>taxize<span class="sc">::</span><span class="fu">use_entrez</span>()</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># Create your key from your (brand-new) account&#39;s. </span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co"># After generating your key set it as ENTREZ_KEY in .Renviron.</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co"># ENTREZ_KEY=&#39;youractualkeynotthisstring&#39;</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co"># For that, use usethis::edit_r_environ()</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>usethis<span class="sc">::</span><span class="fu">edit_r_environ</span>()</span></code></pre></div>
<div id="taxonomic-information-retrieval" class="section level2">
<h2>Taxonomic Information Retrieval</h2>
<p>The <code>taxonomy()</code> function to retrieve taxonomic
information.</p>
<details>
<summary>
<strong>retrieve taxonomy hierarchy</strong>
</summary>
<p>In the following example we will obtain the taxonomic hierarchy of
<code>Arabidopsis thaliana</code> from <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># retrieving the taxonomic hierarchy of &quot;Arabidopsis thaliana&quot;</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co"># from NCBI Taxonomy</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="fu">taxonomy</span>( <span class="at">organism =</span> <span class="st">&quot;Arabidopsis thaliana&quot;</span>, </span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>          <span class="at">db       =</span> <span class="st">&quot;ncbi&quot;</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>          <span class="at">output   =</span> <span class="st">&quot;classification&quot;</span> )</span></code></pre></div>
<details>
<summary>
Show output
</summary>
<pre><code>## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis+thaliana
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0</code></pre>
<pre><code>##                    name          rank      id
## 1    cellular organisms cellular root  131567
## 2             Eukaryota        domain    2759
## 3         Viridiplantae       kingdom   33090
## 4          Streptophyta        phylum   35493
## 5        Streptophytina     subphylum  131221
## 6           Embryophyta         clade    3193
## 7          Tracheophyta         clade   58023
## 8         Euphyllophyta         clade   78536
## 9         Spermatophyta         clade   58024
## 10        Magnoliopsida         class    3398
## 11      Mesangiospermae         clade 1437183
## 12       eudicotyledons         clade   71240
## 13           Gunneridae         clade   91827
## 14         Pentapetalae         clade 1437201
## 15               rosids         clade   71275
## 16              malvids         clade   91836
## 17          Brassicales         order    3699
## 18         Brassicaceae        family    3700
## 19           Camelineae         tribe  980083
## 20          Arabidopsis         genus    3701
## 21 Arabidopsis thaliana       species    3702</code></pre>
</details>
<p>The <code>organism</code> argument takes the scientific name of a
query organism, the <code>db</code> argument specifies that database
from which the corresponding taxonomic information shall be retrieved,
e.g. <code>ncbi</code> (NCBI Taxonomy) and <code>itis</code> (Integrated
Taxonomic Information System) and the <code>output</code> argument
specifies the type of taxonomic information that shall be returned for
the query organism, e.g. <code>classification</code>,
<code>taxid</code>, or <code>children</code>.</p>
<p>The output of <code>classification</code> is a
<code>data.frame</code> storing the taxonomic hierarchy of
<code>Arabidopsis thaliana</code> starting with
<code>cellular organisms</code> up to <code>Arabidopsis thaliana</code>.
The first column stores the taxonomic name, the second column the
taxonomic rank, and the third column the NCBI Taxonomy id for
corresponding taxa.</p>
<p>Analogous <code>classification</code> information can be obtained
from different databases.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># retrieving the taxonomic hierarchy of &quot;Arabidopsis thaliana&quot;</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co"># from the Integrated Taxonomic Information System</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="fu">taxonomy</span>( <span class="at">organism =</span> <span class="st">&quot;Arabidopsis thaliana&quot;</span>, </span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>          <span class="at">db       =</span> <span class="st">&quot;itis&quot;</span>,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>          <span class="at">output   =</span> <span class="st">&quot;classification&quot;</span> )</span></code></pre></div>
<details>
<summary>
Show output
</summary>
<pre><code>## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis thaliana
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0</code></pre>
<pre><code>##                    name          rank     id
## 1               Plantae       kingdom 202422
## 2         Viridiplantae    subkingdom 954898
## 3          Streptophyta  infrakingdom 846494
## 4           Embryophyta superdivision 954900
## 5          Tracheophyta      division 846496
## 6       Spermatophytina   subdivision 846504
## 7         Magnoliopsida         class  18063
## 8               Rosanae    superorder 846548
## 9           Brassicales         order 822943
## 10         Brassicaceae        family  22669
## 11          Arabidopsis         genus  23040
## 12 Arabidopsis thaliana       species  23041</code></pre>
<p>The <code>output</code> argument allows you to directly access
taxonomy ids for a query organism or species.</p>
</details>
</details>
<details>
<summary>
<strong>retrieve taxonomy ID from <code>ncbi</code></strong>
</summary>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># retrieving the taxonomy id of the query organism from NCBI Taxonomy</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="fu">taxonomy</span>( <span class="at">organism =</span> <span class="st">&quot;Arabidopsis thaliana&quot;</span>, </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>          <span class="at">db       =</span> <span class="st">&quot;ncbi&quot;</span>, </span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>          <span class="at">output   =</span> <span class="st">&quot;taxid&quot;</span> )</span></code></pre></div>
<details>
<summary>
Show output
</summary>
<pre><code>## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis+thaliana
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0</code></pre>
<pre><code>##     id
## 1 3702</code></pre>
</details>
</details>
<details>
<summary>
<strong>retrieve taxonomy ID from <code>itis</code></strong>
</summary>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># retrieving the taxonomy id of the query organism from Integrated Taxonomic Information Service</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="fu">taxonomy</span>( <span class="at">organism =</span> <span class="st">&quot;Arabidopsis&quot;</span>, </span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>          <span class="at">db       =</span> <span class="st">&quot;itis&quot;</span>, </span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>          <span class="at">output   =</span> <span class="st">&quot;taxid&quot;</span> )</span></code></pre></div>
<details>
<summary>
Show output
</summary>
<pre><code>## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0</code></pre>
<pre><code>##      id
## 1 23040</code></pre>
</details>
</details>
<p>So far, the following data bases can be accesses to retrieve
taxonomic information:</p>
<ul>
<li><code>db = &quot;itis&quot;</code> : Integrated Taxonomic Information
Service</li>
<li><code>db = &quot;ncbi&quot;</code> : National Center for Biotechnology
Information</li>
</ul>
<details>
<summary>
<strong>How does the <code>taxonomy(db = &quot;ncbi&quot;)</code> output differ
from <code>GenEra</code>?</strong>
</summary>
<p>The taxonomic classifications should be the same between
<code>taxonomy(..., db = &quot;ncbi&quot;)</code> and the taxonomic
classifications in the <code>GenEra</code> output (since it uses NCBI
taxdump as input). But it should be noted that the recent updates to
NCBI taxonomy has meant that the highest order ranks (cellular root,
domain, kingdom etc.) may differ.</p>
</details>
</div>
<div id="retrieve-children-nodes" class="section level2">
<h2>Retrieve Children Nodes</h2>
<p>Another <code>output</code> supported by <code>taxonomy()</code> is
<code>children</code> that returns the immediate children taxa for a
query organism. This feature is useful to determine species
relationships for quantifying recent evolutionary conservation with
Divergence Stratigraphy.</p>
<details>
<summary>
<strong>retrieve children nodes from <code>ncbi</code></strong>
</summary>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># retrieve children taxa of the query organism stored in the correspondning database</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="fu">taxonomy</span>( <span class="at">organism =</span> <span class="st">&quot;Arabidopsis&quot;</span>, </span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>          <span class="at">db       =</span> <span class="st">&quot;ncbi&quot;</span>, </span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>          <span class="at">output   =</span> <span class="st">&quot;children&quot;</span> )</span></code></pre></div>
<details>
<summary>
Show output
</summary>
<pre><code>## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0</code></pre>
<pre><code>##    childtaxa_id
## 1       2608267
## 2       2486701
## 3       1837063
## 4       1547872
## 5       1328956
## 6       1240361
## 7        869751
## 8        869750
## 9        864766
## 10       412662
## 11       378006
## 12       347883
## 13       302551
## 14        97980
## 15        97979
## 16        81970
## 17        59690
## 18        59689
## 19        45251
## 20        45249
## 21        38785
## 22         3702
##                                                        childtaxa_name
## 1                            Arabidopsis arenosa x Arabidopsis lyrata
## 2                            Arabidopsis lyrata x Arabidopsis halleri
## 3                          Arabidopsis thaliana x Arabidopsis halleri
## 4                                               Arabidopsis umezawana
## 5  (Arabidopsis thaliana x Arabidopsis arenosa) x Arabidopsis suecica
## 6                          Arabidopsis thaliana x Arabidopsis arenosa
## 7         Arabidopsis thaliana x Arabidopsis halleri subsp. gemmifera
## 8                           Arabidopsis thaliana x Arabidopsis lyrata
## 9                                         Arabidopsis septentrionalis
## 10                                            Arabidopsis pedemontana
## 11                         Arabidopsis arenosa x Arabidopsis thaliana
## 12                                              Arabidopsis arenicola
## 13                                              Arabidopsis petrogena
## 14                                               Arabidopsis croatica
## 15                                            Arabidopsis cebennensis
## 16                                                Arabidopsis halleri
## 17                                             Arabidopsis kamchatica
## 18                                                 Arabidopsis lyrata
## 19                                               Arabidopsis neglecta
## 20                                                Arabidopsis suecica
## 21                                                Arabidopsis arenosa
## 22                                               Arabidopsis thaliana
##    childtaxa_rank
## 1         species
## 2         species
## 3         species
## 4         species
## 5         species
## 6         species
## 7         species
## 8         species
## 9         species
## 10        species
## 11        species
## 12        species
## 13        species
## 14        species
## 15        species
## 16        species
## 17        species
## 18        species
## 19        species
## 20        species
## 21        species
## 22        species</code></pre>
</details>
</details>
<details>
<summary>
<strong>retrieve children nodes from <code>itis</code></strong>
</summary>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co"># retrieve children taxa of the query organism stored in the correspondning database</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a><span class="fu">taxonomy</span>( <span class="at">organism =</span> <span class="st">&quot;Arabidopsis&quot;</span>, </span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>          <span class="at">db       =</span> <span class="st">&quot;itis&quot;</span>, </span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>          <span class="at">output   =</span> <span class="st">&quot;children&quot;</span> )</span></code></pre></div>
<details>
<summary>
Show output
</summary>
<pre><code>## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0
## ══  1 queries  ═══════════════</code></pre>
<pre><code>## ✔  Found:  Arabidopsis
## ══  Results  ═════════════════
## 
## • Total: 1 
## • Found: 1 
## • Not Found: 0</code></pre>
<pre><code>##    parentname parenttsn rankname             taxonname    tsn
## 1 Arabidopsis     23040  Species  Arabidopsis thaliana  23041
## 2 Arabidopsis     23040  Species Arabidopsis arenicola 823113
## 3 Arabidopsis     23040  Species   Arabidopsis arenosa 823130
## 4 Arabidopsis     23040  Species    Arabidopsis lyrata 823171</code></pre>
</details>
</details>
<p>These results allow us to choose <code>subject</code> organisms for
<a href="other-strata.html">Divergence Stratigraphy</a>.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
