<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2021-02-23" />

<title>Retrieving Taxonomic Information</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Retrieving Taxonomic Information</h1>
<h4 class="date">2021-02-23</h4>



<p>This vignette will introduce users to the retrieval of taxonomic information with <code>myTAI</code>. The <code>taxonomy()</code> function implemented in <code>myTAI</code> relies on the powerful package <a href="https://github.com/ropensci/taxize">taxize</a>. Nevertheless, taxonomic information retrieval has been customized for the <code>myTAI</code> standard and for organism specific information retrieval.</p>
<p>Specifically, the <code>taxonomy()</code> function implemented in <code>myTAI</code> can be used to classify genomes according to phylogenetic classification into Phylostrata (Phylostratigraphy) or to retrieve species specific taxonomic information when performing Divergence Stratigraphy (see <a href="Introduction.html">Introduction</a> for details).</p>
<p>For larger taxonomy queries it may be useful to create an NCBI Account and set up an <a href="https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/">ENTREZ API KEY</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># install.packages(c(&quot;taxize&quot;, &quot;usethis&quot;))</span></span>
<span id="cb1-2"><a href="#cb1-2"></a>taxize<span class="op">::</span><span class="kw">use_entrez</span>()</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co"># Create your key from your (brand-new) account&#39;s. </span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># After generating your key set it as ENTREZ_KEY in .Renviron.</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co"># ENTREZ_KEY=&#39;youractualkeynotthisstring&#39;</span></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co"># For that, use usethis::edit_r_environ()</span></span>
<span id="cb1-7"><a href="#cb1-7"></a>usethis<span class="op">::</span><span class="kw">edit_r_environ</span>()</span></code></pre></div>
<div id="taxonomic-information-retrieval" class="section level2">
<h2>Taxonomic Information Retrieval</h2>
<p>The <code>myTAI</code> package provides the <code>taxonomy()</code> function to retrieve taxonomic information.</p>
<p>In the following example we will obtain the taxonomic hierarchy of <code>Arabidopsis thaliana</code> from <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># retrieving the taxonomic hierarchy of &quot;Arabidopsis thaliana&quot;</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="co"># from NCBI Taxonomy</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>myTAI<span class="op">::</span><span class="kw">taxonomy</span>( <span class="dt">organism =</span> <span class="st">&quot;Arabidopsis thaliana&quot;</span>, </span>
<span id="cb2-4"><a href="#cb2-4"></a>          <span class="dt">db       =</span> <span class="st">&quot;ncbi&quot;</span>,</span>
<span id="cb2-5"><a href="#cb2-5"></a>          <span class="dt">output   =</span> <span class="st">&quot;classification&quot;</span> )</span></code></pre></div>
<pre><code>                   name         rank      id
1    cellular organisms      no rank  131567
2             Eukaryota superkingdom    2759
3         Viridiplantae      kingdom   33090
4          Streptophyta       phylum   35493
5        Streptophytina      no rank  131221
6           Embryophyta      no rank    3193
7          Tracheophyta      no rank   58023
8         Euphyllophyta      no rank   78536
9         Spermatophyta      no rank   58024
10        Magnoliophyta      no rank    3398
11      Mesangiospermae      no rank 1437183
12       eudicotyledons      no rank   71240
13           Gunneridae      no rank   91827
14         Pentapetalae      no rank 1437201
15               rosids     subclass   71275
16              malvids      no rank   91836
17          Brassicales        order    3699
18         Brassicaceae       family    3700
19           Camelineae        tribe  980083
20          Arabidopsis        genus    3701
21 Arabidopsis thaliana      species    3702</code></pre>
<p>The <code>organism</code> argument takes the scientific name of a query organism, the <code>db</code> argument specifies that database from which the corresponding taxonomic information shall be retrieved, e.g. <code>ncbi</code> (NCBI Taxonomy) and <code>itis</code> (Integrated Taxonomic Information System) and the <code>output</code> argument specifies the type of taxonomic information that shall be returned for the query organism, e.g. <code>classification</code>, <code>taxid</code>, or <code>children</code>.</p>
<p>The output of <code>classification</code> is a <code>data.frame</code> storing the taxonomic hierarchy of <code>Arabidopsis thaliana</code> starting with <code>cellular organisms</code> up to <code>Arabidopsis thaliana</code>. The first column stores the taxonomic name, the second column the taxonomic rank, and the third column the NCBI Taxonomy id for corresponding taxa.</p>
<p>Analogous <code>classification</code> information can be obtained from different databases.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># retrieving the taxonomic hierarchy of &quot;Arabidopsis thaliana&quot;</span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># from the Integrated Taxonomic Information System</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>myTAI<span class="op">::</span><span class="kw">taxonomy</span>( <span class="dt">organism =</span> <span class="st">&quot;Arabidopsis thaliana&quot;</span>, </span>
<span id="cb4-4"><a href="#cb4-4"></a>          <span class="dt">db       =</span> <span class="st">&quot;itis&quot;</span>,</span>
<span id="cb4-5"><a href="#cb4-5"></a>          <span class="dt">output   =</span> <span class="st">&quot;classification&quot;</span> )</span></code></pre></div>
<pre><code>              name          rank     id
1          Plantae       Kingdom 202422
2    Viridiplantae    Subkingdom 954898
3     Streptophyta  Infrakingdom 846494
4      Embryophyta Superdivision 954900
5     Tracheophyta      Division 846496
6  Spermatophytina   Subdivision 846504
7    Magnoliopsida         Class  18063
8          Rosanae    Superorder 846548
9      Brassicales         Order 822943
10    Brassicaceae        Family  22669
11     Arabidopsis         Genus  23040</code></pre>
<p>The <code>output</code> argument allows you to directly access taxonomy ids for a query organism or species.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co"># retrieving the taxonomy id of the query organism from NCBI Taxonomy</span></span>
<span id="cb6-2"><a href="#cb6-2"></a>myTAI<span class="op">::</span><span class="kw">taxonomy</span>( <span class="dt">organism =</span> <span class="st">&quot;Arabidopsis thaliana&quot;</span>, </span>
<span id="cb6-3"><a href="#cb6-3"></a>          <span class="dt">db       =</span> <span class="st">&quot;ncbi&quot;</span>, </span>
<span id="cb6-4"><a href="#cb6-4"></a>          <span class="dt">output   =</span> <span class="st">&quot;taxid&quot;</span> )</span></code></pre></div>
<pre><code>    id
1 3702</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># retrieving the taxonomy id of the query organism from Integrated Taxonomic Information Service</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>myTAI<span class="op">::</span><span class="kw">taxonomy</span>( <span class="dt">organism =</span> <span class="st">&quot;Arabidopsis&quot;</span>, </span>
<span id="cb8-3"><a href="#cb8-3"></a>          <span class="dt">db       =</span> <span class="st">&quot;itis&quot;</span>, </span>
<span id="cb8-4"><a href="#cb8-4"></a>          <span class="dt">output   =</span> <span class="st">&quot;taxid&quot;</span> )</span></code></pre></div>
<pre><code>    id
1 23040</code></pre>
<p>So far, the following data bases can be accesses to retrieve taxonomic information:</p>
<ul>
<li><code>db = &quot;itis&quot;</code> : Integrated Taxonomic Information Service</li>
<li><code>db = &quot;ncbi&quot;</code> : National Center for Biotechnology Information</li>
</ul>
</div>
<div id="retrieve-children-nodes" class="section level2">
<h2>Retrieve Children Nodes</h2>
<p>Another <code>output</code> supported by <code>taxonomy()</code> is <code>children</code> that returns the immediate children taxa for a query organism. This feature is useful to determine species relationships for quantifying recent evolutionary conservation with Divergence Stratigraphy.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># retrieve children taxa of the query organism stored in the correspondning database</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>myTAI<span class="op">::</span><span class="kw">taxonomy</span>( <span class="dt">organism =</span> <span class="st">&quot;Arabidopsis&quot;</span>, </span>
<span id="cb10-3"><a href="#cb10-3"></a>          <span class="dt">db       =</span> <span class="st">&quot;ncbi&quot;</span>, </span>
<span id="cb10-4"><a href="#cb10-4"></a>          <span class="dt">output   =</span> <span class="st">&quot;children&quot;</span> )</span></code></pre></div>
<pre><code>   childtaxa_id                                                     childtaxa_name childtaxa_rank
1       1547872                                              Arabidopsis umezawana        species
2       1328956 (Arabidopsis thaliana x Arabidopsis arenosa) x Arabidopsis suecica        species
3       1240361                         Arabidopsis thaliana x Arabidopsis arenosa        species
4        869750                          Arabidopsis thaliana x Arabidopsis lyrata        species
5        412662                                            Arabidopsis pedemontana        species
6        378006                         Arabidopsis arenosa x Arabidopsis thaliana        species
7        347883                                              Arabidopsis arenicola        species
8        302551                                              Arabidopsis petrogena        species
9         97980                                               Arabidopsis croatica        species
10        97979                                            Arabidopsis cebennensis        species
11        81970                                                Arabidopsis halleri        species
12        59690                                             Arabidopsis kamchatica        species
13        59689                                                 Arabidopsis lyrata        species
14        45251                                               Arabidopsis neglecta        species
15        45249                                                Arabidopsis suecica        species
16        38785                                                Arabidopsis arenosa        species
17         3702                                               Arabidopsis thaliana        species</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># retrieve children taxa of the query organism stored in the correspondning database</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>myTAI<span class="op">::</span><span class="kw">taxonomy</span>( <span class="dt">organism =</span> <span class="st">&quot;Arabidopsis&quot;</span>, </span>
<span id="cb12-3"><a href="#cb12-3"></a>          <span class="dt">db       =</span> <span class="st">&quot;itis&quot;</span>, </span>
<span id="cb12-4"><a href="#cb12-4"></a>          <span class="dt">output   =</span> <span class="st">&quot;children&quot;</span> )</span></code></pre></div>
<pre><code>   parentname parenttsn rankname             taxonname    tsn
1 Arabidopsis     23040  Species  Arabidopsis thaliana  23041
2 Arabidopsis     23040  Species Arabidopsis arenicola 823113
3 Arabidopsis     23040  Species   Arabidopsis arenosa 823130
4 Arabidopsis     23040  Species    Arabidopsis lyrata 823171</code></pre>
<p>These results allow us to choose <code>subject</code> organisms for <a href="https://github.com/drostlab/orthologr/blob/master/vignettes/divergence_stratigraphy.Rmd">Divergence Stratigraphy</a>.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
