<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Changelog"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.3.0.9006</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/myTAI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/other-strata.html">Other evolutionary and expression indices</a>
    </li>
    <li>
      <a href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a>
    </li>
    <li>
      <a href="../articles/phylostratigraphy.html">Gene age inference</a>
    </li>
    <li>
      <a href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a>
    </li>
    <li>
      <a href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-stats.html">Statistical testing with myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-transform.html">Transforming dataset for myTAI</a>
    </li>
    <li>
      <a href="../articles/taxonomy.html">Obtaining taxonomy information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="2.3.0.9006" id="mytai-2309006">myTAI 2.3.0.9006<a class="anchor" aria-label="anchor" href="#mytai-2309006"></a></h2>
<p>This update has focused on improving the single-cell phyloset functionality.</p>
<p>The single-cell phylo expression object no longer depends on Seurat. You can construct the ScPhyloExpressionSet either from a count matrix (sparse or dense), using <code>ScPhyloExpressionSet_from_matrix</code>, or from a Seurat object, using <code>ScPhyloExpressionSet_from_seurat</code>. For consistency, one can use <code>BulkPhyloExpressionSet_from_df</code> instead of <code>as_BulkPhyloExpressionSet</code>.</p>
<p>One key functionality of the single-cell object is the ability to switch between different identities when plotting (equivalent to the <code><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Seurat::Idents</a></code> functionality). This is done by setting the <code>::identities_label</code> property of the object. The <code>::available_idents</code> property can be used to see what options the user has in setting the current identity. By setting <code>::idents_colours[[]]</code>, the user can choose a colour pallette for the different identities when plotting, which are saved across different plotting calls.</p>
<p>The computation of TAI values for single cell is now cached. Moreover, we have readded the C++ accelerated code for the computation of TAI, which upon profiling shows to be faster than the R version when handling more than 100000 cells (an adaptive function chooses the appropriate implementation for the object size).</p>
<p>Some of the plotting functionality had been improved and more options were added for plotting (e.g. plot_gene_heatmap now allows for passing a custom colour mapping for the rows (genes), instead of colouring them by their strata; plot_signature for single cell should be more readable).</p>
<p>Bug fixes: - validation of S7 objects now works properly - printing of object information now works properly (instead of dumping all the properties information)</p>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.2.0.9006" id="mytai-2209006">myTAI 2.2.0.9006<a class="anchor" aria-label="anchor" href="#mytai-2209006"></a></h2>
<p>Most importantly, new vignettes were added and the website has been updated. Many thanks to <a href="https://github.com/LotharukpongJS" class="external-link">@LotharukpongJS</a></p>
<p>Functionality wise:</p>
<ul><li>added more input validation across functions</li>
<li>improved plotting in <code>destroy_pattern</code>
</li>
<li>renamed <code>tfPS</code> to <code>tf_PS</code> and fixed bug which prevented strata transformations from happening (PhyloExpressionSet now has an extra field <a href="https://github.com/strata_values" class="external-link">@strata_values</a> which keeps track of the phylostratum values)</li>
<li>added back standard deviation ribbon to plot_signature</li>
<li>finer control over cell identity selection in the sc object</li>
<li>fixed bugs in some of the plotting functions, such as plot_signature_multiple showing the colours in reverse, plot_genes_heatmap not working when given just one gene etc.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.1.0.9000" id="mytai-2109000">myTAI 2.1.0.9000<a class="anchor" aria-label="anchor" href="#mytai-2109000"></a></h2>
<div class="section level3">
<h3 id="major-refactoring-and-new-features-2-1-0-9000">Major Refactoring and New Features<a class="anchor" aria-label="anchor" href="#major-refactoring-and-new-features-2-1-0-9000"></a></h3>
<ul><li>
<strong>Single-cell support:</strong> New <code>ScPhyloExpressionSet</code> S7 class for single-cell data, alongside <code>BulkPhyloExpressionSet</code> for bulk data.</li>
<li>
<strong>Property renaming:</strong> <code>conditions</code> → <code>identities</code> (and <code>conditions_label</code> → <code>identities_label</code>), <code>counts</code>/<code>counts_collapsed</code> → <code>expression</code>/<code>expression_collapsed</code>.</li>
<li><strong>All core logic and plotting now robust to both bulk and single-cell objects.</strong></li>
</ul></div>
<div class="section level3">
<h3 id="function-renaming-and-prefixes-2-1-0-9000">Function Renaming and Prefixes<a class="anchor" aria-label="anchor" href="#function-renaming-and-prefixes-2-1-0-9000"></a></h3>
<ul><li>
<strong>Statistical tests:</strong> All core test functions now use the <code>stat_</code> prefix:
<ul><li>
<code>flatline_test()</code> → <code><a href="../reference/stat_flatline_test.html">stat_flatline_test()</a></code>
</li>
<li>
<code>early_conservation_test()</code> → <code><a href="../reference/stat_early_conservation_test.html">stat_early_conservation_test()</a></code>
</li>
<li>
<code>late_conservation_test()</code> → <code><a href="../reference/stat_late_conservation_test.html">stat_late_conservation_test()</a></code>
</li>
<li>
<code>reductive_hourglass_test()</code> → <code><a href="../reference/stat_reductive_hourglass_test.html">stat_reductive_hourglass_test()</a></code>
</li>
<li>
<code>reverse_hourglass_test()</code> → <code><a href="../reference/stat_reverse_hourglass_test.html">stat_reverse_hourglass_test()</a></code>
</li>
<li>
<code>pairwise_test()</code> → <code><a href="../reference/stat_pairwise_test.html">stat_pairwise_test()</a></code>
</li>
<li>
<code>generic_conservation_test()</code> → <code><a href="../reference/stat_generic_conservation_test.html">stat_generic_conservation_test()</a></code>
</li>
<li>
<code>generate_conservation_txis()</code> → <code><a href="../reference/stat_generate_conservation_txis.html">stat_generate_conservation_txis()</a></code>
</li>
</ul></li>
<li>
<strong>Gene selection/filtering:</strong> All gene selection/filtering functions now use the <code>genes_</code> prefix:
<ul><li>
<code>top_expression_genes()</code> → <code><a href="../reference/genes_top_mean.html">genes_top_mean()</a></code>
</li>
<li>
<code>top_variance_genes()</code> → <code><a href="../reference/genes_top_variance.html">genes_top_variance()</a></code>
</li>
<li>
<code>lowly_expressed_genes()</code> → <code><a href="../reference/genes_lowly_expressed.html">genes_lowly_expressed()</a></code>
</li>
<li>
<code>filter_dyn_expr()</code> → <code><a href="../reference/genes_filter_dynamic.html">genes_filter_dynamic()</a></code>
</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="other-changes-2-1-0-9000">Other Changes<a class="anchor" aria-label="anchor" href="#other-changes-2-1-0-9000"></a></h3>
<ul><li>Many internal and Rd files renamed for consistency (e.g., <code>gene_patterns.R</code> → <code>genes_patterns.R</code>, <code>S7_utils.R</code> → <code>utils_S7.R</code>).</li>
<li>Deprecated/legacy files removed: <code>expression_utils.R</code>, <code>single_cell.R</code>.</li>
<li>All documentation and examples updated to use new function/property names.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="2.0.0.9000" id="mytai-2009000">myTAI 2.0.0.9000<a class="anchor" aria-label="anchor" href="#mytai-2009000"></a></h2>
<div class="section level3">
<h3 id="major-changes-2-0-0-9000">Major Changes<a class="anchor" aria-label="anchor" href="#major-changes-2-0-0-9000"></a></h3>
<div class="section level4">
<h4 id="new-s7-class-system-2-0-0-9000">New S7 Class System<a class="anchor" aria-label="anchor" href="#new-s7-class-system-2-0-0-9000"></a></h4>
<ul><li>Migrated from traditional R data structures to modern S7 classes</li>
<li>New <code>PhyloExpressionSet</code> S7 class replaces the old data.frame-based format</li>
<li>New <code>TestResult</code> S7 class for standardized storage of statistical test results</li>
<li>Computed properties automatically calculate derived values like TXI, pTXI, conditions, etc.</li>
<li>Built-in data validation and type checking through S7 properties</li>
<li>Improved replicate data handling with automatic collapsing</li>
</ul></div>
<div class="section level4">
<h4 id="function-name-modernization-2-0-0-9000">Function Name Modernization<a class="anchor" aria-label="anchor" href="#function-name-modernization-2-0-0-9000"></a></h4>
<p>Function names have been updated to use snake_case convention:</p>
<div class="section level5">
<h5 id="statistical-tests-old--new-2-0-0-9000">Statistical Tests (old → new)<a class="anchor" aria-label="anchor" href="#statistical-tests-old--new-2-0-0-9000"></a></h5>
<ul><li>
<code>FlatLineTest()</code> → <code>flatline_test()</code>
</li>
<li>
<code>ReductiveHourglassTest()</code> → <code>reductive_hourglass_test()</code>
</li>
<li>
<code>EarlyConservationTest()</code> → <code>early_conservation_test()</code>
</li>
<li>
<code>LateConservationTest()</code> → <code>late_conservation_test()</code>
</li>
<li>
<code>ReverseHourglassTest()</code> → <code>reverse_hourglass_test()</code>
</li>
<li>
<code>PairwiseTest()</code> → <code>pairwise_test()</code>
</li>
</ul></div>
<div class="section level5">
<h5 id="visualization-functions-old--new-2-0-0-9000">Visualization Functions (old → new)<a class="anchor" aria-label="anchor" href="#visualization-functions-old--new-2-0-0-9000"></a></h5>
<ul><li>
<code>PlotSignature()</code> → <code><a href="../reference/plot_signature.html">plot_signature()</a></code>
</li>
<li>
<code>PlotPattern()</code> → <code><a href="../reference/plot_signature.html">plot_signature()</a></code>
</li>
<li>
<code>PlotContribution()</code> → <code><a href="../reference/plot_contribution.html">plot_contribution()</a></code>
</li>
<li>
<code>PlotDistribution()</code> → <code><a href="../reference/plot_distribution_strata.html">plot_distribution_strata()</a></code>
</li>
<li>
<code>PlotCategoryExpr()</code> → <code><a href="../reference/plot_strata_expression.html">plot_strata_expression()</a></code>
</li>
<li>
<code>PlotRE()</code> → <code><a href="../reference/plot_relative_expression_line.html">plot_relative_expression_line()</a></code>
</li>
<li>
<code>PlotBarRE()</code> → <code><a href="../reference/plot_relative_expression_bar.html">plot_relative_expression_bar()</a></code>
</li>
<li>
<code>PlotGeneSet()</code> → <code><a href="../reference/plot_gene_profiles.html">plot_gene_profiles()</a></code>
</li>
<li>
<code>PlotMeans</code>, <code>PlotVars</code>, <code>PlotMedians</code> → `plot_strata_expression(aggregate_FUN=“mean”/“var”/“median”)</li>
<li>
<code>PlotSignatureMultiple()</code> → <code><a href="../reference/plot_signature_multiple.html">plot_signature_multiple()</a></code>
</li>
<li>
<code>PlotSignatureTransformed()</code> → <code><a href="../reference/plot_signature_transformed.html">plot_signature_transformed()</a></code>
</li>
<li>
<code>PlotSignatureGeneQuantiles()</code> → <code><a href="../reference/plot_signature_gene_quantiles.html">plot_signature_gene_quantiles()</a></code>
</li>
</ul></div>
<div class="section level5">
<h5 id="utility-functions-old--new-2-0-0-9000">Utility Functions (old → new)<a class="anchor" aria-label="anchor" href="#utility-functions-old--new-2-0-0-9000"></a></h5>
<ul><li>
<code><a href="../reference/TAI.html">TAI()</a></code> → Computed property of PhyloExpressionSet (still accessible via <code><a href="../reference/TAI.html">TAI()</a></code>)</li>
<li>
<code><a href="../reference/TDI.html">TDI()</a></code> → Computed property of PhyloExpressionSet (still accessible via <code><a href="../reference/TDI.html">TDI()</a></code>)</li>
<li>
<code><a href="../reference/TEI.html">TEI()</a></code> → Computed property of PhyloExpressionSet (still accessible via <code><a href="../reference/TEI.html">TEI()</a></code>)</li>
<li>
<code><a href="../reference/TPI.html">TPI()</a></code> → Computed property of PhyloExpressionSet (still accessible via <code><a href="../reference/TPI.html">TPI()</a></code>)</li>
<li>
<code>pTAI()</code>, <code>pTDI</code> → <code><a href="../reference/sTXI.html">sTXI()</a></code> (generalized for all transcriptomic indices)</li>
<li>
<code>pMatrix()</code> → <code><a href="../reference/pTXI.html">pTXI()</a></code>
</li>
<li>
<code>CollapseReplicates()</code> → <code>collapse</code>, Built into PhyloExpressionSet constructor</li>
<li>
<code>Expressed()</code> → <code>lowly_expressed_genes()</code>
</li>
<li>
<code>MatchMap()</code> → <code><a href="../reference/match_map.html">match_map()</a></code>
</li>
<li>
<code>SelectGeneSet()</code> → <code><a href="../reference/select_genes.html">select_genes()</a></code>
</li>
<li>
<code>TopExpressionGenes()</code> → <code>top_expression_genes()</code>
</li>
<li>
<code>TopVarianceGenes()</code> → <code>top_variance_genes()</code>
</li>
<li>
<code>REMatrix()</code> → <code><a href="../reference/rel_exp_matrix.html">rel_exp_matrix()</a></code>
</li>
<li>
<code>RE()</code> → <code><a href="../reference/relative_expression.html">relative_expression()</a></code>
</li>
<li>
<code>omitMatrix()</code> → <code><a href="../reference/omit_matrix.html">omit_matrix()</a></code>
</li>
<li>
<code>is.ExpressionSet()</code> → Built into S7 validation</li>
<li>
<code><a href="../reference/age.apply.html">age.apply()</a></code> → <code><a href="../reference/age.apply.html">age.apply()</a></code> (unchanged)</li>
<li>
<code><a href="../reference/tf.html">tf()</a></code> → <code><a href="../reference/tf.html">tf()</a></code> or <code><a href="../reference/transform_counts.html">transform_counts()</a></code>
</li>
<li>
<code>tfPS()</code> → <code>tfPS()</code> (unchanged)</li>
<li>
<code>tfStability()</code> → <code><a href="../reference/tf_stability.html">tf_stability()</a></code>
</li>
<li>
<code><a href="../reference/taxid.html">taxid()</a></code> → <code><a href="../reference/taxid.html">taxid()</a></code> (unchanged)</li>
</ul></div>
</div>
<div class="section level4">
<h4 id="new-functions-2-0-0-9000">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-2-0-0-9000"></a></h4>
<ul><li>
<code><a href="../reference/destroy_pattern.html">destroy_pattern()</a></code>: Apply GATAI algorithm to identify pattern-contributing genes</li>
<li>
<code><a href="../reference/plot_signature_multiple.html">plot_signature_multiple()</a></code>: Plot multiple signatures on the same plot</li>
<li>
<code><a href="../reference/plot_signature_gene_quantiles.html">plot_signature_gene_quantiles()</a></code>: Plot signature with gene expression quantiles</li>
<li>
<code><a href="../reference/plot_signature_transformed.html">plot_signature_transformed()</a></code>: Plot signatures with different transformations</li>
<li>
<code><a href="../reference/plot_sample_space.html">plot_sample_space()</a></code>: Visualize sample relationships using PCA or UMAP</li>
<li>
<code><a href="../reference/plot_mean_var.html">plot_mean_var()</a></code>: Mean-variance relationship plots</li>
<li>
<code><a href="../reference/plot_gene_profiles.html">plot_gene_profiles()</a></code>: Individual gene expression profiles</li>
<li>
<code><a href="../reference/plot_distribution_expression.html">plot_distribution_expression()</a></code>: Expression distribution plots</li>
<li>
<code><a href="../reference/plot_distribution_pTAI.html">plot_distribution_pTAI()</a></code>: Partial TXI distribution plots</li>
<li>
<code><a href="../reference/plot_distribution_pTAI_qqplot.html">plot_distribution_pTAI_qqplot()</a></code>: Q-Q plots for pTXI distributions</li>
<li>
<code><a href="../reference/plot_distribution_strata.html">plot_distribution_strata()</a></code>: Phylostrata distribution plots</li>
<li>
<code><a href="../reference/plot_gene_heatmap.html">plot_gene_heatmap()</a></code>: Gene expression heatmaps</li>
<li>
<code><a href="../reference/plot_gene_space.html">plot_gene_space()</a></code>: Gene space visualization</li>
<li>
<code><a href="../reference/plot_cullen_frey.html">plot_cullen_frey()</a></code>: Cullen-Frey plots for distribution assessment</li>
<li>
<code><a href="../reference/plot_null_txi_sample.html">plot_null_txi_sample()</a></code>: Null TXI sample plots</li>
<li>
<code>as_PhyloExpressionSet()</code>: Convert data to PhyloExpressionSet S7 object</li>
<li>
<code>get_sc_TAI()</code>: Single-cell TAI computation for Seurat objects=</li>
<li>
<code><a href="../reference/diagnose_test_robustness.html">diagnose_test_robustness()</a></code>: Diagnose statistical test robustness</li>
<li>
<code><a href="../reference/remove_genes.html">remove_genes()</a></code>: Remove genes from PhyloExpressionSet</li>
<li>
<code><a href="../reference/PS_colours.html">PS_colours()</a></code>: Generate phylostratum color palettes</li>
<li>
<code><a href="../reference/ConservationTestResult.html">ConservationTestResult()</a></code>: S7 class for conservation test results</li>
<li>
<code><a href="../reference/TestResult.html">TestResult()</a></code>: S7 class for statistical test results</li>
</ul></div>
<div class="section level4">
<h4 id="enhanced-features-2-0-0-9000">Enhanced Features<a class="anchor" aria-label="anchor" href="#enhanced-features-2-0-0-9000"></a></h4>
<ul><li>Improved performance with parallelized C++ functions using RcppArmadillo</li>
<li>All plots now use ggplot2 with consistent styling</li>
<li>Comprehensive unit tests</li>
<li>Updated to use modern R packages (ggplot2, dplyr, etc.)</li>
</ul></div>
<div class="section level4">
<h4 id="breaking-changes-2-0-0-9000">Breaking Changes<a class="anchor" aria-label="anchor" href="#breaking-changes-2-0-0-9000"></a></h4>
<ul><li>Function names have been updated to snake_case convention</li>
<li>PhyloExpressionSet is now an S7 object instead of a data.frame</li>
<li>Function signatures have been updated for consistency</li>
<li>Some deprecated functions have been removed</li>
<li>New package dependencies: S7, RcppArmadillo</li>
</ul></div>
<div class="section level4">
<h4 id="migration-guide-2-0-0-9000">Migration Guide<a class="anchor" aria-label="anchor" href="#migration-guide-2-0-0-9000"></a></h4>
<p>To migrate from myTAI 1.x to 2.0:</p>
<ol style="list-style-type: decimal"><li>
<p><strong>Convert data format</strong>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Old format (data.frame)</span></span>
<span><span class="va">old_phyex</span> <span class="op">&lt;-</span> <span class="va">PhyloExpressionSetExample</span></span>
<span></span>
<span><span class="co"># New format (S7 object)</span></span>
<span><span class="va">new_phyex</span> <span class="op">&lt;-</span> <span class="fu">as_PhyloExpressionSet</span><span class="op">(</span><span class="va">old_phyex</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p><strong>Update function calls</strong>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Old syntax</span></span>
<span><span class="fu">PlotSignature</span><span class="op">(</span><span class="va">phyex_set</span>, TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># New syntax</span></span>
<span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">phyex_set</span>, conservation_test <span class="op">=</span> <span class="va">flatline_test</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p><strong>Access computed properties</strong>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Old syntax</span></span>
<span><span class="va">tai_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TAI.html">TAI</a></span><span class="op">(</span><span class="va">phyex_set</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># New syntax</span></span>
<span><span class="va">tai_values</span> <span class="op">&lt;-</span> <span class="va">phyex_set</span><span class="op">@</span><span class="va">TXI</span></span>
<span><span class="co"># or</span></span>
<span><span class="va">tai_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TAI.html">TAI</a></span><span class="op">(</span><span class="va">phyex_set</span><span class="op">)</span></span></code></pre></div>
</li>
</ol></div>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.2.9000" id="mytai-1029000">myTAI 1.0.2.9000<a class="anchor" aria-label="anchor" href="#mytai-1029000"></a></h2>
<div class="section level3">
<h3 id="new-functions-1-0-2-9000">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-1-0-2-9000"></a></h3>
<ul><li>New function <code>tfPS()</code> : Perform transformation of phylostratum values, analogous to <code>PS()</code> which transforms expression levels. Currently, <code>tfPS()</code> supports quantile rank transformation.</li>
<li>New function <code>PairwiseTest()</code> : Statistically evaluate the pairwise difference in the phylotranscriptomic pattern between two contrasts based on  or  computations.</li>
<li>New function <code>pairScore()</code> : Compute pairwise difference in TAI (or TDI) score.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-and-issue-fixes-1-0-2-9000">Bug and Issue Fixes<a class="anchor" aria-label="anchor" href="#bug-and-issue-fixes-1-0-2-9000"></a></h3>
<ul><li>Removed <code>taxonomy()</code> and references to it due to the deprecation of <code>taxize()</code>. This is needed for CRAN submission.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.1.9000" id="mytai-1019000">myTAI 1.0.1.9000<a class="anchor" aria-label="anchor" href="#mytai-1019000"></a></h2>
<div class="section level3">
<h3 id="new-functions-1-0-1-9000">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-1-0-1-9000"></a></h3>
<ul><li><p>New function <code>PlotSignatureTransformed()</code> : Plot evolutionary signatures across transcriptomes and RNA-seq transformations</p></li>
<li><p>New function <code>tfStability()</code>: Perform Permutation Tests Under Different Transformations (to test the robustness of the p-values of a given test (e.g. <code>FlatLineTest()</code>, <code>ReductiveHourglassTest()</code>, <code>ReverseHourglassTest()</code>, <code>EarlyConservationTest()</code> and <code>LateConservationTest()</code>) to expression data transformations)</p></li>
<li><p>New function <code>LateConservationTest()</code> : Perform Reductive Late Conservation Test (to test for a high-mid-low (or high-high-low) TAI or TDI pattern)</p></li>
<li><p>New function <code>lcScore()</code> : Compute the Hourglass Score for the LateConservationTest</p></li>
<li><p>internal <code>rcpp</code> functions are now using Eigen to automatically enable parallelization</p></li>
</ul></div>
<div class="section level3">
<h3 id="new-features-1-0-1-9000">New Features<a class="anchor" aria-label="anchor" href="#new-features-1-0-1-9000"></a></h3>
<ul><li><p><code>PlotSignature()</code> updated to be able to perform the <code>TestStatistic = "LateConservationTest"</code>.</p></li>
<li><p><code>PlotSignature()</code> now prints p-value as a subtitle rather than via <code><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">ggplot2::annotate()</a></code>.</p></li>
<li><p><code><a href="../reference/tf.html">tf()</a></code> now has a <code>pseudocount</code> parameter, which is useful for performing logarithmic transformations when there are genes with 0 counts.</p></li>
<li><p><code><a href="../reference/tf.html">tf()</a></code> now supports <code>vst</code> and <code>rlog</code> transformations from <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" class="external-link">DESeq2</a>.</p></li>
<li><p><code><a href="../reference/tf.html">tf()</a></code> now has an <code>intergerise</code> parameter, which is needed when applying <code>vst</code> or <code>rlog</code> transformations.</p></li>
<li><p><code><a href="../reference/tf.html">tf()</a></code> updated documentation for performing rank transformation, which assigns ranks to the gene expression values within each stage, based on their relative positions compared to other values.</p></li>
<li><p>Improvements to existing test functions (<code>ecScore()</code>, <code>rhScore()</code> and <code>reversehourglassScore()</code>) to give a message when the phylotranscriptomic pattern is unlikely to follow the test statistics.</p></li>
<li><p><code>FlatLineTest()</code> - newly returns the ks test statistics for the fitting of gamma</p></li>
<li><p><code>FlatLineTest()</code> - improved fitting</p></li>
<li><p><code>FlatLineTest()</code> - cpp functions are newly parallelized and progress bar is implemented for the computation of permutations</p></li>
</ul></div>
<div class="section level3">
<h3 id="bug-and-issue-fixes-1-0-1-9000">Bug and Issue Fixes<a class="anchor" aria-label="anchor" href="#bug-and-issue-fixes-1-0-1-9000"></a></h3>
<ul><li>Some changes to remove errors and warnings from <code><a href="https://devtools.r-lib.org/reference/test.html" class="external-link">devtools::test()</a></code> and <code><a href="https://devtools.r-lib.org/reference/check.html" class="external-link">devtools::check()</a></code>, when building this package, which has been accumulated from previous updates.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.0" id="mytai-100">myTAI 1.0.0<a class="anchor" aria-label="anchor" href="#mytai-100"></a></h2>
<div class="section level3">
<h3 id="new-functions-1-0-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-1-0-0"></a></h3>
<ul><li>New function <code><a href="../reference/TEI.html">TEI()</a></code>: Compute the Transcriptome Evolutionary Index</li>
<li>New function <code>pMatrixTEI</code>: Compute Partial Transcriptome Evolutionary Index (TEI) Values</li>
<li>New function <code>pStrataTEI</code>: Compute Partial Transcriptome Evolutionary Index (TEI) Strata Values</li>
<li>New function <code>bootTEI</code>: Compute a Permutation Matrix of Transcriptome Evolutionary Index (TEI)</li>
<li>new internal <code>rcpp</code> functions to support parallel C++ computations for <code><a href="../reference/TEI.html">TEI()</a></code>, <code>pMatrixTEI()</code>, <code>pStrataTEI()</code>, <code>bootTEI()</code>
</li>
</ul></div>
<div class="section level3">
<h3 id="new-features-1-0-0">New Features<a class="anchor" aria-label="anchor" href="#new-features-1-0-0"></a></h3>
<ul><li>
<code>CollapseReplicates()</code> now returns <code>tibble</code> objects</li>
<li>
<code>PlotCategoryExpr()</code> received a new argument <code>y.ticks</code>
</li>
</ul></div>
<div class="section level3">
<h3 id="bug-and-issue-fixes-1-0-0">Bug and Issue Fixes<a class="anchor" aria-label="anchor" href="#bug-and-issue-fixes-1-0-0"></a></h3>
<ul><li>
<code>CollapseFromTo()</code> now has an exception when a replicate number <code>1</code> is passed to the function -&gt; previously this would cause an error to occur</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.9.3" id="mytai-093">myTAI 0.9.3<small>2021-02-24</small><a class="anchor" aria-label="anchor" href="#mytai-093"></a></h2>
<ul><li>removing the depreciated <code>std::random_shuffle()</code> function to sample <code>plylostratum</code> or <code>divergence stratum</code> columns and replacing it with <code>std::shuffle()</code>. See full discussion <a href="https://meetingcpp.com/blog/items/stdrandom_shuffle-is-deprecated.html" class="external-link">here</a>.</li>
<li>removing depreciated function calls such as <code><a href="https://dplyr.tidyverse.org/reference/funs.html" class="external-link">dplyr::funs()</a></code> and <code><a href="https://tibble.tidyverse.org/reference/is.tibble.html" class="external-link">tibble::is.tibble()</a></code>
</li>
<li>updated unit tests</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.9.2" id="mytai-092">myTAI 0.9.2<small>2020-01-10</small><a class="anchor" aria-label="anchor" href="#mytai-092"></a></h2>
<ul><li>changing maintainer email-address</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.9.1" id="mytai-091">myTAI 0.9.1<small>2019-03-10</small><a class="anchor" aria-label="anchor" href="#mytai-091"></a></h2>
<ul><li>fixing a unit test that uses <code>set.seed(123)</code> which causes an error in the new R version <code>3.6.0</code> due to the switch from a <code>non-uniform "Rounding" sampler</code> to a <code>"Rejection" sampler</code> in the new R version; the corresponding unit test <code>test-PlotEnrichment.R</code> was adjusted accordingly. Here the CRAN statement:</li>
</ul><blockquote>
<p>Note that this ensures using the (old) non-uniform “Rounding” sampler for all 3.x versions of R, and does not add an R version dependency. Note also that the new “Rejection” sampler which R will use from 3.6.0 onwards by default is definitely preferable over the old one, so that the above should really only be used as a temporary measure for reproduction of the previous behavior (and the run time tests relying on it).</p>
</blockquote>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.9.0" id="mytai-090">myTAI 0.9.0<small>2019-02-06</small><a class="anchor" aria-label="anchor" href="#mytai-090"></a></h2>
<div class="section level4">
<h4 id="new-functions-0-9-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-9-0"></a></h4>
<ul><li><p>new function <code>ReverseHourglassTest()</code> to perform a <code>Reverse Hourglass Test</code>. The Reverse Hourglass Test aims to statistically evaluate the existence of a reverse hourglass pattern based on TAI or TDI computations. The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) does follow an hourglass like shape. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does rather follow a reverse hourglass (low-high-low) shape.</p></li>
<li><p>new function <code>reversehourglassScore()</code> for computing the <code>Reverse Hourglass Score</code> for the <code>Reverse Hourglass Test</code></p></li>
</ul></div>
<div class="section level4">
<h4 id="updated-functionality-0-9-0">Updated Functionality<a class="anchor" aria-label="anchor" href="#updated-functionality-0-9-0"></a></h4>
<ul><li>function <code>PlotSignature()</code> receives a new <code>TestStatistic</code> (<code>TestStatistic = "ReverseHourglassTest"</code>) to perform a <code>revserse hourglass test</code> (= testing the significance of a low-high-low pattern)</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.0" id="mytai-080">myTAI 0.8.0<small>2018-05-23</small><a class="anchor" aria-label="anchor" href="#mytai-080"></a></h2>
<div class="section level4">
<h4 id="updated-functionality-0-8-0">Updated Functionality<a class="anchor" aria-label="anchor" href="#updated-functionality-0-8-0"></a></h4>
<ul><li>fix remaining issues when input is a <code>tibble</code>
</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.0" id="mytai-070">myTAI 0.7.0<small>2018-04-11</small><a class="anchor" aria-label="anchor" href="#mytai-070"></a></h2>
<div class="section level3">
<h3 id="updates-0-7-0">Updates<a class="anchor" aria-label="anchor" href="#updates-0-7-0"></a></h3>
<div class="section level4">
<h4 id="new-functions-0-7-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-7-0"></a></h4>
<ul><li>new function <code>PlotCIRatio()</code> to compute and visualize TAI/TDI etc patters using bootstrapping and confidence intervals (contributed by <a href="https://github.com/ljljolinq1010" class="external-link">@ljljolinq1010</a>)</li>
</ul></div>
<div class="section level4">
<h4 id="update-functionality-0-7-0">Update Functionality<a class="anchor" aria-label="anchor" href="#update-functionality-0-7-0"></a></h4>
<ul><li>all functions can now handle <code>tibble</code> data as input -&gt; before there were errors thrown when input data wasn’t in strict <code>data.frame</code> format</li>
</ul></div>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.6.0" id="mytai-060">myTAI 0.6.0<small>2017-07-03</small><a class="anchor" aria-label="anchor" href="#mytai-060"></a></h2>
<div class="section level4">
<h4 id="updates-0-6-0">Updates<a class="anchor" aria-label="anchor" href="#updates-0-6-0"></a></h4>
<ul><li><p><code>is.ExpressionSet()</code> now prints out more detailed error messages when <a href="https://github.com/drostlab/myTAI/blob/master/vignettes/Introduction.Rmd" class="external-link">ExpressionSet format</a> is violated</p></li>
<li><p>adapt <code>PlotContribution()</code> to new version of <code>dplyr</code> where <code><a href="https://dplyr.tidyverse.org/reference/summarise_each.html" class="external-link">summarise_each()</a></code> is deprecated.</p></li>
</ul><p>Error message occurring after new <code>dplyr</code> release was:</p>
<ol start="2" style="list-style-type: decimal"><li>Failure: <code>PlotContribution()</code> works properly with DivergenceExpressionSet input… (<a href="https://github.com/test-PlotContribution" class="external-link">@test-PlotContribution</a>.R#16) PlotContribution(DivergenceExpressionSetExample, legendName = “DS”) produced messages.</li>
</ol><p><code><a href="https://dplyr.tidyverse.org/reference/summarise_each.html" class="external-link">summarise_each()</a></code> is deprecated. Use <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_all()</a></code>, <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_at()</a></code> or <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_if()</a></code> instead. To map <code>funs</code> over all variables, use <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_all()</a></code> <code><a href="https://dplyr.tidyverse.org/reference/summarise_each.html" class="external-link">summarise_each()</a></code> is deprecated. Use <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_all()</a></code>, <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_at()</a></code> or <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_if()</a></code> instead. To map <code>funs</code> over all variables, use <code><a href="https://dplyr.tidyverse.org/reference/summarise_all.html" class="external-link">summarise_all()</a></code></p>
<p>Is now fixed.</p>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.0" id="mytai-050">myTAI 0.5.0<small>2017-03-14</small><a class="anchor" aria-label="anchor" href="#mytai-050"></a></h2>
<div class="section level4">
<h4 id="new-functions-0-5-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-5-0"></a></h4>
<ul><li><p>new function <code>PlotSignature()</code> allows users to plot evolutionary signatures across transcriptomes (based on ggplot2 -&gt; new main visualization function aiming to replace the <code>PlotPattern()</code> function)</p></li>
<li><p>new function <code><a href="../reference/TPI.html">TPI()</a></code> allows users to compute the Transcriptome Polymorphism Index introduced by <code>Gossmann et al., 2015</code>.</p></li>
<li><p>new function <code>PlotMedians()</code> allows users to compute and visualize the median expression of all age categories</p></li>
<li><p>new function <code>PlotVars()</code> allows users to compute and visualize the expression variance of all age categories</p></li>
</ul></div>
<div class="section level4">
<h4 id="updates-0-5-0">Updates<a class="anchor" aria-label="anchor" href="#updates-0-5-0"></a></h4>
<ul><li><p><code>PlotContribution()</code> is now based on ggplot2 and loses base graphics arguments</p></li>
<li><p>now R/RcppExports.R and src/rcpp_funcs.cpp are included in the package due to previous compilation problems (see also <a href="https://stackoverflow.com/questions/34585560/travis-ci-r-package-error-in-documentation" class="external-link">stackoverflow discussion</a>)</p></li>
<li><p><code>MatchMap()</code> is now based on <code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">dplyr::inner_join()</a></code> to match age category table with a gene expression dataset</p></li>
<li><p><code>PlotCorrelation()</code> has been extended and optimized for producing high publication quality plots</p></li>
<li><p><code>PlotMeans()</code> is now based on ggplot2 and lost all base graphics arguments.</p></li>
<li><p><code>PlotRE()</code> is now based on ggplot2 and lost all base graphics arguments.</p></li>
</ul></div>
<div class="section level4">
<h4 id="vignettes-0-5-0">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-5-0"></a></h4>
<ul><li>In <code>Introduction</code> vignette: complete restructuring of the Introduction</li>
<li>In <code>Introduction</code> vignette: add new ggplot2 based examples</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.0" id="mytai-040">myTAI 0.4.0<small>2016-05-04</small><a class="anchor" aria-label="anchor" href="#mytai-040"></a></h2>
<div class="section level4">
<h4 id="new-functions-0-4-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-4-0"></a></h4>
<ul><li>a new function <code>PlotSelectedAgeDistr()</code> allowing unsers to visualize the PS or DS gene distribution of a subset of genes stored in the input ExpressionSet object</li>
<li>a new function <code>PlotGroupDiffs()</code> allowing users to plot the significant differences between gene expression distributions of PS or DS groups</li>
<li>a new function <code>GroupDiffs()</code> allowing users to perform statistical tests to quantify the gene expression level differences between all genes of defined PS or DS groups</li>
</ul></div>
<div class="section level4">
<h4 id="updates-0-4-0">Updates<a class="anchor" aria-label="anchor" href="#updates-0-4-0"></a></h4>
<ul><li><p><code>PlotDistribution()</code> now uses ggplot2 to visualize the PS or DS distribution and is also based on the new function <code>PlotSelectedAgeDistr()</code>; furthermore it loses arguments <code>plotText</code> and <code>...</code> and gains a new argument <code>legendName</code></p></li>
<li><p>remove arguments ‘main.text’ and ‘…’ from <code>PlotCorrelation()</code></p></li>
<li><p><code>PlotCorrelation()</code> is now based on ggplot2</p></li>
<li><p><code>PlotGroupDiffs()</code> receives a new argument <code>gene.set</code> allowing users to statistically quantify the group specific PS/DS differences of a selected set of genes</p></li>
<li><p>analogously to <code>PlotGroupDiffs()</code> the function <code>GroupDiffs()</code> also receives a new argument <code>gene.set</code> allowing users to statistically quantify the group specific PS/DS differences of a selected set of genes</p></li>
<li><p>Fixing wrong x-axis labeling in <code>PlotCategoryExpr()</code> when <code>type = "stage-centered"</code> is specified</p></li>
<li><p><code>PlotCategoryExpr()</code> now also prints out the PS/DS absolute frequency distribution of the selected <code>gene.set</code></p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.3.0" id="mytai-030">myTAI 0.3.0<small>2015-07-31</small><a class="anchor" aria-label="anchor" href="#mytai-030"></a></h2>
<div class="section level4">
<h4 id="vignettes-0-3-0">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-3-0"></a></h4>
<ul><li>adding examples for <code>PlotCategoryExpr()</code> to <code>Advanced</code> Vignette</li>
<li>adding examples for <code>PlotReplicateQuality()</code> to <code>Expression</code> vignette</li>
</ul></div>
<div class="section level4">
<h4 id="new-functions-0-3-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-3-0"></a></h4>
<ul><li>a new function <code>PlotCategoryExpr()</code> allowing users to plot the expression levels of each age or divergence category as boxplot, dot plot or violin plot</li>
<li>a new function <code>PlotReplicateQuality()</code> allowing users to visualize the quality of biological replicates</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.2.1" id="mytai-021">myTAI 0.2.1<small>2015-07-23</small><a class="anchor" aria-label="anchor" href="#mytai-021"></a></h2>
<div class="section level4">
<h4 id="vignettes-0-2-1">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-2-1"></a></h4>
<ul><li>fixed a wrong example in the Enrichment vignette (<a href="https://github.com/HajkD/myTAI/commit/8d52fd60c274361dc9028dec3409abf60a738d8a" class="external-link uri">https://github.com/HajkD/myTAI/commit/8d52fd60c274361dc9028dec3409abf60a738d8a</a>)</li>
</ul></div>
<div class="section level4">
<h4 id="updates-0-2-1">Updates<a class="anchor" aria-label="anchor" href="#updates-0-2-1"></a></h4>
<ul><li>
<code>PlotGeneSet()</code> and <code>SelectGeneSet()</code> now have a new argument <code>use.only.map</code> specifying whether or not instead of using a standard <code>ExpressionSet</code> a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to the function.</li>
<li>a wrong version of the edgeR Bioconductor package was imported causing version 0.2.0 to fail R CMD Check on unix based systems</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.2.0" id="mytai-020">myTAI 0.2.0<small>2015-07-10</small><a class="anchor" aria-label="anchor" href="#mytai-020"></a></h2>
<div class="section level4">
<h4 id="vignettes-0-2-0">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-2-0"></a></h4>
<ul><li><p>adding new vignette <strong>Taxonomy</strong> providing spep by step instructions on retrieving taxonomic information for any organism of interest</p></li>
<li><p>adding new vignette <strong>Expression Analysis</strong> providing use cases to perform gene expression data analysis with <code>myTAI</code></p></li>
<li><p>adding new vignette <strong>Enrichment</strong> providing step-by-step instructions on how to perform PS and DS enrichment analyses with <code>PlotEnrichment()</code></p></li>
<li><p>adding examples for <code>pStrata()</code>, <code>pMatrix()</code>, <code>pTAI()</code>, <code>pTDI()</code>, and <code>PlotContribution()</code> to the <strong>Introduction</strong> Vignette</p></li>
</ul></div>
<div class="section level4">
<h4 id="new-functions-0-2-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-2-0"></a></h4>
<ul><li><p>a new function <code>taxonomy()</code> allows users to retrieve taxonomic information for any organism of interest; this function has been taken from the <a href="https://github.com/ropensci/biomartr" class="external-link">biomartr</a> package and was removed from <code>biomartr</code> afterwards. Please notice, that in myTAI version 0.1.0 the Introduction vignette referenced to the <code>taxonomy()</code> function in <code>biomartr</code>. This is no longer the case (since myTAI version 0.2.0), because now <code>taxonomy()</code> is implemented in myTAI.</p></li>
<li><p>the new <code>taxonomy()</code> function is based on the powerful R package <a href="https://github.com/ropensci/taxize" class="external-link">taxize</a>.</p></li>
<li><p>a new function <code>SelectGeneSet()</code> allows users to fastly select a subset of genes in an ExpressionSet</p></li>
<li><p>a new function <code>DiffGenes()</code> allows users to perform differential gene expression analysis with ExpressionSet objects</p></li>
<li><p>a new function <code>EnrichmentTest()</code> allows users to perform a Fisher’s exact test based enrichment analysis of over or underrepresented Phylostrata or Divergence Strata within a given gene set without having to plot the result</p></li>
<li><p>a new function <code>PlotGeneSet()</code> allows users to visualize the expression profiles of a given gene set</p></li>
<li><p>a new function <code>PlotEnrichment()</code> allows users to visualize the Phylostratum or Divergence Stratum enrichment of a given Gene Set as well as computing Fisher’s exact test to quantify the statistical significance of enrichment</p></li>
<li><p>a new function <code>PlotContribution()</code> allows users to visualize the Phylostratum or Divergence Stratum contribution to the global TAI/TDI pattern</p></li>
<li><p>a new function <code>pTAI()</code> allows users to compute the phylostratum contribution to the global TAI pattern</p></li>
<li><p>a new function <code>pTDI()</code> allows users to compute the divergence stratum contribution to the global TDI pattern</p></li>
</ul></div>
<div class="section level4">
<h4 id="updates-0-2-0">Updates<a class="anchor" aria-label="anchor" href="#updates-0-2-0"></a></h4>
<ul><li>
<code>FilterRNASeqCT()</code> has been renamed to <code>Expressed()</code> allowing users to apply this filter function to RNA-Seq data as well as to microarray data</li>
<li>
<code>PlotRE()</code> and <code>PlotMeans()</code> are now based on colors from the RColorBrewer package (default)</li>
<li>
<code>PlotRE()</code> and <code>PlotMeans()</code> now have a new argument <code>colors</code> allowing unsers to choose custom colors for the visualized relative or mean expression profiles</li>
<li>
<code>geom.mean()</code> and <code>harm.mean()</code> now are external functions accessible to the <code>myTAI</code> user</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.0" id="mytai-010">myTAI 0.1.0<small>2015-05-24</small><a class="anchor" aria-label="anchor" href="#mytai-010"></a></h2>
<div class="section level4">
<h4 id="main-news-0-1-0">Main News<a class="anchor" aria-label="anchor" href="#main-news-0-1-0"></a></h4>
<ul><li>now all functions have unit tests</li>
</ul></div>
<div class="section level4">
<h4 id="new-functions-0-1-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-1-0"></a></h4>
<ul><li><p>a new function <code>pStrata()</code> allows users to compute partial TAI/TDI values for all Phylostrata or Divergence Strata</p></li>
<li><p>a new function <code>CollapseReplicates()</code> allows users to combine replicate expression levels in ExpressionSet objects</p></li>
<li><p>a new function <code>FilterRNASeqCT()</code> allows users to filter expression levels of <code>ExpressionSet</code> objects deriving from RNA-Seq count tables</p></li>
</ul></div>
<div class="section level4">
<h4 id="updates-0-1-0">Updates<a class="anchor" aria-label="anchor" href="#updates-0-1-0"></a></h4>
<ul><li><p>function <code>MatchMap()</code> now receives a new argument <code>remove.duplicates</code> allowing users to delete duplicate gene ids (that might be stored in the input PhyoMap or DivergenceMap) during the process of matching a Map with an ExpressionSet</p></li>
<li><p><code>FlatLineTest()</code>, <code>ReductiveHourglassTest()</code>, <code>EarlyConservationTest()</code>, and <code>PlotPattern()</code> implement a new argument <code>custom.perm.matrix</code> allowing users to pass their own (custom) permutation matrix to the corresponding function. All subsequent test statistics and p-value/std.dev computations are then based on this custom permutation matrix</p></li>
<li><p><code>EarlyConservationTest()</code> and <code>ReductiveHourglassTest()</code> now have a new parameter <code>gof.warning</code> allowing users to choose whether or not non significant goodness of fit results should be printed as warning</p></li>
<li><p>now when specifying <code>TestStatistic = NULL</code> in <code>PlotPattern()</code> only the TAI/TDI profile is drawn (without performing any test statistics); this is equavalent to performing: <code>plot(TAI(PhyloExpressionSetExample)</code></p></li>
<li><p>function <code>combinatorialSignificance()</code> is now named <code>CombinatorialSignificance()</code></p></li>
<li><p>changing the title and description of the <code>myTAI</code> package</p></li>
<li><p>some minor changes in vignettes and within the documentation of functions</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.2" id="mytai-002">myTAI 0.0.2<small>2014-12-27</small><a class="anchor" aria-label="anchor" href="#mytai-002"></a></h2>
<div class="section level3">
<h3 id="new-features-in-v-0-0-2">New Features in v. 0.0.2<a class="anchor" aria-label="anchor" href="#new-features-in-v-0-0-2"></a></h3>
<ul><li><p><code>combinatorialSignificance()</code>, <code>FlatLineTest()</code>, <code>ReductiveHourglassTest()</code>, and <code>EarlyConservationTest()</code> now support multicore processing</p></li>
<li><p><code>MatchMap()</code> has been entirely rewritten and is now based on <a href="https://github.com/tidyverse/dplyr" class="external-link">dplyr</a>; additionally it now has a new argument <code>accumulate</code> that allows you to accumulate multiple expression levels to a unique expressiion level for a unique gene id</p></li>
</ul><div class="section level4">
<h4 id="vignettes-0-0-2">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-0-2"></a></h4>
<p>All three Vignettes: <code>Introduction</code>, <code>Intermediate</code>, and <code>Advanced</code> have been updated and extended.</p>
</div>
<div class="section level4">
<h4 id="bug-fixes-0-0-2">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-0-2"></a></h4>
<ul><li><p>two small bugs in <code>ReductiveHourglassTest()</code> and <code>EarlyConservationTest()</code> have been fixed that caused that instead of displaying 3 or 4 plots (<code>par(mfrow=c(1,3))</code> or <code>par(mfrow=c(2,2))</code>) only 1 plot has been generated</p></li>
<li><p>a small bug in <code>PlotMeans()</code> that caused the visualization of a wrong y-axis label when plotting only one group of Phylostrata or Divergence Strata</p></li>
</ul></div>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.1" id="mytai-001">myTAI 0.0.1<small>2014-11-01</small><a class="anchor" aria-label="anchor" href="#mytai-001"></a></h2>
<p>Introducing myTAI 0.0.1:</p>
<p>A framework to perform phylotranscriptomics analyses for Evolutionary Developmental Biology research.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Ștefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

