<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2024-10-03" />

<title>Gene Expression Analysis with myTAI</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Gene Expression Analysis with
<code>myTAI</code></h1>
<h4 class="date">2024-10-03</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In the <a href="Introduction.html">Introduction</a> vignette we
introduced and discussed how phylotranscriptomics can be applied to
capture evolutionary signals in (developmental) transcriptomes.
Furthermore, in the <a href="Enrichment.html">Enrichment Analyses</a>
vignette we provide a use case to correlate specific groups or sets of
genes with their predicted evolutionary origin. Here, we aim to combine
previously introduced techniques with <em>classic</em> gene expression
analyses to detect possible functional causes for the observed
transcriptome conservation.</p>
<p>In other words, phylotranscriptomics allows us to detect stages or
periods of evolutionary conservation and is able to predict the
evolutionary origin of process or trait specific genes based on
enrichment analyses. By combining evolutionary enrichment analyses with
the functional annotation of process or trait specific genes (see <a href="https://github.com/ropensci/biomartr/blob/master/vignettes/Functional_Annotation.Rmd">Functional
Annotation</a> for details) the detection of evolutionary signals can be
correlated with functional processes. Then, performing gene expression
analyses on corresponding process or trait specific genes allows users
to detect potential causes of stage/period specific evolutionary
transcriptome conservation.</p>
<p>The following sections introduce main gene expression data analysis
techniques implemented in <code>myTAI</code>:</p>
<ul>
<li><p>Detection of Differentially Expressed Genes (DEGs)</p></li>
<li><p>Fold-Change</p></li>
<li><p>Welch t-test</p></li>
<li><p>Wilcoxon Rank Sum Test (Mann-Whitney U test)</p></li>
<li><p>Negative Binomial (Exact Tests)</p></li>
<li><p>Collapsing Replicate Samples</p></li>
<li><p>Filter for Expressed Genes</p></li>
<li><p>Compute the Statistical Significance of Each Replicate
Combination</p></li>
</ul>
</div>
<div id="detection-of-differenentially-expressed-genes-degs" class="section level2">
<h2>Detection of Differenentially Expressed Genes (DEGs)</h2>
<p>A variety of methods have been published to detect differentially
expressed genes. Some methods are based on non-statistical
quantification of expression differences (e.g. fold-change and
log-fold-change), but most methods are based on statistical tests to
quantify the significance of differences in gene expression between
samples. These statistical methods can furthermore be divided into two
methodological categories: parametric tests and non-parametric tests.
The <code>DiffGenes()</code> function available in <code>myTAI</code>
implements the most popular and useful methods to detect differentially
expressed genes. In the literature, different methods have been
introduced and discussed for microarray technologies versus RNA-Seq
technologies.</p>
<p>In this section we will introduce all methods implemented in
<code>DiffGenes()</code> using small examples and will furthermore,
discuss published advantages and disadvantages of each method and each
mRNA quantification technology.</p>
<p><strong>Note that when using <code>DiffGenes()</code> it is assumed
that your input dataset has been normalized before passing it to
<code>DiffGenes()</code>. For RNA-Seq data <code>DiffGenes()</code>
assumes that the libraries have been normalized to have the same size,
i.e., to have the same expected column sum under the null hypothesis (or
the lib.size argument in <code>DiffGenes()</code> is specified
accordingly).</strong></p>
</div>
<div id="fold-changes" class="section level2">
<h2>Fold-Changes</h2>
<p>A fold change in gene expression is simply the ratio of the gene
expression level of one sample against a second sample: <span class="math inline">\(\frac{e_{i1}}{e_{i2}}\)</span>, where <span class="math inline">\(e_{i1}\)</span> is the expression level of gene
<span class="math inline">\(i\)</span> in sample one and <span class="math inline">\(e_{i2}\)</span> is the expression level of gene
<span class="math inline">\(i\)</span> in sample two. In case replicate
expression levels are present for each sample the ratio of means of the
corresponding replicates is computed: <span class="math inline">\(\frac{\bar{e}_{i1}}{\bar{e}_{i2}}\)</span>, where
<span class="math inline">\(\bar{e}_{i1}\)</span> is the mean of
replicate expression levels of gene <span class="math inline">\(i\)</span> in sample one and <span class="math inline">\(\bar{e}_{i2}\)</span> is the mean of replicate
expression levels of gene <span class="math inline">\(i\)</span> in
sample two.</p>
<ul>
<li><p><strong>Advantages:</strong> Given a small number of replicate
values the statistical evaluation of differentially expressed genes
might be biased (depending on the statistical test chosen) by underlying
sample distributions which are not fulfilled or because a small number
of replicate values is not sufficient enough to perform non-parametric
tests. Here, fold-changes provide a simple way to quantify gene
expression differences between samples by <span class="math inline">\(n\)</span>-fold enrichment. In our opinion,
although the process of choosing a threshold for defining genes as being
differentially expressed or not based on fold-change values is purely
subjective and relies on common sense, in some cases this procedure will
be more suitable than defining differentially expressed genes based on
p-values obtained from a test statistic with violated test
assumptions.</p></li>
<li><p><strong>Disadvantages:</strong> If used appropriately,
statistical tests not only systematically quantify the significance of
the observed gene-by-gene differences of expression, but furthermore,
accounts the variance of replicate expression levels when comparing the
mean difference of replicate expression levels between samples. Hence,
the gene specific variance between replicates is also quantified by the
p-value returned by the sufficient test statistic which is not
quantified by a simple fold-change measure.</p></li>
</ul>
<div id="example-fold-change" class="section level3">
<h3>Example: Fold-Change</h3>
<p>For the following example we assume that
<code>PhyloExpressionSetExample[1:5,1:8]</code> stores 5 genes and 3
developmental stages with 2 replicate expression levels per stage.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the fold-change measure</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>                  <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>                  <span class="at">method        =</span> <span class="st">&quot;foldchange&quot;</span>,</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>                  <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="fu">head</span>(DEGs)</span></code></pre></div>
<pre><code>   Phylostratum      GeneID    S1-&gt;S2    S1-&gt;S3    S2-&gt;S1    S2-&gt;S3    S3-&gt;S1    S3-&gt;S2
1            1 at1g01040.2 1.6713881 2.0806706 0.5983051 1.2448758 0.4806143 0.8032930
2            1 at1g01050.1 1.0273222 1.2709185 0.9734045 1.2371177 0.7868325 0.8083305
3            1 at1g01070.1 1.3087379 1.4044799 0.7640949 1.0731560 0.7120073 0.9318310
4            1 at1g01080.2 0.7779572 0.7286769 1.2854177 0.9366542 1.3723503 1.0676299
5            1 at1g01090.1 0.3803866 0.2288961 2.6289042 0.6017460 4.3687939 1.6618307</code></pre>
<p>The resulting output shows all combinations of fold-changes between
samples (developmental stages). Here, <code>S1-&gt;S2</code> denotes
that the fold-change was computed for expression levels of stage
<code>S1</code> against stage <code>S2</code>.</p>
</div>
<div id="example-log-fold-change" class="section level3">
<h3>Example: Log-Fold-Change</h3>
<p><strong>When selecting <code>method = &quot;log-foldchange&quot;</code> it is
assumed that the input <code>ExpressionSet</code> stores
<code>log2</code> expression levels. Here, we transform absolute
expression levels stored in <code>PhyloExpressionSetExample</code> to
<code>log2</code> expression levels using the <code>tf()</code> function
before log-fold-changes are computed.</strong></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the logfold-change measure</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>log.DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> <span class="fu">tf</span>(PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],log2),</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>                      <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>                      <span class="at">method        =</span> <span class="st">&quot;log-foldchange&quot;</span>,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>                      <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="fu">head</span>(log.DEGs)</span></code></pre></div>
<pre><code>  Phylostratum      GeneID      S1-&gt;S2     S1-&gt;S3      S2-&gt;S1      S2-&gt;S3     S3-&gt;S1      S3-&gt;S2
1            1 at1g01040.2  0.74104679  1.0570486 -0.74104679  0.31600182 -1.0570486 -0.31600182
2            1 at1g01050.1  0.03888868  0.3458715 -0.03888868  0.30698280 -0.3458715 -0.30698280
3            1 at1g01070.1  0.38817621  0.4900360 -0.38817621  0.10185975 -0.4900360 -0.10185975
4            1 at1g01080.2 -0.36223724 -0.4566488  0.36223724 -0.09441158  0.4566488  0.09441158
5            1 at1g01090.1 -1.39446159 -2.1272350  1.39446159 -0.73277345  2.1272350  0.73277345</code></pre>
<p>The resulting output stores all combinations of log fold-changes
between samples (developmental stages).</p>
</div>
</div>
<div id="welch-t-test" class="section level2">
<h2>Welch t-test</h2>
<p>The <code>Welch t-test</code> is a parametric test to statistically
quantify the difference of sample means in cases where the assumption of
homogeneity of variance (equal variances in the two populations) is
violated (Boslaugh, 2013). The <code>Welch t-test</code> is a sufficient
parameter test for small sample sizes and thus, has been used to detect
differentially expressed genes based on p-values returned by the test
statistic (Hahne et al., 2008).</p>
<p>In detail, the test statistic is computed as follows:</p>
<p><span class="math inline">\(t = \frac{\bar{x}_1 -
\bar{x}_2}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}}\)</span></p>
<p>where <span class="math inline">\(\bar{x}_1\)</span> and <span class="math inline">\(\bar{x}_2\)</span> are sample means, <span class="math inline">\(s_1^2\)</span> and <span class="math inline">\(s_2^2\)</span> are the sample variances, and <span class="math inline">\(n_1\)</span> and <span class="math inline">\(n_2\)</span> are the sample sizes.</p>
<p>The degrees of freedom for Welch’s t-test are then computed as
follows:</p>
<p><span class="math inline">\(df = \frac{\big(\frac{s_1^2}{n_1} +
\frac{s_2^2}{n_2}\big)^2}{\frac{s_1^4}{n_1^2 (n_1 - 1)} +
\frac{s_2^4}{n_2^2 (n_2 - 1)}}\)</span></p>
<p>To perform a sufficient <code>Welch t-test</code> the following
assumptions about the input data need to be fulfilled to test whether
two samples come from populations with equal means:</p>
<p><strong>Assumptions about input data</strong></p>
<ul>
<li>independent samples</li>
<li>continuous data</li>
<li>(approximate) normality</li>
</ul>
<p>Nevertheless, although in most cases <code>log2</code> expression
levels are used to perform the <code>Welch t-test</code> assuming that
expression levels are log-normal distributed which approximates a normal
distribution in infinity, in most cases the small number of replicates
is not sufficient enough to fulfill the (approximate) normality
assumption of the <code>Welch t-test</code>.</p>
<p>Due to this fact, non-parametric, sampling based, or generalized
linear model based methods have been proposed to quantify p-values of
differential expression. Nevertheless, the <code>DiffGenes()</code>
function implements the <code>Welch t-test</code> for the detection of
differentially expressed genes, allowing users to compare the results
with more recent DEG detection methods/methodologies also implemented in
<code>DiffGenes()</code>.</p>
<ul>
<li><strong>Advantages:</strong>
<ul>
<li>DEG detection based on statistical quantification</li>
<li>Parametric test resulting in a strong test statistic</li>
<li>Can handle small sample sizes</li>
</ul></li>
<li><strong>Disadvantages:</strong>
<ul>
<li>Test assumptions must be fulfilled to return sufficient
p-values<br />
</li>
<li>Can hardly assure normality with very sample sizes of <span class="math inline">\(n = 3,4,5,..\)</span> (replicates)</li>
<li>Pairwise comparisons between different stages or experiments</li>
</ul></li>
</ul>
<div id="example-welch-t-test" class="section level3">
<h3>Example: Welch t-test</h3>
<p>Performing <code>Welch t-test</code> with <code>DiffGenes()</code>
can be done by specifying <code>method = &quot;t.test&quot;</code>. Internally
<code>DiffGenes()</code> performs a two sided <code>Welch t-test</code>.
This means that the <code>Welch t-test</code> quantifies only whether or
not a gene is significantly differentially expressed, but not the
direction of enrichment (over-expressed or under-expressed).</p>
<p>The <code>PhyloExpressionSetExample</code> we use in the following
example stores absolute expression levels. In case your
<code>ExpressionSet</code> also stores absolute expression levels (which
is likely due to the <code>ExpressionSet</code> standard for
Phylotranscriptomics analyses), you can use the <code>tf()</code>
function implemented in <code>myTAI</code> to transform absolute
expression levels to <code>log2</code> expression levels before
performing <code>DiffGenes()</code> with a <code>Welch t-test</code>,
e.g. <code>tf(PhyloExpressionSetExample[1:5,1:8],log2)</code>. In
general, using <code>log2</code> transformed expression levels as input
<code>ExpressionSet</code> of <code>DiffGenes()</code> allows us to (at
least) assume that samples (replicate expression levels) used to perform
the <code>Welch t-test</code> are log-normal distributed and therefore,
somewhat approximate normal distributed.</p>
<p>Please notice however, that RNA-Seq data can include count values of
0. So when transforming absolute counts to <code>log2</code> counts
infinity values of <code>log2(0) = -Inf</code> will be produced and
therefore, p-value computations will not be possible. To avoid this case
you could either remove RNA-Seq count values of 0 from the input dataset
using the <code>Expressed()</code> function (see section <em>Filter for
Expressed Genes</em>), e.g. pass
<code>tf(Expressed(PhyloExpressionSetExample[1:5,1:8], cut.off = 1),log2)</code>
as <code>ExpressionSet</code> argument to <code>DiffGenes()</code> or
shift all count values by a constant value, e.g. pass
<code>tf(PhyloExpressionSetExample[1:5,1:8], function(x) log2(x + 1))</code>
as <code>ExpressionSet</code> argument to <code>DiffGenes()</code>.</p>
<p>Internally, <code>DiffGenes()</code> will also check for 0 values in
input data and will automatically shift all expression levels by
<code>+1</code> in case 0 values are included.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by a Welch t-test</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>ttest.DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> <span class="fu">tf</span>(PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],log2),</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>                        <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>                        <span class="at">method        =</span> <span class="st">&quot;t.test&quot;</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>                        <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>ttest.DEGs</span></code></pre></div>
<pre><code>  Phylostratum      GeneID     S1&lt;-&gt;S2    S1&lt;-&gt;S3    S2&lt;-&gt;S3
1            1 at1g01040.2 0.027832572 0.04020203 0.13481563
2            1 at1g01050.1 0.852379466 0.31471871 0.36326955
3            1 at1g01070.1 0.003200692 0.00113536 0.02236621
4            1 at1g01080.2 0.086426813 0.03092924 0.45999438
5            1 at1g01090.1 0.090387087 0.04638872 0.04978092</code></pre>
<p>The resulting <code>data.frame</code> stores the p-values of
stage-wise comparisons for each gene. To adjust p-values for multiple
testing of stage-wise comparisons you can specify the
<code>p.adjust.method</code> argument with one of the p-value adjustment
methods implemented in <code>DiffGenes()</code>.</p>
<p>In detail, correcting for multiple testing allows to appropriately
choose selection cut-offs for p-values fulfilling the differential
expression criteria. Hahne et al., 2008 (p. 87) give a nice example of
correcting for multiple testing to determine appropriate selection
cut-offs.</p>
<p>Please consult the documentation of <code>?p.adjust</code> to see
which p-value adjustment methods are implemented in
<code>DiffGenes()</code>.</p>
<p>Please also consult these reviews <a href="http://www.stat.columbia.edu/~gelman/research/published/multiple2f.pdf">Gelman
et al., 2008</a>, and <a href="http://www.gs.washington.edu/academics/courses/akey/56008/lecture/lecture10.pdf">Slides</a>)
to decide whether or not to apply p-value adjustment to your own
dataset.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by a Welch t-test</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co"># and furthermore, adjust p-values for multiple comparison</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co"># using the Benjamini &amp; Hochberg (1995) method: method = &quot;BH&quot;</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>ttest.DEGs.p_adjust <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet   =</span> <span class="fu">tf</span>(PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],log2),</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>                                 <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>                                 <span class="at">method          =</span> <span class="st">&quot;t.test&quot;</span>,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                                 <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>                                 <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>ttest.DEGs.p_adjust</span></code></pre></div>
<pre><code>  Phylostratum      GeneID    S1&lt;-&gt;S2   S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.06958143 0.0579859 0.2246927
2            1 at1g01050.1 0.85237947 0.3147187 0.4540869
3            1 at1g01070.1 0.01600346 0.0056768 0.1118311
4            1 at1g01080.2 0.11298386 0.0579859 0.4599944
5            1 at1g01090.1 0.11298386 0.0579859 0.1244523</code></pre>
<p>The resulting p-value adjusted <code>data.frame</code> can be used to
filter for differentially expressed genes. Here, specifying the
arguments: <code>comparison</code>, <code>alpha</code>, and
<code>filter.method</code> in <code>DiffGenes()</code> allows users to
obtain only significant differentially expressed genes.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by a Welch t-test</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co"># and furthermore, adjust p-values for multiple comparison</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co"># using the Benjamini &amp; Hochberg (1995) method: method = &quot;BH&quot;</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co"># and filter for significantly differentially expressed genes (alpha = 0.05) </span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>ttest.DEGs.p_adjust.filtered <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet   =</span> <span class="fu">tf</span>(PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span> ,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],log2),</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                                          <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>                                          <span class="at">method          =</span> <span class="st">&quot;t.test&quot;</span>,</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>                                          <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>                                          <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>),</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>                                          <span class="at">comparison      =</span> <span class="st">&quot;above&quot;</span>,</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>                                          <span class="at">alpha           =</span> <span class="fl">0.05</span>,</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>                                          <span class="at">filter.method   =</span> <span class="st">&quot;n-set&quot;</span>,</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>                                          <span class="at">n               =</span> <span class="dv">1</span>)</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="co"># look at the genes fulfilling the filter criteria </span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>ttest.DEGs.p_adjust.filtered</span></code></pre></div>
<pre><code>  Phylostratum      GeneID    S1&lt;-&gt;S2   S1&lt;-&gt;S3   S2&lt;-&gt;S3
3            1 at1g01070.1 0.03200692 0.0113536 0.2192432</code></pre>
<p>In this example, only 1 out of 10 genes fulfills the p-value criteria
(<code>alpha = 0.05</code>) in at least one stage comparison.</p>
<div id="rank-top-p-values" class="section level4">
<h4>Rank top p-values</h4>
<p>Finally, users can rank genes in increasing p-value order for each
stage comparison by typing:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>ttest.DEGs.p_adjust <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet   =</span> <span class="fu">tf</span>(PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],log2),</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>                                 <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>                                 <span class="at">method          =</span> <span class="st">&quot;t.test&quot;</span>,</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>                                 <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>                                 <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="fu">head</span>(ttest.DEGs.p_adjust[<span class="fu">order</span>(ttest.DEGs.p_adjust[ , <span class="st">&quot;S1&lt;-&gt;S2&quot;</span>], <span class="at">decreasing =</span> <span class="cn">FALSE</span>) , <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span></code></pre></div>
<pre><code>    Phylostratum      GeneID  S1&lt;-&gt;S2
54             1 at1g02400.1 0.151388
119            1 at1g03870.1 0.151388
137            1 at1g04380.1 0.151388
289            1 at1g08110.4 0.151388
383            1 at1g10360.1 0.151388
413            1 at1g11040.1 0.151388</code></pre>
<p>Here the line
<code>ttest.DEGs.p_adjust[order(ttest.DEGs.p_adjust[ , &quot;S1&lt;-&gt;S2&quot;], decreasing = FALSE) , 1:3]</code>
will sort p-values of stage comparison <code>&quot;S1&lt;-&gt;S2&quot;</code> in
increasing order.</p>
</div>
</div>
</div>
<div id="wilcoxon-mann-whitney-test-mann-whitney-u-test" class="section level2">
<h2>Wilcoxon-Mann-Whitney test (Mann-Whitney U test)</h2>
<p>The Wilcoxon-Mann-Whitney test is a <em>nonparametric</em> test to
quantify the shift in empirical distribution parameters.
<em>Nonparametric</em> tests are useful when sample populations do not
meet the test assumptions of <em>parametric</em> tests.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by a Wilcoxon-Mann-Whitney test</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>Wilcox.DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>                        <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>                        <span class="at">method        =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>                        <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>Wilcox.DEGs</span></code></pre></div>
<pre><code>  Phylostratum      GeneID   S1&lt;-&gt;S2   S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.3333333 0.3333333 0.3333333
2            1 at1g01050.1 1.0000000 0.3333333 0.3333333
3            1 at1g01070.1 0.3333333 0.3333333 0.3333333
4            1 at1g01080.2 0.3333333 0.3333333 0.6666667
5            1 at1g01090.1 0.3333333 0.3333333 0.3333333</code></pre>
<p>Again, users can adjust p-values by specifying the
<code>p.adjust.method</code> argument.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by a Wilcoxon-Mann-Whitney test</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co"># and furthermore, adjust p-values for multiple comparison</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="co"># using the Benjamini &amp; Hochberg (1995) method: method = &quot;BH&quot;</span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a><span class="co"># and filter for significantly differentially expressed genes (alpha = 0.05)</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>Wilcox.DEGs.adj <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet  =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>                            <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>                            <span class="at">method          =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a>                            <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>),</span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>                            <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a>Wilcox.DEGs.adj</span></code></pre></div>
<pre><code>  Phylostratum      GeneID   S1&lt;-&gt;S2   S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.4166667 0.3333333 0.4166667
2            1 at1g01050.1 1.0000000 0.3333333 0.4166667
3            1 at1g01070.1 0.4166667 0.3333333 0.4166667
4            1 at1g01080.2 0.4166667 0.3333333 0.6666667
5            1 at1g01090.1 0.4166667 0.3333333 0.4166667</code></pre>
</div>
<div id="negative-binomial-exact-tests" class="section level2">
<h2>Negative Binomial (Exact Tests)</h2>
<p>Exact Tests for Differences between two groups of negative-binomial
counts implemented in <code>DiffGenes()</code> are based on the
<code>edgeR</code> function <code>exactTest()</code>. Please consult the
<a href="https://www.bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR
Users Guide</a> for mathematical details.</p>
<div id="install-edger-package" class="section level3">
<h3>Install edgeR Package</h3>
<p>The detection of DEGs using negative binomial models is based on the
powerful implementations provided by the <a href="https://www.bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a>
package. Hence, before using the negative binomial models in
<code>DiffGenes()</code> users need to install the edgeR package.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># install edgeR</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;http://bioconductor.org/biocLite.R&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="fu">biocLite</span>(<span class="st">&quot;edgeR&quot;</span>)</span></code></pre></div>
</div>
<div id="double-tail-method" class="section level3">
<h3>Double Tail Method</h3>
<p>This method computes two-sided p-values by doubling the smaller tail
probability (see <code>?exactTestByDeviance</code> for details). To
compute p-values for stagewise comparisons based on negative binomial
models, the <code>DiffGenes()</code> argument
<code>method = &quot;doubletail&quot;</code>, the number of replicates per stage
<code>nrep</code>, and <code>lib.size</code> quantifying the library
size to equalize sample library sizes by quantile-to-quantile
normalization need to be specified (see also
<code>?equalizeLibSizes</code>).</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by the Double Tail Method</span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>DoubleTail.DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>                        <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>                        <span class="at">method        =</span> <span class="st">&quot;doubletail&quot;</span>,</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>                        <span class="at">lib.size      =</span> <span class="dv">1000</span>,</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>                        <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>DoubleTail.DEGs</span></code></pre></div>
<pre><code>  Phylostratum      GeneID    S1&lt;-&gt;S2     S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.26026604 0.110233012 0.6304508
2            1 at1g01050.1 0.95314428 0.598102712 0.6398757
3            1 at1g01070.1 0.55461941 0.456018563 0.8774231
4            1 at1g01080.2 0.58130025 0.487028051 0.8860005
5            1 at1g01090.1 0.03615134 0.001773543 0.2645537</code></pre>
<p>Again, users can adjust p-values by specifying the
<code>p.adjust.method</code> argument.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by the Double Tail Method</span></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="co"># and furthermore, adjust p-values for multiple comparison</span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="co"># using the Benjamini &amp; Hochberg (1995) method: method = &quot;BH&quot;</span></span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a><span class="co"># and filter for significantly differentially expressed genes (alpha = 0.05)</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>DoubleTail.DEGs.adj <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet  =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>                                <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>                                <span class="at">method          =</span> <span class="st">&quot;doubletail&quot;</span>,</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>                                <span class="at">lib.size        =</span> <span class="dv">1000</span>,</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>                                <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>),</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a>                                <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a>DoubleTail.DEGs.adj</span></code></pre></div>
<pre><code>  Phylostratum      GeneID   S1&lt;-&gt;S2     S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.6506651 0.275582530 0.8860005
2            1 at1g01050.1 0.9531443 0.598102712 0.8860005
3            1 at1g01070.1 0.7266253 0.598102712 0.8860005
4            1 at1g01080.2 0.7266253 0.598102712 0.8860005
5            1 at1g01090.1 0.1807567 0.008867715 0.8860005</code></pre>
</div>
<div id="small-p-method" class="section level3">
<h3>Small-P Method</h3>
<p>This method performs the method of small probabilities as proposed by
Robinson and Smyth (2008) (see <code>exactTestBySmallP</code> for
details). To compute p-values for stagewise comparisons based on
negative binomial models, the <code>DiffGenes()</code> argument
<code>method = &quot;doubletail&quot;</code>, the number of replicates per stage
<code>nrep</code>, and <code>lib.size</code> quantifying the library
size to equalize sample library sizes by quantile-to-quantile
normalization need to be specified (see also
<code>?equalizeLibSizes</code>).</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by the Small-P Method</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>SmallP.DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>                        <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>                        <span class="at">method        =</span> <span class="st">&quot;smallp&quot;</span>,</span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>                        <span class="at">lib.size      =</span> <span class="dv">1000</span>,</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>                        <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a>SmallP.DEGs</span></code></pre></div>
<pre><code>  Phylostratum      GeneID    S1&lt;-&gt;S2     S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.26026604 0.110233012 0.6304508
2            1 at1g01050.1 0.95314428 0.598102712 0.6398757
3            1 at1g01070.1 0.55461941 0.456018563 0.8774231
4            1 at1g01080.2 0.58130025 0.487028051 0.8860005
5            1 at1g01090.1 0.03615134 0.001773543 0.2645537</code></pre>
<p>Again, users can adjust p-values by specifying the
<code>p.adjust.method</code> argument.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by the Small-P Method</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="co"># and furthermore, adjust p-values for multiple comparison</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="co"># using the Benjamini &amp; Hochberg (1995) method: method = &quot;BH&quot;</span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a><span class="co"># and filter for significantly differentially expressed genes (alpha = 0.05)</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>SmallP.DEGs.adj <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet  =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>                                <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>                                <span class="at">method          =</span> <span class="st">&quot;smallp&quot;</span>,</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>                                <span class="at">lib.size        =</span> <span class="dv">1000</span>,</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>                                <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>),</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>                                <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>SmallP.DEGs.adj</span></code></pre></div>
<pre><code>  Phylostratum      GeneID   S1&lt;-&gt;S2     S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.6506651 0.275582530 0.8860005
2            1 at1g01050.1 0.9531443 0.598102712 0.8860005
3            1 at1g01070.1 0.7266253 0.598102712 0.8860005
4            1 at1g01080.2 0.7266253 0.598102712 0.8860005
5            1 at1g01090.1 0.1807567 0.008867715 0.8860005</code></pre>
</div>
<div id="deviance-method" class="section level3">
<h3>Deviance Method</h3>
<p>This method uses the deviance goodness of fit statistics to define
the rejection region, and is therefore equivalent to a conditional
likelihood ratio test (see <code>exactTestByDeviance</code> for
details). To compute p-values for stagewise comparisons based on
negative binomial models, the <code>DiffGenes()</code> argument
<code>method = &quot;doubletail&quot;</code>, the number of replicates per stage
<code>nrep</code>, and <code>lib.size</code> quantifying the library
size to equalize sample library sizes by quantile-to-quantile
normalization need to be specified (see also
<code>?equalizeLibSizes</code>).</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by the Deviance</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>Deviance.DEGs <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>                        <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>                        <span class="at">method        =</span> <span class="st">&quot;deviance&quot;</span>,</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>                        <span class="at">lib.size      =</span> <span class="dv">1000</span>,</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>                        <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>Deviance.DEGs</span></code></pre></div>
<pre><code>  Phylostratum      GeneID    S1&lt;-&gt;S2     S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.26026604 0.110233012 0.6304508
2            1 at1g01050.1 0.95314428 0.598102712 0.6398757
3            1 at1g01070.1 0.55461941 0.456018563 0.8774231
4            1 at1g01080.2 0.58130025 0.487028051 0.8860005
5            1 at1g01090.1 0.03615134 0.001773543 0.2645537</code></pre>
<p>Again, users can adjust p-values by specifying the
<code>p.adjust.method</code> argument.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PhyloExpressionSetExample&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="co"># Detection of DEGs using the p-value returned by the Deviance Method</span></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="co"># and furthermore, adjust p-values for multiple comparison</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a><span class="co"># using the Benjamini &amp; Hochberg (1995) method: method = &quot;BH&quot;</span></span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a><span class="co"># and filter for significantly differentially expressed genes (alpha = 0.05)</span></span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a>Deviance.DEGs.adj <span class="ot">&lt;-</span> <span class="fu">DiffGenes</span>(<span class="at">ExpressionSet    =</span> PhyloExpressionSetExample[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a>                                <span class="at">nrep            =</span> <span class="dv">2</span>,</span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a>                                <span class="at">method          =</span> <span class="st">&quot;deviance&quot;</span>,</span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a>                                <span class="at">lib.size        =</span> <span class="dv">1000</span>,</span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a>                                <span class="at">stage.names     =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>),</span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a>                                <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a></span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a><span class="co"># look at the results</span></span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a>Deviance.DEGs.adj</span></code></pre></div>
<pre><code>  Phylostratum      GeneID   S1&lt;-&gt;S2     S1&lt;-&gt;S3   S2&lt;-&gt;S3
1            1 at1g01040.2 0.6506651 0.275582530 0.8860005
2            1 at1g01050.1 0.9531443 0.598102712 0.8860005
3            1 at1g01070.1 0.7266253 0.598102712 0.8860005
4            1 at1g01080.2 0.7266253 0.598102712 0.8860005
5            1 at1g01090.1 0.1807567 0.008867715 0.8860005</code></pre>
</div>
</div>
<div id="replicate-quality-check" class="section level2">
<h2>Replicate Quality Check</h2>
<p>Users can also perform replicate quality checks to quantify the
variability between replicate expression levels fo each stage
separately.</p>
<p>The <code>PlotReplicateQuality()</code> is designed to perform
customized replicate variability checks for any
<code>ExpressionSet</code> object storing replicates.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a><span class="co"># visualize the sd() between replicates</span></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a><span class="fu">PlotReplicateQuality</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[ , <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>                     <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>                     <span class="at">legend.pos   =</span> <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>                     <span class="at">ylim          =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.2</span>),</span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>                     <span class="at">lwd           =</span> <span class="dv">6</span>)</span></code></pre></div>
<p>The resulting plot visualizes the kernel density estimates for the
variance (log variance) between replicates. Each curve represents the
density function for the replicate variation within one stage or
experiment. In this case the variance between replicates of
<code>Stage 1</code> to <code>Stage 3</code> (each including 2
replicates) seem to deviate from each other allowing the conclusion that
each stage has a different expression level variability between
replicates.</p>
<p>The <code>FUN</code> argument implemented in
<code>PlotReplicateQuality()</code> allows users to furthermore, specify
customized criteria quantifying replicate variability. Please notice
that the function specified in <code>FUN</code> will be performed
separately on each gene and stage.</p>
<p>In the following example the median absolute deviation function
<code>mad()</code> is used to quantify replicate variability.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a><span class="co"># visualize the mad() between replicates</span></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="fu">PlotReplicateQuality</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample[ , <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>],</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>                     <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>                     <span class="at">FUN           =</span> mad,</span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>                     <span class="at">legend.pos    =</span> <span class="st">&quot;topright&quot;</span>,</span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>                     <span class="at">ylim          =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.015</span>),</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>                     <span class="at">lwd           =</span> <span class="dv">6</span>)</span></code></pre></div>
<p>In general, users are not limited to specific functions implemented
in R. By writing customized functions such as
<code>FUN = function(x) return((x - mean(x))^2)</code> users can define
their own criteria to quantify replicate variability and can then apply
this criteria to <code>PlotReplicateQuality()</code> by specifying the
<code>FUN</code> argument.</p>
</div>
<div id="collapsing-replicate-samples" class="section level2">
<h2>Collapsing Replicate Samples</h2>
<p>After performing differential gene expression analyses, replicate
expression levels are collapsed to a single stage specific expression
level. For this purpose, <code>myTAI</code> implements the
<code>CollapseReplicates()</code> function, allowing users to combine
replicate expression levels stored in a standard
<code>PhyloExpressionSet</code> or <code>DivergenceExpressionSet</code>
object to a stage specific expression level using a specified window
function.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">library</span>(myTAI)</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a><span class="co"># load example data</span></span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a><span class="co"># generate an example PhyloExpressionSet with replicates</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>ExampleReplicateExpressionSet <span class="ot">&lt;-</span> PhyloExpressionSetExample[ ,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>]</span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a><span class="co"># rename stages</span></span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a><span class="fu">names</span>(ExampleReplicateExpressionSet)[<span class="dv">3</span><span class="sc">:</span><span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Stage_1_Repl_1&quot;</span>,<span class="st">&quot;Stage_1_Repl_2&quot;</span>,</span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a>                                               <span class="st">&quot;Stage_2_Repl_1&quot;</span>,<span class="st">&quot;Stage_2_Repl_2&quot;</span>,</span>
<span id="cb32-12"><a href="#cb32-12" tabindex="-1"></a>                                               <span class="st">&quot;Stage_3_Repl_1&quot;</span>,<span class="st">&quot;Stage_3_Repl_2&quot;</span>)</span>
<span id="cb32-13"><a href="#cb32-13" tabindex="-1"></a><span class="co"># have a look at the example dataset</span></span>
<span id="cb32-14"><a href="#cb32-14" tabindex="-1"></a><span class="fu">head</span>(ExampleReplicateExpressionSet, <span class="dv">5</span>)</span></code></pre></div>
<pre><code>  Phylostratum      GeneID Stage_1_Repl_1 Stage_1_Repl_2 Stage_2_Repl_1 Stage_2_Repl_2 Stage_3_Repl_1 Stage_3_Repl_2
1            1 at1g01040.2       2173.635      1911.2001       1152.555      1291.4224       1000.253       962.9772
2            1 at1g01050.1       1501.014      1817.3086       1665.309      1564.7612       1496.321      1114.6435
3            1 at1g01070.1       1212.793      1233.0023        939.200       929.6195        864.218       877.2060
4            1 at1g01080.2       1016.920       936.3837       1181.338      1329.4734       1392.643      1287.9746
5            1 at1g01090.1      11424.567     16778.1685      34366.649     39775.6405      56231.569     66980.3673</code></pre>
<p>Now, assume that this example <code>PhyloExpressionSet</code> stores
three developmental stages and 2 biological replicates for each
developmental stage. Of course, we could now compute and visualize the
TAI profile by typing:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="co"># visualize the TAI profile over 3 stages of development</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a><span class="co"># and 2 replicates per stage</span></span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a><span class="fu">PlotPattern</span>(<span class="at">ExpressionSet =</span> ExampleReplicateExpressionSet,</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>            <span class="at">type          =</span> <span class="st">&quot;l&quot;</span>,</span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>            <span class="at">lwd           =</span> <span class="dv">6</span>)</span></code></pre></div>
<p>Usually, one would expect that variations in replicate values are
smaller than variations between developmental stages. In this example
however, we constructed replicate values that vary larger than
expression levels between developmental stages. For many applications it
might be useful to visualize TAI/TDI values of replicates as well, but
normally replicate values are collapsed to one gene and stage specific
value after differential gene expression analyses and replicate quality
control have been performed.</p>
<p>The following example illustrates how to collapse replicates with
<code>CollapseReplicates()</code>:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="co"># combine the expression levels of the 2 replicates (const) per stage</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="co"># using geom.mean as window function and rename new stages to: &quot;S1&quot;,&quot;S2&quot;,&quot;S3&quot;</span></span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>CollapssedPhyloExpressionSet <span class="ot">&lt;-</span> <span class="fu">CollapseReplicates</span>(</span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>                                       <span class="at">ExpressionSet =</span> ExampleReplicateExpressionSet,</span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a>                                       <span class="at">nrep          =</span> <span class="dv">2</span>,</span>
<span id="cb35-6"><a href="#cb35-6" tabindex="-1"></a>                                       <span class="at">FUN           =</span> geom.mean,</span>
<span id="cb35-7"><a href="#cb35-7" tabindex="-1"></a>                                       <span class="at">stage.names   =</span> <span class="fu">c</span>(<span class="st">&quot;S1&quot;</span>,<span class="st">&quot;S2&quot;</span>,<span class="st">&quot;S3&quot;</span>))</span>
<span id="cb35-8"><a href="#cb35-8" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" tabindex="-1"></a><span class="co"># have a look at the collapsed PhyloExpressionSet</span></span>
<span id="cb35-10"><a href="#cb35-10" tabindex="-1"></a><span class="fu">head</span>(CollapssedPhyloExpressionSet)</span></code></pre></div>
<pre><code>   Phylostratum      GeneID         S1         S2         S3
1            1 at1g01040.2  2038.1982  1220.0147   981.4381
2            1 at1g01050.1  1651.6070  1614.2524  1291.4582
3            1 at1g01070.1  1222.8557   934.3975   870.6878
4            1 at1g01080.2   975.8215  1253.2189  1339.2866
5            1 at1g01090.1 13844.9740 36972.3612 61371.0937
6            1 at1g01120.1   815.3288   894.8987   905.8272</code></pre>
<p>The <code>nrep</code> argument specifies either a constant number of
replicates per stage or a numeric vector storing variable numbers of
replicates for each developmental stage. In our example, each
developmental stage had a constant (equal) number of replicates per
developmental stage (<code>nrep = 2</code>). In case a variable stage
specific number of replicates is present, one could specify
<code>nrep = c(2,3,2)</code> defining the case that developmental stage
1 stores 2 replicates, stage 2 stores 3 replicates, and stage 3 again,
stores 2 replicates.</p>
<p>The argument <code>FUN</code> specifies the window function to
collapse replicate expression levels to a single stage specific value.
In this example, we chose the <code>geom.mean()</code> (geometric mean)
function implemented in <code>myTAI</code>, because our example
<code>PhyloExpressionSet</code> stores absolute expression levels.
Notice that the mathematical equivalent of performing arithmetic mean
(<code>mean()</code>) computations on <code>log</code> expression levels
is to perform the geometric mean (<code>geom.mean()</code>) on absolute
expression levels.</p>
<p>The <code>stage.names</code> argument then specifies the new names of
collapsed stages.</p>
</div>
<div id="filter-for-expressed-genes" class="section level2">
<h2>Filter for Expressed Genes</h2>
<p>After differential gene expression analyses and replicate aggregation
have been performed, some studies filter gene expression levels in
RNA-Seq count tables or microarray expression matrices for non-expressed
or outlier genes. For example, in most studies performing RNA-Seq
experiments FPKM/RPKM values &lt; 1 are remove from the processed
(final) count table.</p>
<p>For this purpose <code>myTAI</code> implements the
<code>Expressed()</code> function to filter (remove) expression levels
in RNA-Seq count tables or microarray expression matrices which do not
pass a defined expression threshold.</p>
<p>The <code>Expressed()</code> function takes a standard
<code>PhyloExpressionSet</code> or <code>DivergenceExpressionSet</code>
object storing a RNA-Seq count table (CT) or microarray gene expression
matrix and removes genes from this count table or gene expression matrix
that have an expression level below a defined <code>cut.off</code>
value.</p>
<p><code>Expressed()</code> allows users to choose from several gene
extraction methods (see <code>?Expressed</code> for details):</p>
<ul>
<li><p><code>const</code>: all genes that have at least one stage that
undercuts or exceeds the expression <code>cut.off</code> will be
excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage
<code>ExpressionSet</code> genes passing the expression level
<code>cut.off</code> in 6 stages will be retained in the
<code>ExpressionSet</code>.</p></li>
<li><p><code>min-set</code>: genes passing the expression level
<code>cut.off</code> in <code>ceiling(n/2)</code> stages will be
retained in the <code>ExpressionSet</code>, where <code>n</code> is the
number of stages in the <code>ExpressionSet</code>.</p></li>
<li><p><code>n-set</code>: genes passing the expression level
<code>cut.off</code> in <code>n</code> stages will be retained in the
<code>ExpressionSet</code>. Here, the argument <code>n</code> is
defining the number of stages for which the threshold criteria should be
fulfilled.</p></li>
</ul>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="co"># check number of genes in PhyloExpressionSetExample</span></span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a><span class="fu">nrow</span>(PhyloExpressionSetExample)</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="co">#&gt; [1] 25260</span></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a><span class="co"># remove genes that have an expression level below 8000</span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a><span class="co"># in at least one developmental stage</span></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>FilterConst <span class="ot">&lt;-</span> <span class="fu">Expressed</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>                         <span class="at">cut.off       =</span> <span class="dv">8000</span>,</span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>                         <span class="at">comparison    =</span> <span class="st">&quot;below&quot;</span>, </span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>                         <span class="at">method        =</span> <span class="st">&quot;const&quot;</span>)</span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a><span class="fu">nrow</span>(FilterConst) <span class="co"># check number of retained genes</span></span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a><span class="co">#&gt; [1] 449</span></span></code></pre></div>
<p>Users will observe that only 449 out of 25260 genes in
<code>PhyloExpressionSetExample</code> have an absolute expression level
above <code>8000</code> when omitting genes using
<code>method = &#39;const&#39;</code>. The argument <code>comparison</code>
specifies whether genes having expression levels below, above, or below
AND above (both) the <code>cut.off</code> value should be removed from
the dataset.</p>
<p>The following comparison methods can be selected:</p>
<ul>
<li><code>comparison = &quot;below&quot;</code>: define genes as not expressed
which undercut the <code>cut-off</code> threshold.</li>
<li><code>comparison = &quot;above&quot;</code>: define genes as outliers which
exceed the <code>cut-off</code> threshold.</li>
<li><code>comparison = &quot;both&quot;</code>: remove genes fulfilling the
<code>comparison = &quot;below&quot;</code> <strong>AND</strong>
<code>comparison = &quot;above&quot;</code> criteria.</li>
</ul>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="co"># again: check number of genes in PhyloExpressionSetExample</span></span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a><span class="fu">nrow</span>(PhyloExpressionSetExample)</span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a><span class="co">#&gt; [1] 25260</span></span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a><span class="co"># remove genes that have an expression level above 12000</span></span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a><span class="co"># in at least one developmental stage (outlier removal)</span></span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>FilterConst.above <span class="ot">&lt;-</span> <span class="fu">Expressed</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>                               <span class="at">cut.off       =</span> <span class="dv">12000</span>,</span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a>                               <span class="at">comparison    =</span> <span class="st">&quot;above&quot;</span>, </span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a>                               <span class="at">method        =</span> <span class="st">&quot;const&quot;</span>)</span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a></span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a><span class="fu">nrow</span>(FilterConst.above) <span class="co"># check number of retained genes</span></span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a><span class="co">#&gt; [1] 23547</span></span></code></pre></div>
<p>For this example 25260 - 23547 = 1713 have been classified as
outliers (expression levels above 12000) and were removed from the
dataset.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="co"># again: check number of genes in PhyloExpressionSetExample</span></span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a><span class="fu">nrow</span>(PhyloExpressionSetExample)</span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a><span class="co">#&gt; [1] 25260</span></span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a><span class="co"># remove genes that have an expression level below 8000 AND above 12000</span></span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a><span class="co"># in at least one developmental stage (non-expressed genes AND outlier removal)</span></span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a>FilterConst.both <span class="ot">&lt;-</span>  <span class="fu">Expressed</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a>                               <span class="at">cut.off       =</span> <span class="fu">c</span>(<span class="dv">8000</span>,<span class="dv">12000</span>),</span>
<span id="cb39-9"><a href="#cb39-9" tabindex="-1"></a>                               <span class="at">comparison    =</span> <span class="st">&quot;both&quot;</span>, </span>
<span id="cb39-10"><a href="#cb39-10" tabindex="-1"></a>                               <span class="at">method        =</span> <span class="st">&quot;const&quot;</span>)</span>
<span id="cb39-11"><a href="#cb39-11" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" tabindex="-1"></a><span class="fu">nrow</span>(FilterConst.both) <span class="co"># check number of retained genes</span></span>
<span id="cb39-13"><a href="#cb39-13" tabindex="-1"></a><span class="co">#&gt; [1] 2</span></span></code></pre></div>
<p>When selecting <code>comparison = &#39;both&#39;</code>, the
<code>cut.off</code> argument receives 2 threshold values: the
<em>below</em> <code>cut.off</code> as first element and the
<em>above</em> <code>cut.off</code> as second element. In this case
<code>cut.off = c(8000,12000)</code>. Here, only 2 genes fulfill these
criteria.</p>
<p>Analogously, users can specify the number of stages that should
fulfill the threshold criteria using the <code>n-set</code> method.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="co"># remove genes that have an expression level below 8000</span></span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a><span class="co"># in at least 5 developmental stages (in this case: n = 2 stages fulfilling the criteria)</span></span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>FilterNSet <span class="ot">&lt;-</span> <span class="fu">Expressed</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>                        <span class="at">cut.off       =</span> <span class="dv">8000</span>,</span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>                        <span class="at">method        =</span> <span class="st">&quot;n-set&quot;</span>,</span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>                        <span class="at">comparison    =</span> <span class="st">&quot;below&quot;</span>,</span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>                        <span class="at">n             =</span> <span class="dv">2</span>)</span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a><span class="fu">nrow</span>(FilterMinSet) <span class="co"># check number of retained genes</span></span>
<span id="cb40-10"><a href="#cb40-10" tabindex="-1"></a><span class="co">#&gt; [1] 20</span></span></code></pre></div>
<p>Here, 20 genes are fulfilling these criteria.</p>
</div>
<div id="compute-the-statistical-significance-of-each-replicate-combination" class="section level2">
<h2>Compute the Statistical Significance of Each Replicate
Combination</h2>
<p>In some cases (high variability of replicates) it might be useful to
verify that there is no sequence of replicates (for all possible
combination of replicates) that results in a non-significant
<code>TAI</code> or <code>TDI</code> pattern, when the initial pattern
with combined replicates was shown to be significant.</p>
<p>The <code>CombinatorialSignificance()</code> function implemented in
<code>myTAI</code> allows users to compute the p-values quantifying the
statistical significance of the underlying pattern for all combinations
of replicates.</p>
<div id="a-small-example" class="section level3">
<h3>A small Example:</h3>
<p>Assume a <code>PhyloExpressionSet</code> stores 3 developmental
stages with 3 replicates measured for each stage. The 9 replicates in
total are denoted as: <span class="math inline">\(1.1, 1.2, 1.3, 2.1,
2.2, 2.3, 3.1, 3.2, 3.3\)</span>. Now the function computes the
statistical significance of each pattern derived by the corresponding
combination of replicates, e.g.</p>
<ul>
<li><p>1.1, 2.1, 3.1 : p-value for combination 1</p></li>
<li><p>1.1, 2.2, 3.1 : p-value for combination 2</p></li>
<li><p>1.1, 2.3, 3.1 : p-value for combination 3</p></li>
<li><p>1.2, 2.1, 3.1 : p-value for combination 4</p></li>
<li><p>1.2, 2.1, 3.1 : p-value for combination 5</p></li>
<li><p>1.2, 2.1, 3.1 : p-value for combination 6</p></li>
<li><p>1.3, 2.1, 3.1 : p-value for combination 7</p></li>
<li><p>1.3, 2.2, 3.1 : p-value for combination 8</p></li>
<li><p>1.3, 2.3, 3.1 : p-value for combination 9</p></li>
<li><p>…</p></li>
</ul>
<p>This procedure yields 27 p-values for the <span class="math inline">\(3^3\)</span> (<span class="math inline">\(n^m\)</span>) replicate combinations, where <span class="math inline">\(n\)</span> denotes the number of developmental
stages and <span class="math inline">\(m\)</span> denotes the number of
replicates per stage.</p>
<p>Note that in case users have a large amount of stages/experiments and
a large amount of replicates the computation time will increase by <span class="math inline">\(n^m\)</span>. For 11 stages and 4 replicates,
<span class="math inline">\(4^{11}\)</span> = 4194304 p-values have to
be computed. Each p-value computation itself is based on a permutation
test running with <span class="math inline">\(1,000, 10,000,
...\)</span> or more permutations. Be aware that this might take some
time.</p>
<p>The p-value vector returned by this function can then be used to plot
the p-values to see whether an critical value <span class="math inline">\(\alpha\)</span> is exceeded or not (e.g. <span class="math inline">\(\alpha = 0.05\)</span>).</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># load a standard PhyloExpressionSet</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a><span class="co"># we assume that the PhyloExpressionSetExample</span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a><span class="co"># consists of 3 developmental stages</span></span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a><span class="co"># and 2 replicates for stage 1, 3 replicates for stage 2,</span></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a><span class="co"># and 2 replicates for stage 3</span></span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a><span class="co"># FOR REAL ANALYSES PLEASE USE: permutations = 1000 or 10000</span></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a><span class="co"># BUT </span><span class="al">NOTE</span><span class="co"> THAT THIS TAKES MUCH MORE COMPUTATION TIME</span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a>p.vector <span class="ot">&lt;-</span> <span class="fu">CombinatorialSignificance</span>(<span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a>                                      <span class="at">replicates    =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">2</span>),</span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a>                                      <span class="at">TestStatistic =</span> <span class="st">&quot;FlatLineTest&quot;</span>,</span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a>                                      <span class="at">permutations  =</span> <span class="dv">100</span>,</span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a>                                      <span class="at">parallel      =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code> [1] 2.436296e-03 2.288593e-02 1.608399e-03 1.185615e-02 1.835306e-06 1.077012e-05
 [7] 2.025515e-07 5.148342e-07 1.654885e-07 6.251145e-06 9.265520e-10 1.047479e-06</code></pre>
<p>Users will observe that none of the replicate combinations resulted
in p-values &gt; 0.05 and thus we can assume that the
phylotranscriptomic pattern computed based on collapsed replicates is
not biased by insignificant replicate combinations.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="fu">any</span>(p.vector <span class="sc">&gt;</span> <span class="fl">0.05</span>)</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a><span class="co">#&gt; FALSE</span></span></code></pre></div>
<p><code>CombinatorialSignificance()</code> can perform all significance
tests introduced in the <a href="Introduction.html">Introduction</a> and
<a href="Intermediate.html">Intermediate</a> vignettes.</p>
<p>Furthermore, the <code>parallel</code> argument allows users to
perform significance computations in parallel on a multicore machine.
This will speed up p-value computations for a large number of
combinations.</p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
