<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2024-10-03" />

<title>Intermediate Phylotranscriptomics Analyses</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Intermediate Phylotranscriptomics
Analyses</h1>
<h4 class="date">2024-10-03</h4>



<p>In the <strong>Introduction</strong> vignette users learned how to
perform the most fundamental computations and visualizations for
phylotranscriptomics analyses using the <code>myTAI</code> package.</p>
<p>This <strong>Intermediate</strong> vignette introduces more detailed
analyses and more specialized techniques to investigate the observed
phylotranscriptomics patterns (<code>TAI</code>, <code>TDI</code>,
<code>RE</code>, etc.) and introduces the statistical framework to
quantify the significance of observed phenomena.</p>
<ul>
<li><a href="#flat-line-test">1. Details: Flat Line Test</a></li>
<li><a href="#reductive-hourglass-test">2. Details: Reductive Hourglass
Test</a></li>
<li><a href="#reductive-early-conservation-test">3. Details: Reductive
Early Conservation Test</a></li>
<li><a href="#reverse-hourglass-test">4. Details: Reverse Hourglass
Test</a></li>
<li><a href="#data-transformation">5. Data Transformation for log(TAI)
and sqrt(TAI) patterns</a></li>
</ul>
<div id="investigating-the-statistical-significance-of-phylotranscriptomic-patterns" class="section level2">
<h2>Investigating the Statistical Significance of Phylotranscriptomic
Patterns</h2>
<p>Three methods have been proposed to quantify the statistical
significance of the observed phylotranscriptomics patterns (Quint et
al., 2012; Drost et al., 2015).</p>
<ul>
<li><strong>Flat Line Test</strong></li>
<li><strong>Reductive Hourglass Test</strong></li>
<li><strong>Reductive Early Conservation Test</strong></li>
</ul>
<p>Here, we will build the test statistic of each test step by step so
that future modifications or new test statistics can be built upon the
existing methods implemented in the <code>myTAI</code> package.</p>
<div id="flat-line-test" class="section level3">
<h3>Flat Line Test</h3>
<p>The <strong>Flat Line Test</strong> is a permutation test quantifying
the statistical significance of an observed phylotranscriptomic pattern.
The goal is to detect any evolutionary signal within a developmental
time course that significantly deviates from a flat line.</p>
<p>To build the test statistic we start with a standard
PhyloExpressionSet. The <code>myTAI</code> package provides an example
PhyloExpressionSet named <code>PhyloExpressionSetExample</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(myTAI)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co"># load an example PhyloExpressionSet stored in the myTAI package</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample) </span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co"># look at the standardized data set format</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">head</span>(PhyloExpressionSetExample, <span class="dv">3</span>) </span></code></pre></div>
<pre><code>  Phylostratum      GeneID   Zygote Quadrant Globular     Heart  Torpedo      Bent    Mature
1            1 at1g01040.2 2173.635 1911.200 1152.555 1291.4224 1000.253  962.9772 1696.4274
2            1 at1g01050.1 1501.014 1817.309 1665.309 1564.7612 1496.321 1114.6435 1071.6555
3            1 at1g01070.1 1212.793 1233.002  939.200  929.6195  864.218  877.2060  894.8189</code></pre>
<p>Users will observe that the first column of the
<code>PhyloExpressionSetExample</code> stores the Phylostratum
assignments of the corresponding genes. The permutation test is based on
random sampling of the Phylostratum assignment of genes. The underlying
assumption is that the <code>TAI</code> profile of correctly assigned
Phylostrata is significantly deviating from <code>TAI</code> profiles
based on randomly assigned Phylostrata.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">TAI</span>(PhyloExpressionSetExample) </span></code></pre></div>
<pre><code>  Zygote Quadrant Globular    Heart  Torpedo     Bent   Mature 
3.229942 3.225614 3.107135 3.116693 3.073993 3.176511 3.390334</code></pre>
<p>Visualization:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co"># Visualize the TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample, </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>               <span class="at">p.value       =</span> <span class="cn">FALSE</span> ) </span></code></pre></div>
<pre><code>  Zygote Quadrant Globular    Heart  Torpedo     Bent   Mature 
3.543700 3.573999 3.554707 3.543376 3.582905 3.562813 3.523409 </code></pre>
<p>Visualization:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Visualize the TAI profile based on randomly assigned Phylostrata</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span>  randomPhyloExpressionSetExample, </span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>               <span class="at">p.value       =</span> <span class="cn">FALSE</span> ) </span></code></pre></div>
<p>Users will observe that the visual pattern of the correctly assigned
<code>TAI</code> profile and the randomly assigned <code>TAI</code>
profile differ qualitatively.</p>
<p>Now we investigate the variance of the two observed patterns.</p>
<pre><code>[1] 0.01147725</code></pre>
<pre><code>[1] 0.0004102549</code></pre>
<p>We observe that the variance of the randomly assigned TAI profile is
much smaller than the variance of the correctly assigned TAI profile.
Here we use the variance to quantify the <em>flatness</em> of a given
TAI profile. In theory the variance of a perfect flat line would be
zero. So any TAI profile that is close to zero would resemble a flat
line. But how exactly are the variances of randomly assigned TAI
profiles distributed? For this purpose the <code>bootMatrix()</code>
function was implemented.</p>
<p>The <code>bootMatrix()</code> takes an PhyloExpressionSet or
DivergenceExpressionSet as input and computes N <code>TAI</code> or
<code>TDI</code> profiles based on randomly assigned Phylostrata or
Divergence-Strata.</p>
<pre><code>    Zygote Quadrant Globular    Heart  Torpedo     Bent   Mature
1 3.578000 3.577732 3.552230 3.556756 3.555627 3.553580 3.510978
2 3.590339 3.564544 3.589012 3.598754 3.583721 3.597553 3.633530
3 3.615284 3.616286 3.569486 3.566816 3.575761 3.554789 3.601522
4 3.501172 3.506554 3.486261 3.491282 3.531536 3.541634 3.494997
5 3.450857 3.458339 3.447274 3.463692 3.444261 3.468433 3.497926
6 3.539454 3.543827 3.595298 3.588149 3.582898 3.588062 3.565487</code></pre>
<p>Based on this <code>booMatrix</code> we can compute the variance of
each random TAI profile.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># compute the variance of the random TAI profile for each row</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>variance_vector <span class="ot">&lt;-</span> <span class="fu">apply</span>(randomTAIs, <span class="dv">1</span> , var)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># and visualize the distribution of variances</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="fu">hist</span>(variance_vector, <span class="at">breaks =</span> <span class="dv">100</span>)</span></code></pre></div>
<p>Now it is interesting to see where we can find the variance of the
correctly assigned TAI.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># variance of the TAI profile based on correctly assigned Phylostrata</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>var_real <span class="ot">&lt;-</span> <span class="fu">var</span>(<span class="fu">TAI</span>(PhyloExpressionSetExample)) </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co"># visualize the distribution of variances</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="fu">hist</span>( <span class="at">x      =</span> <span class="fu">c</span>(variance_vector,var_real), </span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>      <span class="at">breaks =</span> <span class="dv">100</span>, </span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>      <span class="at">xlab   =</span> <span class="st">&quot;variance&quot;</span>, </span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>      <span class="at">main   =</span> <span class="st">&quot;Histogram of variance_vector&quot;</span> )</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co"># and plot a red line at the position where we can find the </span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="co"># real variance</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> var_real, <span class="at">lwd =</span> <span class="dv">5</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p>This plot illustrates that variances based on random <code>TAI</code>
profiles seem to have a smaller variance than the variance based on the
correct <code>TAI</code> profile. To obtain a p-value that now
quantifies this difference, we need to fit the histogram of
<code>variance_vector</code> with a specific probability
distribution.</p>
<p>Visually it would be possible to choose a gamma distribution to fit
the histogram of <code>variance_vector</code>. To validate this choice a
<a href="https://openlibrary.org/works/OL1978341W/Probabilistic_techniques_in_exposure_assessment">Cullen
and Frey graph</a> provided by the <strong>fitdistrplus</strong> package
can be used.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co"># install.packages(&quot;fitdistrplus&quot;)</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co"># plot a Cullen and Frey graph</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>fitdistrplus<span class="sc">::</span><span class="fu">descdist</span>(variance_vector)</span></code></pre></div>
<p>Based on the observation that a gamma distribution is a suitable fit
for <code>variance_vector</code>, we can now estimate the parameters of
the gamma distribution that fits the data.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># estimate the parameters: shape and rate using &#39;moment matching estimation&#39;</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>gamma_MME <span class="ot">&lt;-</span> fitdistrplus<span class="sc">::</span><span class="fu">fitdist</span>(variance_vector,<span class="at">distr =</span> <span class="st">&quot;gamma&quot;</span>, <span class="at">method =</span> <span class="st">&quot;mme&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="co"># estimate shape:</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>shape <span class="ot">&lt;-</span> gamma_MME<span class="sc">$</span>estimate[<span class="dv">1</span>]</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co"># estimate the rate:</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>rate <span class="ot">&lt;-</span> gamma_MME<span class="sc">$</span>estimate[<span class="dv">2</span>]</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a><span class="co"># define an expression written as function as input for the curve() function</span></span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>gamma_distr <span class="ot">&lt;-</span> <span class="cf">function</span>(x){ <span class="fu">return</span>(<span class="fu">dgamma</span>(<span class="at">x =</span> x, <span class="at">shape =</span> shape, <span class="at">rate =</span> rate)) }</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a><span class="co"># plot the density function and the histogram of variance_vector</span></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a><span class="fu">curve</span>( <span class="at">expr =</span> gamma_distr,</span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>       <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fu">min</span>(variance_vector),<span class="fu">max</span>(<span class="fu">c</span>(variance_vector,var_real))),</span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>       <span class="at">col  =</span> <span class="st">&quot;steelblue&quot;</span>,</span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a>       <span class="at">lwd  =</span> <span class="dv">5</span>,</span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="st">&quot;Variances&quot;</span>, </span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a>       <span class="at">ylab =</span> <span class="st">&quot;Frequency&quot;</span> )</span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a><span class="co"># plot the histogram of variance_vector</span></span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a>histogram <span class="ot">&lt;-</span> <span class="fu">hist</span>(variance_vector,<span class="at">prob =</span> <span class="cn">TRUE</span>,<span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">breaks =</span> <span class="dv">100</span>)</span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a><span class="fu">rug</span>(variance_vector)</span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a></span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a><span class="co"># plot a red line at the position where we can find the real variance</span></span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> var_real, <span class="at">lwd =</span> <span class="dv">5</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p>Using the gamma distribution with estimated parameters the
corresponding p-value of <code>var_real</code> can be computed.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># p-value of var_real</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">pgamma</span>(var_real, <span class="at">shape =</span> shape,<span class="at">rate =</span> rate, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Hence, the variance of the correct TAI profile significantly deviates
from random TAI profiles and this allows us to assume that the
underlying TAI profile captures a real evolutionary signal.</p>
</div>
<div id="using-the-flatlinetest-function" class="section level3">
<h3>Using the <code>FlatLineTest()</code> Function</h3>
<p>This entire procedure of computing the p-value having the variance of
TAI profiles as test statistic is done by the
<code>FlatLineTest()</code> function.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Perform the FlatLineTest</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="fu">FlatLineTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample, </span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>              <span class="at">permutations  =</span> <span class="dv">1000</span> )</span></code></pre></div>
<p>This function returns the p-value of the test statistic.</p>
<p>Additionally the <code>FlatLineTest()</code> function allows to
investigate the goodness of the test statistic.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># perform the FlatLineTest and investigate the goodness of the test statistic</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="fu">FlatLineTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample, </span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>              <span class="at">permutations  =</span> <span class="dv">1000</span>, </span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>              <span class="at">plotHistogram =</span> <span class="cn">TRUE</span> )</span></code></pre></div>
<p>The <code>plotHistogram</code> argument specifies whether analytics
plots shall be drawn to quantify the goodness of the test statistic
returned by the <code>FlatLineTest</code>.</p>
<p>The three resulting plots show:</p>
<ul>
<li><p>a <a href="https://openlibrary.org/works/OL1978341W/Probabilistic_techniques_in_exposure_assessment">Cullen
and Frey graph</a></p></li>
<li><p>a histogram of the test statistic and the corresponding <a href="https://en.wikipedia.org/wiki/Gamma_distribution">gamma
distribution</a> that was fitted to the test statistic</p></li>
<li><p>a plot showing the p-values (<code>p_flt</code>) for 10
individual runs. Since the underlying test statistic is generated by a
permutation test, the third plot returned by <code>FlatLineTest()</code>
shows the influence of different permutations to the corresponding
p-value</p></li>
</ul>
<p>In other words, to test whether or not the underlying permutation of
the permutation test is causing the significance of the p-value, you can
specify the <code>runs</code> argument within the
<code>FlatLineTest()</code> function to perform several independent
runs. In case there exists a permutation that causes a previous
significant p-value to become non-significant, the corresponding
phylotranscriptomic pattern shouldn’t be considered as statistically
significant.</p>
</div>
<div id="reductive-hourglass-test" class="section level3">
<h3>Reductive Hourglass Test</h3>
<p>The <strong>Reductive Hourglass Test</strong> has been developed to
statistically evaluate the existence of a phylotranscriptomic hourglass
pattern based on TAI or TDI computations. The corresponding p-value
quantifies the probability that a given TAI or TDI pattern (or any
phylotranscriptomics pattern) does not follow an hourglass like shape. A
p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics
pattern does indeed follow an hourglass (high-low-high) shape.</p>
<p>To build the test statistic again we start with a standard
PhyloExpressionSet.</p>
<pre><code>  Phylostratum      GeneID   Zygote Quadrant Globular     Heart  Torpedo      Bent    Mature
1            1 at1g01040.2 2173.635 1911.200 1152.555 1291.4224 1000.253  962.9772 1696.4274
2            1 at1g01050.1 1501.014 1817.309 1665.309 1564.7612 1496.321 1114.6435 1071.6555
3            1 at1g01070.1 1212.793 1233.002  939.200  929.6195  864.218  877.2060  894.8189
</code></pre>
<p>And again compute the <code>TAI()</code> profile of the
<code>PhyloExpressionSetExample</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">TAI</span>(PhyloExpressionSetExample) </span></code></pre></div>
<pre><code>  Zygote Quadrant Globular    Heart  Torpedo     Bent   Mature 
3.229942 3.225614 3.107135 3.116693 3.073993 3.176511 3.390334 
</code></pre>
<p>Visualization:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># visualize the TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample, </span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>               <span class="at">p.value       =</span> <span class="cn">FALSE</span> ) </span></code></pre></div>
<p>The Reductive Hourglass Test is a permutation test based on the
following test statistic.</p>
<ol style="list-style-type: decimal">
<li><p>A set of developmental stages is partitioned into three modules -
<code>early</code>, <code>mid</code>, and <code>late</code> - based on
prior biological knowledge (see Drost et al., 2015 for
details).</p></li>
<li><p>The mean TAI or TDI value for each of the three modules <span class="math inline">\(T_{early}\)</span>, <span class="math inline">\(T_{mid}\)</span>, and <span class="math inline">\(T_{late}\)</span> are computed.</p></li>
<li><p>The two differences D1 = <span class="math inline">\(T_{early}\)</span> - <span class="math inline">\(T_{mid}\)</span> and D2 = <span class="math inline">\(T_{late}\)</span> - <span class="math inline">\(T_{mid}\)</span> are calculated.</p></li>
<li><p>The minimum <span class="math inline">\(D_{min}\)</span> of D1
and D2 is computed as final test statistic of the <strong>Reductive
Hourglass Test</strong>.</p></li>
</ol>
<p>In order to determine the statistical significance of an observed
minimum difference <span class="math inline">\(D_{min}\)</span> the
following permutation test was performed. Based on the
<code>bootMatrix()</code> <span class="math inline">\(D_{min}\)</span>
is calculated from each of the permuted TAI or TDI profiles,
approximated by a Gaussian distribution with method of moments estimated
parameters returned by <code>fitdistrplus::fitdist()</code>, and the
corresponding p-value is computed by <code>pnorm</code> given the
estimated parameters of the Gaussian distribution. The goodness of fit
for the random vector <span class="math inline">\(D_{min}\)</span> is
statistically quantified by a Lilliefors (Kolmogorov-Smirnov) test for
normality.</p>
<p>To perform the <strong>Reductive Hourglass Test</strong> you can use
the <code>ReductiveHourglassTest()</code> function. Using this function
you need to divide the given phylotranscriptomic pattern into three
developmental modules:</p>
<ul>
<li>early module</li>
<li>mid module</li>
<li>late module</li>
</ul>
<p>This can be done using the <code>modules</code> argument:
<code>module = list(early = 1:2, mid = 3:5, late = 6:7)</code>. In this
example (<code>PhyloExpressionSetExample</code>) we divide the
corresponding developmental process into the three modules:</p>
<ul>
<li>early module: Stages 1 - 2 = Zygote and Quadrant</li>
<li>mid module: Stages 3 - 5 = Globular, Heart, and Torpedo</li>
<li>late module: Stages 6 - 7 = Bent and Mature</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Perform the Reductive Hourglass Test</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="fu">ReductiveHourglassTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>                        <span class="at">modules       =</span> <span class="fu">list</span>(<span class="at">early =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">mid =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">late =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>                        <span class="at">lillie.test   =</span> <span class="cn">TRUE</span> ) </span></code></pre></div>
<p>The corresponding output shows the p-value returned by the
<strong>Reductive Hourglass Test</strong>, the standard deviation of
randomly permuted TAI profiles returned by <code>bootMatrix()</code>
(<code>apply(bootMatrix(PhyloExpressionSetExample) , 2 , sd)</code>) and
in case the argument <code>lillie.test = TRUE</code>, a logical value
representing the goodness of fit statistic returned by the Lilliefors
(Kolmogorov-Smirnov) test for normality. In case
<code>lillie.test</code> is <code>TRUE</code> the corresponding
Lilliefors (Kolmogorov-Smirnov) test passed the goodness of fit
criterion. In case <code>lillie.test</code> is <code>FALSE</code> the
corresponding goodness of fit by a normal distribution is not
statistically significant.</p>
<p>Analogous to the <code>plotHistogram</code> argument that is present
in the <code>FlatLineTest()</code> function, the
<code>ReductiveHourglassTest()</code> function also takes an argument
<code>plotHistogram</code>. When <code>plotHistogram = TRUE</code>, the
<code>ReductiveHourglassTest()</code> function returns a multi-plot
showing:</p>
<ul>
<li><p>A Cullen and Frey skewness-kurtosis plot. This plot illustrates
which distributions seem plausible to fit the resulting permutation
vector <span class="math inline">\(D_{min}\)</span>. Here a normal
distribution seems most plausible.</p></li>
<li><p>A histogram of <span class="math inline">\(D_{min}\)</span>
combined with the density plot is visualized. <span class="math inline">\(D_{min}\)</span> is then fitted by a normal
distribution. The corresponding parameters are estimated by moment
matching estimation.</p></li>
<li><p>A plot showing the p-values for N independent runs to verify that
a specific p-value is biased by a specific permutation order.</p></li>
<li><p>A bar plot showing the number of cases in which the underlying
goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test for
normality) has shown to be significant (<code>TRUE</code>) or not
significant (<code>FALSE</code>). This allows to quantify the
permutation bias and their implications on the goodness of fit.</p></li>
</ul>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># perform the Reductive Hourglass Test and plot the test statistic</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="fu">ReductiveHourglassTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>                        <span class="at">modules       =</span> <span class="fu">list</span>(<span class="at">early =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">mid =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">late =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>                        <span class="at">plotHistogram =</span> <span class="cn">TRUE</span>, </span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>                        <span class="at">lillie.test   =</span> <span class="cn">TRUE</span> ) </span></code></pre></div>
<p>The corresponding output shows the <code>summary statistics</code> of
the fitted normal distribution as well as the p-value, standard
deviation, and Lilliefors (Kolmogorov-Smirnov) test result.</p>
<p>This example output nicely illustrates that although the Lilliefors
(Kolmogorov-Smirnov) test for normality is violated for some
permutations, the Cullen and Frey graph shows that there is no better
approximation than a normal distribution (which is also supported
visually by investigating the fitted frequency distribution). The
corresponding p-value returned by <code>ReductiveHourglassTest()</code>
is significant and illustrates that the observed phylotranscriptomic
pattern of PhyloExpressionSetExample does follow the Hourglass Model
assumption.</p>
</div>
<div id="reductive-early-conservation-test" class="section level3">
<h3>Reductive Early Conservation Test</h3>
<p>The Early Conservation Test has been developed to statistically
evaluate the existence of a monotonically increasing phylotranscriptomic
pattern based on TAI or TDI computations. The corresponding p-value
quantifies the probability that a given TAI or TDI pattern (or any
phylotranscriptomic pattern) does not follow an early conservation like
pattern. A p-value &lt; 0.05 indicates that the corresponding
phylotranscriptomics pattern does indeed follow an early conservation
(low-high-high) shape.</p>
<p>To build the test statistic again we start with a standard
PhyloExpressionSet.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">library</span>(myTAI)</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="co"># load an example PhyloExpressionSet stored in the myTAI package</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample) </span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a><span class="co"># look at the standardized data set format</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a><span class="fu">head</span>(PhyloExpressionSetExample, <span class="dv">3</span>) </span></code></pre></div>
<p>And again compute the <code>TAI()</code> profile of the
<code>PhyloExpressionSetExample</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="co"># TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="fu">TAI</span>(PhyloExpressionSetExample) </span></code></pre></div>
<pre><code>  Zygote Quadrant Globular    Heart  Torpedo     Bent   Mature 
3.229942 3.225614 3.107135 3.116693 3.073993 3.176511 3.390334</code></pre>
<p>Visualization:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># Visualize the TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample, </span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>               <span class="at">p.value       =</span> <span class="cn">FALSE</span> ) </span></code></pre></div>
<p>The reductive early conservation test is a permutation test based on
the following test statistic.</p>
<ol style="list-style-type: decimal">
<li><p>A set of developmental stages is partitioned into three modules -
<code>early</code>, <code>mid</code>, and <code>late</code> - based on
prior biological knowledge.</p></li>
<li><p>The mean TAI or TDI value for each of the three modules <span class="math inline">\(T_{early}\)</span>, <span class="math inline">\(T_{mid}\)</span>, and <span class="math inline">\(T_{late}\)</span> are computed.</p></li>
<li><p>The two differences D1 = <span class="math inline">\(T_{mid}\)</span> - <span class="math inline">\(T_{early}\)</span> and D2 = <span class="math inline">\(T_{late}\)</span> - <span class="math inline">\(T_{early}\)</span> are calculated.</p></li>
<li><p>The minimum <span class="math inline">\(D_{min}\)</span> of D1
and D2 is computed as final test statistic of the <strong>Reductive
Early Conservation Test</strong>.</p></li>
</ol>
<p>In order to determine the statistical significance of an observed
minimum difference <span class="math inline">\(D_{min}\)</span> the
following permutation test was performed. Based on the
<code>bootMatrix()</code> <span class="math inline">\(D_{min}\)</span>
is calculated from each of the permuted TAI or TDI profiles,
approximated by a Gaussian distribution with method of moments estimated
parameters returned by <code>fitdistrplus::fitdist()</code>, and the
corresponding p-value is computed by <code>pnorm</code> given the
estimated parameters of the Gaussian distribution. The goodness of fit
for the random vector <span class="math inline">\(D_{min}\)</span> is
statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test for
normality.</p>
<p>To perform the <strong>Reductive Early Conservation Test</strong> you
can use the <code>EarlyConservationTest()</code> function. Using this
function you need to divide the given phylotranscriptomics pattern into
three developmental modules:</p>
<ul>
<li>early module</li>
<li>mid module</li>
<li>late module</li>
</ul>
<p>This can be done using the <code>modules</code> argument:
<code>module = list(early = 1:2, mid = 3:5, late = 6:7)</code>. In this
example (<code>PhyloExpressionSetExample</code>) we divide the
corresponding developmental process into the three modules:</p>
<ul>
<li>early module: Stages 1 - 2 = Zygote and Quadrant</li>
<li>mid module: Stages 3 - 5 = Globular, Heart, and Torpedo</li>
<li>late module: Stages 6 - 7 = Bent and Mature</li>
</ul>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="co"># Perform the Reductive Early Conservation Test</span></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a><span class="fu">EarlyConservationTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>                       <span class="at">modules       =</span> <span class="fu">list</span>(<span class="at">early =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">mid =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">late =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>                       <span class="at">lillie.test   =</span> <span class="cn">TRUE</span> ) </span></code></pre></div>
<p>Analogous to the <code>plotHistogram</code> argument that is present
in the <code>FlatLineTest()</code> and
<code>ReductiveHourglassTest()</code> function, the
<code>EarlyConservationTest()</code> function also takes an argument
<code>plotHistogram</code>. When <code>plotHistogram = TRUE</code>, the
<code>EarlyConservationTest()</code> function returns a multi-plot
showing:</p>
<ul>
<li><p>A Cullen and Frey skewness-kurtosis plot. This plot illustrates
which distributions seem plausible to fit the resulting permutation
vector <span class="math inline">\(D_{min}\)</span>. Again a normal
distribution seems most appropriate.</p></li>
<li><p>A histogram of <span class="math inline">\(D_{min}\)</span>
combined with the density plot is visualized. <span class="math inline">\(D_{min}\)</span> is then fitted by a normal
distribution. The corresponding parameters are estimated by moment
matching estimation.</p></li>
<li><p>A plot showing the p-values for N independent runs to verify that
a specific p-value is biased by a specific permutation order.</p></li>
<li><p>A bar plot showing the number of cases in which the underlying
goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test for
normality) has shown to be significant (<code>TRUE</code>) or not
significant (<code>FALSE</code>). This allows to quantify the
permutation bias and their implications on the goodness of fit.</p></li>
</ul>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="co"># perform the Reductive Early Conservation Test and plot the test statistic</span></span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a><span class="fu">EarlyConservationTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>                       <span class="at">modules       =</span> <span class="fu">list</span>(<span class="at">early =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">mid =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">late =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>                       <span class="at">plotHistogram =</span> <span class="cn">TRUE</span>, </span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a>                       <span class="at">lillie.test   =</span> <span class="cn">TRUE</span> ) </span></code></pre></div>
<p>This example output nicely illustrates that although the Lilliefors
(Kolmogorov-Smirnov) test for normality is violated, the <a href="https://openlibrary.org/works/OL1978341W/Probabilistic_techniques_in_exposure_assessment">Cullen
and Frey graph</a> shows that there is no better approximation than a
normal distribution (which is also supported visually by investigating
the fitted frequency distribution). The corresponding p-value returned
by the <code>EarlyConservationTest()</code> is highly non-significant
and illustrates that the observed phylotranscriptomic pattern of
<code>PhyloExpressionSetExample</code> does not follow the Early
Conservation Model assumption.</p>
<p>This example shall illustrate that finding the right test statistic
is a multi-step process of investigating different properties of the
underlying permutation test. Although single aspects might fit or fit
not corresponding criteria, the overall impression (sum of all
individual analyses) must be considered to obtain a valid p-value.</p>
</div>
<div id="reverse-hourglass-test" class="section level3">
<h3>Reverse Hourglass Test</h3>
<p>The <strong>Reverse Hourglass Test</strong> has been developed to
statistically evaluate the existence of a
<code>reverse hourglass pattern</code> based on TAI or TDI computations.
The corresponding p-value quantifies the probability that a given TAI or
TDI pattern (or any phylotranscriptomics pattern) does not follow an
reverse hourglass like shape. A p-value &lt; 0.05 indicates that the
corresponding phylotranscriptomics pattern does indeed follow a reverse
hourglass (low-high-low) shape.</p>
<p>To build the test statistic again we start with a standard
PhyloExpressionSet.</p>
<pre><code>  Phylostratum      GeneID   Zygote Quadrant Globular     Heart  Torpedo      Bent    Mature
1            1 at1g01040.2 2173.635 1911.200 1152.555 1291.4224 1000.253  962.9772 1696.4274
2            1 at1g01050.1 1501.014 1817.309 1665.309 1564.7612 1496.321 1114.6435 1071.6555
3            1 at1g01070.1 1212.793 1233.002  939.200  929.6195  864.218  877.2060  894.8189
</code></pre>
<p>And again compute the <code>TAI()</code> profile of the
<code>PhyloExpressionSetExample</code>.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a><span class="fu">TAI</span>(PhyloExpressionSetExample) </span></code></pre></div>
<pre><code>  Zygote Quadrant Globular    Heart  Torpedo     Bent   Mature 
3.229942 3.225614 3.107135 3.116693 3.073993 3.176511 3.390334 
</code></pre>
<p>Visualization:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co"># visualize the TAI profile of correctly assigned Phylostrata</span></span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample, </span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>               <span class="at">p.value       =</span> <span class="cn">FALSE</span> ) </span></code></pre></div>
<p>The Reverse Hourglass Test is a permutation test based on the
following test statistic.</p>
<ol style="list-style-type: decimal">
<li><p>A set of developmental stages is partitioned into three modules -
<code>early</code>, <code>mid</code>, and <code>late</code> - based on
prior biological knowledge (see Drost et al., 2015 for
details).</p></li>
<li><p>The mean TAI or TDI value for each of the three modules <span class="math inline">\(T_{early}\)</span>, <span class="math inline">\(T_{mid}\)</span>, and <span class="math inline">\(T_{late}\)</span> are computed.</p></li>
<li><p>The two differences D1 = <span class="math inline">\(T_{mid}\)</span> - <span class="math inline">\(T_{early}\)</span> and D2 = <span class="math inline">\(T_{mid}\)</span> - <span class="math inline">\(T_{late}\)</span> are calculated.</p></li>
<li><p>The minimum <span class="math inline">\(D_{min}\)</span> of D1
and D2 is computed as final test statistic of the <strong>Reverse
Hourglass Test</strong>.</p></li>
</ol>
<p>In order to determine the statistical significance of an observed
minimum difference <span class="math inline">\(D_{min}\)</span> the
following permutation test was performed. Based on the
<code>bootMatrix()</code> <span class="math inline">\(D_{min}\)</span>
is calculated from each of the permuted TAI or TDI profiles,
approximated by a Gaussian distribution with method of moments estimated
parameters returned by <code>fitdistrplus::fitdist()</code>, and the
corresponding p-value is computed by <code>pnorm</code> given the
estimated parameters of the Gaussian distribution. The goodness of fit
for the random vector <span class="math inline">\(D_{min}\)</span> is
statistically quantified by a Lilliefors (Kolmogorov-Smirnov) test for
normality.</p>
<p>To perform the <strong>Reverse Hourglass Test</strong> you can use
the <code>ReverseHourglassTest()</code> function. Using this function
you need to divide the given phylotranscriptomic pattern into three
developmental modules:</p>
<ul>
<li>early module</li>
<li>mid module</li>
<li>late module</li>
</ul>
<p>This can be done using the <code>modules</code> argument:
<code>module = list(early = 1:2, mid = 3:5, late = 6:7)</code>. In this
example (<code>PhyloExpressionSetExample</code>) we divide the
corresponding developmental process into the three modules:</p>
<ul>
<li>early module: Stages 1 - 2 = Zygote and Quadrant</li>
<li>mid module: Stages 3 - 5 = Globular, Heart, and Torpedo</li>
<li>late module: Stages 6 - 7 = Bent and Mature</li>
</ul>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="co"># Perform the Reverse Hourglass Test</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a><span class="fu">ReverseHourglassTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>                        <span class="at">modules       =</span> <span class="fu">list</span>(<span class="at">early =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">mid =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">late =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>                        <span class="at">lillie.test   =</span> <span class="cn">TRUE</span> ) </span></code></pre></div>
<pre><code>$p.value
[1] 1

$std.dev
[1] 0.05598325 0.05469016 0.05386737 0.05248817 0.05161525 0.05411684 0.05605765

$lillie.test
[1] TRUE</code></pre>
<p>The corresponding output shows the p-value returned by the
<strong>Reductive Hourglass Test</strong>, the standard deviation of
randomly permuted TAI profiles returned by <code>bootMatrix()</code>
(<code>apply(bootMatrix(PhyloExpressionSetExample) , 2 , sd)</code>) and
in case the argument <code>lillie.test = TRUE</code>, a logical value
representing the goodness of fit statistic returned by the Lilliefors
(Kolmogorov-Smirnov) test for normality. In case
<code>lillie.test</code> is <code>TRUE</code> the corresponding
Lilliefors (Kolmogorov-Smirnov) test passed the goodness of fit
criterion. In case <code>lillie.test</code> is <code>FALSE</code> the
corresponding goodness of fit by a normal distribution is not
statistically significant.</p>
<p>Analogous to the <code>plotHistogram</code> argument that is present
in the <code>FlatLineTest()</code> function, the
<code>ReductiveHourglassTest()</code> function also takes an argument
<code>plotHistogram</code>. When <code>plotHistogram = TRUE</code>, the
<code>ReductiveHourglassTest()</code> function returns a multi-plot
showing:</p>
<ul>
<li><p>A Cullen and Frey skewness-kurtosis plot. This plot illustrates
which distributions seem plausible to fit the resulting permutation
vector <span class="math inline">\(D_{min}\)</span>. Here a normal
distribution seems most plausible.</p></li>
<li><p>A histogram of <span class="math inline">\(D_{min}\)</span>
combined with the density plot is visualized. <span class="math inline">\(D_{min}\)</span> is then fitted by a normal
distribution. The corresponding parameters are estimated by moment
matching estimation.</p></li>
<li><p>A plot showing the p-values for N independent runs to verify that
a specific p-value is biased by a specific permutation order.</p></li>
<li><p>A bar plot showing the number of cases in which the underlying
goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test for
normality) has shown to be significant (<code>TRUE</code>) or not
significant (<code>FALSE</code>). This allows to quantify the
permutation bias and their implications on the goodness of fit.</p></li>
</ul>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="co"># perform the Reverse Hourglass Test and plot the test statistic</span></span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a><span class="fu">ReverseHourglassTest</span>( <span class="at">ExpressionSet =</span> PhyloExpressionSetExample,</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>                        <span class="at">modules       =</span> <span class="fu">list</span>(<span class="at">early =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">mid =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">late =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>),</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>                        <span class="at">plotHistogram =</span> <span class="cn">TRUE</span>, </span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a>                        <span class="at">lillie.test   =</span> <span class="cn">TRUE</span> ) </span></code></pre></div>
<p>The corresponding output shows the <code>summary statistics</code> of
the fitted normal distribution as well as the p-value, standard
deviation, and Lilliefors (Kolmogorov-Smirnov) test result.</p>
<p>This example output nicely illustrates that although the Lilliefors
(Kolmogorov-Smirnov) test for normality is violated for some
permutations, the Cullen and Frey graph shows that there is no better
approximation than a normal distribution (which is also supported
visually by investigating the fitted frequency distribution). The
corresponding p-value returned by <code>ReverseHourglassTest()</code> is
significant and illustrates that the observed phylotranscriptomic
pattern of PhyloExpressionSetExample does follow the Hourglass Model
assumption.</p>
</div>
</div>
<div id="data-transformation" class="section level2">
<h2>Data Transformation</h2>
<p>Motivated by the discussion raised by Piasecka et al., 2013, the
influence of gene expression transformation on the global
phylotranscriptomic patterns does not seem negligible. Hence, different
transformations can result in qualitatively different TAI or TDI
patterns.</p>
<p>Initially, the TAI and TDI formulas were defined for absolute
expression levels. So using the initial TAI and TDI formulas with
transformed expression levels can result in qualitatively different
patterns when compared with non-transformed expression levels, but might
also belong to a different class of models, since different valid
expression level transformation functions result in different
patterns.</p>
<p>The purpose of the <code>tf()</code> function is to allow the user to
study the qualitative impact of different transformation functions on
the global TAI and TDI pattern, or on any subsequent phylotranscriptomic
analysis.</p>
<p>The examples using the <code>PhyloExpressionSetExample</code> data
set show that using common gene expression transformation functions:
log2 (Quackenbush, 2001 and 2002), sqrt (Yeung et al., 2001), boxcox, or
inverse hyperbolic sine transformation, each transformation results in
qualitatively different patterns. Nevertheless, for each resulting
pattern the statistical significance can be tested using either the
<code>FlatLineTest()</code>, <code>ReductiveHourglassTest()</code>, or
<code>EarlyConservationTest()</code> (Drost et al., 2015) to quantify
the significance of observed patterns.</p>
<p>The <code>tf()</code> function takes a standard
<code>PhyloExpressionSet</code> or <code>DivergenceExpressionSet</code>
and transformation function and returns the corresponding
<code>ExpressionSet</code> with transformed gene expression levels.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">library</span>(myTAI)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a><span class="co"># a simple example is to transform the gene expression levels of a given PhyloExpressionSet</span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a><span class="co"># using a sqrt or log2 transformation</span></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>PES.sqrt <span class="ot">&lt;-</span> <span class="fu">tf</span>(PhyloExpressionSetExample, sqrt)</span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a><span class="fu">head</span>(PES.sqrt)</span></code></pre></div>
<pre><code>#&gt;   Phylostratum      GeneID    Zygote  Quadrant  Globular     Heart   Torpedo      Bent   Mature
#&gt; 1            1 at1g01040.2  46.62226  43.71728  33.94930  35.93637  31.62678  31.03187 41.18771
#&gt; 2            1 at1g01050.1  38.74292  42.62990  40.80820  39.55706  38.68230  33.38628 32.73615
#&gt; 3            1 at1g01070.1  34.82517  35.11413  30.64637  30.48966  29.39759  29.61766 29.91352
#&gt; 4            1 at1g01080.2  31.88919  30.60039  34.37060  36.46195  37.31813  35.88836 29.34724
#&gt; 5            1 at1g01090.1 106.88576 129.53057 185.38244 199.43831 237.13197 258.80566 88.16213
#&gt; 6            1 at1g01120.1  29.05239  28.06409  29.31939  30.52242  30.70579  29.50021 28.15589</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>PES.log2 <span class="ot">&lt;-</span> <span class="fu">tf</span>(PhyloExpressionSetExample, log2)</span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a><span class="fu">head</span>(PES.log2)</span></code></pre></div>
<pre><code>  Phylostratum      GeneID    Zygote  Quadrant  Globular     Heart   Torpedo      Bent    Mature
1            1 at1g01040.2 11.085894 10.900263 10.170620 10.334745  9.966149  9.911358 10.728284
2            1 at1g01050.1 10.551722 10.827588 10.701574 10.611727 10.547204 10.122367 10.065625
3            1 at1g01070.1 10.244117 10.267960  9.875289  9.860497  9.755251  9.776772  9.805452
4            1 at1g01080.2  9.989991  9.870956 10.206206 10.376639 10.443610 10.330888  9.750306
5            1 at1g01090.1 13.479852 14.034298 15.068722 15.279598 15.779093 16.031451 12.924174
6            1 at1g01120.1  9.721170  9.621306  9.747567  9.863595  9.880877  9.765307  9.630730</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># in case a given PhyloExpressionSet already stores gene expression levels</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a><span class="co"># that are log2 transformed and need to be re-transformed to absolute</span></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a><span class="co"># expression levels, to perform subsequent phylotranscriptomics analyses</span></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a><span class="co"># (that are defined for absolute expression levels), </span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a><span class="co"># one can re-transform a PhyloExpressionSet like this:</span></span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a>PES.absolute <span class="ot">&lt;-</span> <span class="fu">tf</span>(PES.log2 , <span class="cf">function</span>(x) <span class="dv">2</span><span class="sc">^</span>x)</span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a><span class="co"># which should be the same as  PhyloExpressionSetExample :</span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a><span class="fu">head</span>(PhyloExpressionSetExample)</span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a><span class="fu">head</span>(PES.absolute)</span></code></pre></div>
<pre><code>&gt; head(PhyloExpressionSetExample)
  Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart    Torpedo       Bent    Mature
1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224  1000.2529   962.9772 1696.4274
2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612  1496.3207  1114.6435 1071.6555
3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195   864.2180   877.2060  894.8189
4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734  1392.6429  1287.9746  861.2605
5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405 56231.5689 66980.3673 7772.5617
6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180   942.8453   870.2625  792.7542
&gt; head(PES.absolute)
  Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart    Torpedo       Bent    Mature
1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224  1000.2529   962.9772 1696.4274
2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612  1496.3207  1114.6435 1071.6555
3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195   864.2180   877.2060  894.8189
4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734  1392.6429  1287.9746  861.2605
5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405 56231.5689 66980.3673 7772.5617
6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180   942.8453   870.2625  792.7542</code></pre>
<p>When transforming the <code>ExpressionMatrix</code> of the
<code>PhyloExpressionSetExample</code> using different transformation
functions, the resulting phylotranscriptomic patterns qualitatively
differ:</p>
<div id="log2-transformation-tai" class="section level3">
<h3>log2 transformation (TAI)</h3>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a><span class="co"># plotting the TAI using log2 transformed expression levels</span></span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a><span class="co"># and performing the Flat Line Test to obtain the p-value</span></span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> <span class="fu">tf</span>(PhyloExpressionSetExample, log2),</span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>               <span class="at">TestStatistic =</span> <span class="st">&quot;FlatLineTest&quot;</span>,</span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>               <span class="at">xlab          =</span> <span class="st">&quot;Ontogeny&quot;</span>, </span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>               <span class="at">ylab          =</span> <span class="st">&quot;TAI&quot;</span> )</span></code></pre></div>
</div>
<div id="sqrt-transformation-tai" class="section level3">
<h3>sqrt transformation (TAI)</h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">data</span>(PhyloExpressionSetExample)</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="co"># plotting the TAI using sqrt transformed expression levels</span></span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a><span class="co"># and performing the Flat Line Test to obtain the p-value</span></span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> <span class="fu">tf</span>(PhyloExpressionSetExample, sqrt),</span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a>               <span class="at">TestStatistic =</span> <span class="st">&quot;FlatLineTest&quot;</span>,</span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a>               <span class="at">xlab          =</span> <span class="st">&quot;Ontogeny&quot;</span>, </span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>               <span class="at">ylab          =</span> <span class="st">&quot;TAI&quot;</span> )</span></code></pre></div>
<p>For the <code>PhyloExpressionSetExample</code> all transformations
result in a significant phylotranscriptomics pattern deviating from a
flat line.</p>
<p>Nevertheless, it is not clear which transformation is the most
appropriate one since the original TAI and TDI measure were defined for
absolute expression levels.</p>
<p>The same accounts for <code>TDI</code> profiles:</p>
</div>
<div id="log2-transformation-tdi" class="section level3">
<h3>log2 transformation (TDI)</h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">data</span>(DivergenceExpressionSetExample)</span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a><span class="co"># plotting the TDI using log2 transformed expression levels</span></span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a><span class="co"># and performing the Flat Line Test to obtain the p-value</span></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> <span class="fu">tf</span>(DivergenceExpressionSetExample, log2),</span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>               <span class="at">TestStatistic =</span> <span class="st">&quot;FlatLineTest&quot;</span>,</span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a>               <span class="at">xlab          =</span> <span class="st">&quot;Ontogeny&quot;</span>, </span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a>               <span class="at">ylab          =</span> <span class="st">&quot;TDI&quot;</span> )</span></code></pre></div>
</div>
<div id="sqrt-transformation-tdi" class="section level3">
<h3>sqrt transformation (TDI)</h3>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">data</span>(DivergenceExpressionSetExample)</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a><span class="co"># plotting the TDI using sqrt transformed expression levels</span></span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a><span class="co"># and performing the Flat Line Test to obtain the p-value</span></span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a><span class="fu">PlotSignature</span>( <span class="at">ExpressionSet =</span> <span class="fu">tf</span>(DivergenceExpressionSetExample, sqrt),</span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>               <span class="at">TestStatistic =</span> <span class="st">&quot;FlatLineTest&quot;</span>,</span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a>               <span class="at">xlab          =</span> <span class="st">&quot;Ontogeny&quot;</span>, </span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a>               <span class="at">ylab          =</span> <span class="st">&quot;TDI&quot;</span> )</span></code></pre></div>
<p>As a result, observed patterns should always be quantified using
statistical tests (for ex. <code>FlatLineTest()</code>,
<code>ReductiveHourglassTest()</code>, and
<code>EarlyConservationTest()</code>). In case the observed pattern is
significant, qualitative differences of the observed patterns based on
different data transformations must be investigated in more detail,
since most data transformations are known to cause different effects on
a measure that isn’t robust against data transformations.</p>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Domazet-Loso, T. &amp; Tautz, D. <strong>A phylogenetically based
transcriptome age index mirrors ontogenetic divergence
patterns</strong>. <em>Nature</em> <strong>468</strong>, 815-8
(2010).</p>
<p>Quint, M. <em>et al</em>. <strong>A transcriptomic hourglass in plant
embryogenesis</strong>. <em>Nature</em> <strong>490</strong>, 98-101
(2012).</p>
<p>Drost, H.G. <em>et al</em>. <strong>Evidence for Active Maintenance
of Phylotranscriptomic Hourglass Patterns in Animal and Plant
Embryogenesis</strong>. <em>Mol. Biol. Evol.</em> <strong>32</strong>
(5), 1221-1231 (2015).</p>
<p>Piasecka, B. <em>et al</em>. <strong>The hourglass and the early
conservation models - co-existing patterns of developmental constraints
in vertebrates</strong>. <em>PLoS Genet</em>.
<strong>9</strong>:e1003476 (2013).</p>
<p>Quackenbush, J. Computational Analysis of Microarray Data. <em>Nature
Reviews</em>. <strong>2</strong>, 418-427 (2001).</p>
<p>Quackenbush, J. Microarray data normalization and transformation.
<em>Nature Genetics</em>. <strong>32</strong>, 496-501 (2002).</p>
<p>Yeung, K.Y. <em>et al</em>. Model-based clustering and data
transformations for gene expression data. <em>Bioinformatics</em>.
<strong>17</strong>, 977-987 (2001).</p>
<p>Yeung, K.Y. <em>et al</em>. Supplement to <em>Model-based clustering
and data transformations for gene expression data - Data Transformations
and the Gaussian mixture assumption</em>. <em>Bioinformatics</em>.
<strong>17</strong>, 977-987 (2001).</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
