<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2024-10-03" />

<title>Discussing Genomic Phylostratigraphy</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>







<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Discussing Genomic Phylostratigraphy</h1>
<h4 class="date">2024-10-03</h4>



<div id="genomic-phylostratography-based-gene-age-inference" class="section level3">
<h3>Genomic phylostratography based gene age inference</h3>
<p>Discussed intensely in the past years (<a href="https://www.sciencedirect.com/science/article/pii/S016895251300111X">Capra
et al., 2013</a>; <a href="https://www.nature.com/articles/nmeth.3830.pdf">Altenhoff et al.,
2016</a>; <a href="https://academic.oup.com/gbe/article/8/6/1812/2574026">Liebeskind
et al., 2016</a>; <a href="https://academic.oup.com/mbe/article-lookup/doi/10.1093/molbev/msw284">Domazet-Loso
et al., 2017</a>; <a href="https://academic.oup.com/mbe/article/38/1/31/5900268">Futo et al.,
2021</a>; <a href="https://link.springer.com/article/10.1186/s13059-023-02895-z">Barrera-Redondo
et al., 2023</a>; etc.), gene age inference is not a trivial task and
might be biased in some currently existing approaches (<a href="https://academic.oup.com/gbe/article/8/6/1812/2574026">Liebeskind
et al., 2016</a>; <a href="https://academic.oup.com/bib/article-lookup/doi/10.1093/bib/bby074">Yin
et al., 2018</a>; <a href="https://academic.oup.com/gbe/article/10/11/2906/5140197">Casola
2018</a>; <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000862">Weisman
et al., 2020</a>; <a href="https://www.sciencedirect.com/science/article/pii/S0960982222007217">Weisman
et al., 2022</a>; etc.).</p>
<p>In particular, Moyers &amp; Zhang argue that <a href="https://www.sciencedirect.com/science/article/pii/S0168952507002995">genomic
phylostratigraphy</a> (a prominent BLAST based gene age inference
method)</p>
<ol style="list-style-type: decimal">
<li><p>underestimates gene age for a considerable fraction of
genes,</p></li>
<li><p>is biased for rapidly evolving proteins which are short, and/or
their most conserved block of sites is small, and</p></li>
<li><p>these biases create spurious nonuniform distributions of various
gene properties among age groups, many of which cannot be predicted a
priori (<a href="https://academic.oup.com/mbe/article/32/1/258/2925575">Moyers
&amp; Zhang, 2015</a>; <a href="https://academic.oup.com/mbe/article/33/5/1245/2579728">Moyers
&amp; Zhang, 2016</a>; <a href="https://academic.oup.com/gbe/article/8/6/1812/2574026">Liebeskind
et al., 2016</a>).</p></li>
</ol>
<p>However, these arguments were based on simulated data and were
inconclusive due to <a href="https://academic.oup.com/mbe/article/33/11/3031/2272516">errors in
their analyses</a>. Furthermore, <a href="https://academic.oup.com/mbe/article-lookup/doi/10.1093/molbev/msw284">Domazet-Loso
et al., 2017</a> provide convincing evidence that there is
<strong>no</strong> phylostratigraphic bias. As a response, <a href="https://academic.oup.com/gbe/article/9/6/1519/3867914">Moyers
&amp; Zhang, 2017</a> published a counter-study stating that a
phylostratigraphic trend claimed by <a href="https://academic.oup.com/mbe/article-lookup/doi/10.1093/molbev/msw284">Domazet-Loso
et al., 2017</a> to be robust to error disappears when genes likely to
be error-resistant are analyzed. <a href="https://academic.oup.com/gbe/article/9/6/1519/3867914">Moyers
&amp; Zhang, 2017</a> further suggest a more robust methodology for
controlling for the effects of error by first restricting to those genes
which can be simulated and then removing those genes which, through
simulation, have been shown to be error-prone (see also <a href="https://academic.oup.com/gbe/article/10/8/2037/5061555">Moyers
&amp; Zhang, 2018</a>).</p>
<p>In general, an objective benchmarking set representing the tree of
life is still missing and therefore any procedure aiming to quantify
gene ages will be biased to some degree. Based on this debate <a href="https://academic.oup.com/gbe/article/8/6/1812/2574026">Liebeskind
et al., 2016</a> suggest inferring gene age by combining several common
orthology inference algorithms to create gene age datasets and then
characterize the error around each age-call on a per-gene and
per-algorithm basis. Using this approach systematic error was found to
be a large factor in estimating gene age, suggesting that simple
consensus algorithms are not enough to give a reliable point estimate
(<a href="https://academic.oup.com/gbe/article/8/6/1812/2574026">Liebeskind
et al., 2016</a>). This was also observed by <a href="https://academic.oup.com/gbe/article/10/8/2037/5061555">Moyers
&amp; Zhang, 2018</a>) when running alternative tools such as
<code>PSIBLAST</code>, <code>HMMER</code>, <code>OMA</code>, etc.
However, by generating a consensus gene age and quantifying the possible
error in each workflow step, <a href="https://academic.oup.com/gbe/article/8/6/1812/2574026">Liebeskind
et al., 2016</a> provide a very useful <a href="https://github.com/marcottelab/Gene-Ages">database</a> of
consensus gene ages for a variety of genomes.</p>
<p>Alternatively, <a href="https://doi.org/10.1093/bib/bbw034">Smith
&amp; Pease, 2016</a> argues that <em>de novo</em> gene birth/death and
gene family expansion/contraction studies should avoid drawing direct
inferences of evolutionary relatedness from measures of sequence
similarity alone, and should instead, where possible, use more rigorous
phylogeny-based methods. For this purpose, I recommend researchers to
consult the <a href="https://orthology.phylomedb.org/">phylomedb
database</a> to retrieve phylogeny-based gene orthology relationships
and use these age estimates in combination with <code>myTAI</code>.
Similarly, users can also retrieve gene ages from <a href="http://eggnog5.embl.de/#/app/home">EggNOG database</a> using <a href="https://github.com/kullrich/orthomap"><code>orthomap</code></a>.
Alternatively, users might find the simulation based removal approach
proposed by <a href="https://academic.oup.com/gbe/article/10/8/2037/5061555">Moyers
&amp; Zhang, 2018</a> more suitable.</p>
<p>Evidently, these advancements in gene age research are very recent
and gene age inference is a very young and active field of genomic
research. Therefore, many more studies need to address the robust and
realistic inference of gene age and a community standard is still
missing.</p>
<p>Despite the ongoing debate about how to correctly infer gene age,
users of <code>myTAI</code> can perform any gene age inference method
they find most appropriate for their biological question and pass this
gene age inference table as input to <code>myTAI</code>. To do so, users
need to follow the following data format specifications to use their
gene age inference table with <code>myTAI</code>. However, even when
users rely on established procedures such as phylostratigraphy the gene
age inference bias will be present as ‘systematic error’ in all
developmental stages for which TAI or TDI computations are performed.
Thus, <em>stages of constraint will be detectable in any case.</em>
Since TAI or TDI computations are intended to enable screening for
conserved or constrained stages in developmental or biological processes
for further downstream experimental studies, even simple approaches such
as phylostratigraphy can give first evidence for the existence of
transcriptomic constraints within a biological process. If researchers
then wish to extract the exact candidate genes that might potentially
cause such transcriptome constraints then I would advise to rely on more
superior approaches of gene age inference as discussed here.</p>
<div id="overlooked-technical-aspects-of-blast-and-blast-like-tools" class="section level4">
<h4>Overlooked technical aspects of BLAST and BLAST-like tools</h4>
<p>In my opinion, what is completely missing in this entire debate is
the bioinformatics/technical aspect of using BLAST or any other
BLAST-like tool for gene age inference. Recently, it was intensely
discussed how BLAST hits are biased by the use of the default argument
<code>max_target_seqs</code> (<a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty833/5106166">Shah
et al., 2018</a>). The main issue of how this
<code>max_target_seqs</code> is set is that:</p>
<blockquote>
<p>According to the BLAST documentation itself (2008), this parameter
represents the ‘number of aligned sequences to keep’. This statement is
commonly interpreted as meaning that BLAST will return the top N
database hits for a sequence query if the value of max_target_seqs is
set to N. For example, in a recent article (Wang et al., 2016) the
authors explicitly state ‘Setting ’max target seqs’ as ‘1’ only the best
match result was considered’. <strong>To our surprise, we have recently
discovered that this intuition is incorrect.</strong> Instead, BLAST
returns the first N hits that exceed the specified E-value threshold,
which may or may not be the highest scoring N hits. The invocation using
the parameter ‘-max_target_seqs 1’ simply returns the first good hit
found in the database, not the best hit as one would assume. Worse yet,
the output produced depends on the order in which the sequences occur in
the database. For the same query, different results will be returned by
BLAST when using different versions of the database even if all versions
contain the same best hit for this database sequence. Even ordering the
database in a different way would cause BLAST to return a different ‘top
hit’ when setting the max_target_seqs parameter to 1. - Shah et al.,
2018</p>
</blockquote>
<p>The solution to this issue seems to be that any BLAST search must be
performed with a significantly high <code>-max_target_seqs</code>,
e.g. <code>-max_target_seqs 10000</code> (see <a href="https://gist.github.com/sujaikumar/504b3b7024eaf3a04ef5" class="uri">https://gist.github.com/sujaikumar/504b3b7024eaf3a04ef5</a>
for details) and best hits must be filtered subsequently. It is not
clear from any of the studies referenced above how the best BLAST hit
was retrieved and which <code>-max_target_seqs</code> values were used
to perform BLAST searches in the respective study. Thus, the
comparability of the results between studies is impossible and any
individual claim made in these studies might be biased.</p>
<p>In addition, the <code>-max_target_seqs</code> argument issue seems
not to be the only issue that might influence technical differences in
BLAST hit results. <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty1018/5239655">Gonzalez-Pech
et al., 2018</a> discuss another problem of retrieving the best BLAST
hits based on <code>E-value</code> thresholds.</p>
<blockquote>
<p>Many users assume that BLAST alignment hits with E-values less than
or equal to the predefined threshold (e.g. 10<sup>-5</sup> via the
specification of evalue 1e-5) are identified after the search is
completed, in a final step to rank all alignments by E-value, from the
smallest (on the top of the list of results) to the largest E-value (at
the bottom of the list). <strong>However, the E-value filtering step
does not occur at the final stage of BLAST; it occurs earlier during the
scanning phase (Altschul et al., 1997; Camacho et al., 2009). During
this phase, a gapped alignment is generated using less-sensitive
heuristic parameters (Camacho et al., 2009); alignments with an E-value
that satisfies the defined cut-off are included in the subsequent phase
of the BLAST algorithm (and eventually reported). During the final
(trace-back) phase, these gapped alignments are further adjusted using
moresensitive heuristic parameters (Camacho et al., 2009), and the
E-value for each of these refined alignments is then
recalculated.</strong> - Gonzalez-Pech et al., 2018</p>
</blockquote>
<p>This means that if one study mentioned above ran a BLAST search with
a BLAST parameter configuration of lets say
<code>-max_target_seqs 250</code> (default value in BLAST) and
<code>evalue 10</code> (default value in BLAST) and then subsequently
selected the best hit which returned the smallest <code>E-value</code>
and another study used the parameter configuration
<code>-max_target_seqs 1</code> and <code>evalue 0.0001</code> then the
results of both studies would not be comparable and the proposed gene
age inference bias might simply result from a technical difference in
running BLAST searches.</p>
<p>In more detail, even if one study for example ran BLAST with
<code>evalue 10</code> (default value in BLAST) and then subsequently
filtered for hits that resulted in <code>evalue &lt; 0.0001</code>
whereas another study ran BLAST directly with
<code>evalue 0.0001</code>, according to <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty1018/5239655">Gonzalez-Pech
et al., 2018</a> these studies although referring to the same
<code>E-value</code> threshold for filtering hits will result in
different sets of filtered BLAST hits.</p>
</div>
<div id="homology-detection-failure" class="section level4">
<h4>Homology detection failure</h4>
<p>In addition, small and fast-evolving genes are often wrongly
annotated as young genes due to <code>homology detection failure</code>.
This issue and potential solution in the context of genomic
phylostratigraphy have been discussed by <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000862">Weisman
et al., 2020</a>.</p>
<blockquote>
<p>An alternative explanation for a lineage-specific gene is that
nothing particularly special has happened in the gene’s evolutionary
history. The gene has homologs outside of the lineage (no de novo
origination), and no novel function has emerged (no
neofunctionalization), but despite this lack of novelty, computational
similarity searches (e.g., BLAST) have failed to detect the
out-of-lineage homologs. We refer to such unsuccessful searches as
<em>homology detection failure</em>. As homologs diverge in sequence
from one another, the statistical significance of their similarity
declines. <em>Over evolutionary time, with a constant rate of sequence
evolution, the degree of similarity may fall below the chosen
significance threshold, resulting in a failure to detect the
homolog</em>. - Weisman et al., 2020</p>
</blockquote>
<p>Users can run their test, <a href="https://github.com/caraweisman/abSENSE"><code>abSENSE</code></a>
to detect the <em>probability that a homolog of a given gene would fail
to be detected by a homology search</em>. We note that a recently
developed tool <a href="https://github.com/josuebarrera/GenEra"><code>GenEra</code></a>
can automate <code>abSENSE</code> analysis for every gene in a given
genome and provide an output compatible with <code>myTAI</code>.</p>
</div>
<div id="synteny-based-phylostratigraphy" class="section level4">
<h4>Synteny-based phylostratigraphy</h4>
<p>A recently introduced approach is called
<code>synteny-based phylostratigraphy</code> (<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3023-y">Arendsee
et al., 2019</a>). Here, the authors provide a comparative analysis of
genes across evolutionary clades, augmenting standard phylostratigraphy
with a detailed, synteny-based analysis. Whereas standard
phylostratigraphy searches the proteomes of related species for
similarities to focal genes, their <code>fagin</code> pipeline first
finds syntenic genomic intervals and then searches within these
intervals for any trace of similarity. It searches the
(<code>in silico</code> translated) amino acid sequence of all
unannotated ORFs as well as all known CDS within the syntenic search
space of the target genomes. If no amino acid similarity is found within
the syntenic search space, their <code>fagin</code> pipeline will search
for nucleotide similarity. Finding nucleotide sequence similarity, but
not amino acid similarity, is consistent with a <code>de novo</code>
origin of the focal gene. If no similarity of any sort is found, their
<code>fagin</code> pipeline will use the syntenic data to infer a
possible reason. Thus, they detect indels, scrambled synteny, assembly
issues, and regions of uncertain synteny (<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3023-y">Arendsee
et al., 2019</a>).</p>
</div>
<div id="accelerated-and-sensitive-sequence-alignment" class="section level4">
<h4>Accelerated and sensitive sequence alignment</h4>
<p>While BLASTP is considered the gold-standard tool for sequence
alignment in many labs, newer approaches (such as <a href="https://github.com/bbuchfink/diamond">DIAMOND v2</a> and <a href="https://github.com/soedinglab/mmseqs2/">MMSeqs2</a>) have been and
are being developed. They retain the sensitivity of BLAST while being
thousands to hundred-folds faster. This speed-up is crucial as sequence
databases increase in size. This is important for genomic
phylostratigraphy and <a href="https://github.com/bbuchfink/diamond">DIAMOND v2</a> and <a href="https://github.com/soedinglab/mmseqs2/">MMSeqs2</a> have been
employed in various phylostratigraphic and phylotranscriptomic analyses
(either explicitly or under-the-hood in tools such as <a href="https://github.com/davidemms/OrthoFinder">OrthoFinder</a>, which
uses DIAMOND by default) (see <a href="https://www.biorxiv.org/content/10.1101/2022.08.21.504376v1">Domazet-Loso
et al., 2022</a>; <a href="https://www.pnas.org/doi/full/10.1073/pnas.2216351120">Ma &amp;
Zheng 2023</a>; <a href="https://academic.oup.com/g3journal/article/13/6/jkad077/7097621">Manley
et al., 2023</a>).</p>
<p>With <code>DIAMOND</code>, users can overcome the overlooked
technical aspects of BLAST such as the <code>max_target_seqs</code>
issue by setting <code>--max-target-seqs</code> to <code>0</code> (or
simply <code>-k0</code>). Furthermore, the
<code>--ultra-sensitive</code> mode can be employed, which is as
sensitive as BLAST while being 80x faster (see <a href="https://www.nature.com/articles/s41592-021-01101-x/figures/1">Fig
1. in the DIAMOND paper</a>).</p>
<p><a href="https://github.com/josuebarrera/GenEra">GenEra</a>, a
recently developed gene age inference tool, utilises <a href="https://github.com/bbuchfink/diamond">DIAMOND v2</a> with
appropriate parameter settings by default. Users can also specify the
use of <a href="https://github.com/soedinglab/mmseqs2/">MMSeqs2</a> as
well as the <em>protein structure aligner</em> <a href="https://github.com/steineggerlab/foldseek">Foldseek</a> and other
tools (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02895-z">Barrera-Redondo
et al., 2023</a>). Potential issues such as <em>homology detection
failure</em> can be overcome by providing a pair-wise evolutionary
distance file as an input. Furthermore, users can set thresholds such as
for <code>taxonomic representativeness</code> to filter evolutionarily
inconsistent matches to the database, i.e. matches stemming from
database contamination or horizontal gene transfer.</p>
</div>
<div id="inferring-gene-age-using-orthogroups" class="section level4">
<h4>Inferring gene age using orthogroups</h4>
<p>The first studies in gene age inference used the framework of genomic
phylostratigraphy, where each gene of the focal species is compared
against a sequence database to find the most distantly related homologue
(<a href="https://www.sciencedirect.com/science/article/pii/S0168952507002995">Domazet-Loso
et al., 2007</a>). Some more recent studies have instead inferred gene
age using the node in the phylogeny that contains all members of an
<code>orthogroup</code>.</p>
<p>An orthogroup is a <em>group of genes descended from a single gene in
the last common ancestor of a clade of species</em>. This framework has
been very useful for phylogenetic inference and comparative
transcriptomics. Since an orthogroup is a subset of all sequence
homologues in the given dataset, a gene can hit genes outside of its
orthogroup (using the same parameters on a pairwise sequence aligner).
<a href="https://github.com/davidemms/OrthoFinder">OrthoFinder</a> (a
tool for phylogenetic orthology inference for comparative genomics) is
commonly used in <em>orthogroup</em>-based gene age inference. The
result of OrthoFinder includes the files <code>Orthogroups.tsv</code>
(now <a href="https://github.com/davidemms/OrthoFinder#orthogroups-directory-deprecated">depreciated</a>)
and <code>Phylogenetic_Hierarchical_Orthogroups/N0.tsv</code>, which
have been used for <em>orthogroup</em>-based TAI studies (e.g. <a href="https://www.pnas.org/doi/full/10.1073/pnas.2216351120">Ma &amp;
Zheng 2023</a>; <a href="https://www.biorxiv.org/content/10.1101/2023.07.04.546637v1.full">Nishimiya-Fujisawa
et al., 2023</a>). Incorporating these results into <code>myTAI</code>
can be done using <a href="https://github.com/kullrich/orthomap"><code>orthomap</code></a>
(see <a href="https://orthomap.readthedocs.io/en/latest/">documentation</a>).</p>
<p>Methodologically, it should be stressed that while
<code>Orthogroups.tsv</code> and
<code>Phylogenetic_Hierarchical_Orthogroups/N0.tsv</code> contain
orthogroups with more than one member (i.e. orthogroups with at least
one orthologue or paralogue), users should be aware that these output
files of OrthoFinder do not include singletons, which is a signature of
a novel or species-specific gene. To incorporate singletons, users
should include the genes in <code>Orthogroups_UnassignedGenes.tsv</code>
output or, alternatively, singletons in the <code>Orthogroups.txt</code>
file, and assign the singletons to the species-specific phylostrata in
the study.</p>
<p>The <em>orthogroup</em>-based approach seems to be motivated by an
assumption that the more restricted set of genes that consists of
orthologues and in-paralogues (paralogues that arose after the species
split) tends to have more similar attributes (e.g. conservation in
function and/or expression pattern) than the larger set of genes that
contains all sequence homologues. It follows that this framework should
better reflect the ‘gene age’. This assumption draws parallels to the
<code>orthologue conjecture</code> (i.e. orthologues tend to have more
conserved attributes than paralogues), which is heavily debated (<a href="https://www.nature.com/articles/nrg3456#Sec8">Gabaldon &amp;
Koonin 2013</a>; <a href="https://www.pnas.org/doi/abs/10.1073/pnas.1707515115">Dunn et al.,
2018</a>; <a href="https://pubmed.ncbi.nlm.nih.gov/32657391/">Stamboulian et al.,
2020</a>; etc.). In several cases, orthologues do not have more
conserved attributes than paralogues. Indeed, ‘shifts’ or
‘substitutions’ in the expression profile of paralogues that arose
before the species split are observed in single-cell studies (<a href="https://elifesciences.org/articles/66747">Tarashansky et al.,
2021</a>; <a href="https://www.nature.com/articles/s41559-021-01580-3">Shafer et al.,
2022</a>). Overall, it remains unclear whether inferring gene age using
the <em>orthogroup</em>-based approach better captures evolutionary
information than the classic genomic phylostratigraphy approach
(all/unrestricted sequence homologues), given that several issues
pertaining to genomic phylostratigraphy have been addressed (see <a href="https://link.springer.com/article/10.1186/s13059-023-02895-z">Barrera-Redondo
et al., 2023</a>).</p>
</div>
</div>
<div id="summary" class="section level3">
<h3>Summary</h3>
<p>Hence, all of the above mentioned approaches are far from being
perfect and much more research is needed to systematically compare
different approaches for gene age inference.</p>
<p>The overall rational behind gene age inference is to assign each
protein coding gene of an organism of interest with an evolutionary age
estimate which aims to quantify its potential origin within the tree of
life (detectable sequence homologue; orphan gene - see <a href="https://www.nature.com/articles/nrg3053">Tautz &amp; Domazet-Loso,
2011</a>). Hence, gene age inference generates a table storing the gene
age in the first column and the corresponding gene id of the organism of
interest in the second column. This table is named
<em>phylostratigraphic map</em> (or <em>phylomap</em>).</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
