<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with myTAI • myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with myTAI">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/Introduction.html">Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/Intermediate.html">Intermediate</a></li>
    <li><a class="dropdown-item" href="../articles/Advanced.html">Advanced</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Discussion</h6></li>
    <li><a class="dropdown-item" href="../articles/Phylostratigraphy.html">Genomic Phylostratigraphy</a></li>
    <li><a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Taxonomy.html">Taxonomic Retrieval</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting started with myTAI</h1>
            
            <h4 data-toc-skip class="date">2025-02-07</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/vignettes/myTAI.Rmd" class="external-link"><code>vignettes/myTAI.Rmd</code></a></small>
      <div class="d-none name"><code>myTAI.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="scientific-background">Scientific background<a class="anchor" aria-label="anchor" href="#scientific-background"></a>
</h3>
<p>Today, phenomena such as morphological mutations, diseases or
developmental processes are primarily investigated on the molecular
level using transcriptomics approaches. Transcriptomes denote the total
number of quantifiable transcripts present at a specific stage in a
biological process. In disease or developmental (defect) studies,
transcriptomes are usually measured over several time points. In
treatment studies aiming to quantify differences in the transcriptome
due to biotic stimuli, abiotic stimuli, or diseases usually treatment /
disease versus non-treatment / non-disease transcriptomes are compared.
In either case, comparing changes in transcriptomes over time or between
treatments allows us to identify genes and gene regulatory mechanisms
that might be involved in governing the biological process of
investigation. Although classic transcriptomics studies are based on an
established methodology, little is known about the evolution and
conservation mechanisms underlying such transcriptomes. Understanding
the evolutionary mechanism that change transcriptomes over time,
however, might give us a new perspective on how diseases emerge in the
first place or how morphological changes are triggered by changes of
developmental transcriptomes.</p>
<p>Evolutionary transcriptomics aims to capture and quantify the
evolutionary conservation of genes that contribute to the transcriptome
during a specific stage of the biological process of interest. The
resulting temporal conservation pattern then enables to detect stages of
development or other biological processes that are evolutionarily
conserved (<a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btx835/4772684" class="external-link">Drost
et al., 2018</a>). This quantification on the highest level is achieved
through transcriptome indices (e.g. <a href="https://drostlab.github.io/myTAI/articles/Introduction.html#transcriptome-age-index">Transcriptome
Age Index or Transcriptome Divergence Index</a>) which aim to quantify
the average evolutionary age <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02895-z" class="external-link">Barrera-Redondo
et al., 2023</a> or sequence conservation <a href="https://academic.oup.com/mbe/article/32/5/1221/1125964" class="external-link">Drost et
al., 2015</a> of genes that contribute to the transcriptome at a
particular stage. In general, evolutionary transcriptomics can be used
as a method to quantify the evolutionary conservation of transcriptomes
at particular developmental stages and to investigate how transcriptomes
underlying biological processes are constrained or channeled due to
events in evolutionary history (Dollo’s law) (<a href="https://www.sciencedirect.com/science/article/pii/S0959437X16302040" class="external-link">Drost
et al., 2017</a>).</p>
<p><strong>Please note, since myTAI relies on gene age inference and
there has been an extensive debate about the best approaches for gene
age inference in the last years, please follow my <a href="https://drostlab.github.io/myTAI/articles/Phylostratigraphy.html">updated
discussion about the gene age inference literature</a>. With <a href="https://github.com/josuebarrera/GenEra" class="external-link">GenEra</a>, we addressed
all previously raised issues and we encourage users to run
<code>GenEra</code> when aiming to infer gene ages for further
<code>myTAI</code> analyses.</strong></p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Users can install <code>myTAI</code> from <a href="https://CRAN.R-project.org/package=myTAI" class="external-link">CRAN</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install myTAI 0.9.3 from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"myTAI"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install the developer version containing the newest features</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"drostlab/myTAI"</span><span class="op">)</span></span></code></pre></div>
<p><code>myTAI</code> is using <code>OpenMP</code> to run cpp code in
parallel, thus you might need to install additional software.</p>
<p>Ensure that the dependencies requirements are met:</p>
<details><summary><strong>Mac</strong>
</summary><div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">brew</span> install llvm libomp</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="bu">cd</span> /usr/local/lib</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /usr/local/opt/libomp/lib/libomp.dylib ./libomp.dylib</span></code></pre></div>
</details><details><summary><strong>Linux</strong>
</summary><ol style="list-style-type: decimal">
<li>
<p>Check if you have an OpenMP-enabled compiler: Open a terminal and
run the command <code>gcc --version</code> or <code>g++ --version</code>
to check if you have the GNU Compiler Collection (GCC) installed. OpenMP
support is typically included in GCC.</p>
<p>If you don’t have GCC installed, you can install it using your
distribution’s package manager. For example, on Ubuntu, you can use sudo
apt-get install build-essential to install GCC.</p>
</li>
<li>
<p>Install the libomp library: Open a terminal and run the command
appropriate for your package manager:</p>
<p>On Ubuntu or Debian-based systems:
<code>sudo apt-get install libomp-dev</code> or
<code>sudo apt install libomp-dev</code></p>
<p>On Fedora or CentOS systems:
<code>sudo dnf install libomp-devel</code></p>
<p>On Arch Linux: <code>sudo pacman -S libomp</code></p>
</li>
</ol></details><details><summary><strong>Windows</strong>
</summary><p>On Windows, the most common OpenMP implementation is provided by
Microsoft Visual C++ (MSVC) compiler, which includes OpenMP support by
default. So, if you are using MSVC as your C++ compiler, you should have
OpenMP support without any additional installations.</p>
</details>
</div>
<div class="section level3">
<h3 id="running-mytai">Running myTAI<a class="anchor" aria-label="anchor" href="#running-mytai"></a>
</h3>
<div class="section level4">
<h4 id="load-example-data">Load example data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drostlab/myTAI" class="external-link">myTAI</a></span><span class="op">)</span></span>
<span><span class="co"># example dataset covering 7 stages of A thaliana embryo development</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PhyloExpressionSetExample"</span><span class="op">)</span></span>
<span><span class="co"># transform absolute expression levels to log2 expression levels</span></span>
<span><span class="va">ExprExample</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tf.html">tf</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="quantify-transcriptome-conservation-using-tai">Quantify transcriptome conservation using TAI<a class="anchor" aria-label="anchor" href="#quantify-transcriptome-conservation-using-tai"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># visualize global Transcriptome Age Index pattern</span></span>
<span><span class="fu"><a href="../reference/PlotSignature.html">PlotSignature</a></span><span class="op">(</span><span class="va">ExprExample</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="reference/figures/unnamed-chunk-10-1.png" alt="plotSignature"><div class="figcaption">plotSignature</div>
</div>
</div>
<div class="section level4">
<h4 id="quantify-expression-level-distributions-for-each-gene-age-category">Quantify expression level distributions for each gene age
category<a class="anchor" aria-label="anchor" href="#quantify-expression-level-distributions-for-each-gene-age-category"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot expression level distributions for each age (=PS) category </span></span>
<span><span class="co"># and each developmental stage </span></span>
<span><span class="fu"><a href="../reference/PlotCategoryExpr.html">PlotCategoryExpr</a></span><span class="op">(</span><span class="va">ExprExample</span>, <span class="st">"PS"</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="reference/figures/unnamed-chunk-11-1.png" alt="plotCategoryExpr"><div class="figcaption">plotCategoryExpr</div>
</div>
</div>
<div class="section level4">
<h4 id="quantify-mean-expression-of-individual-gene-age-categories">Quantify mean expression of individual gene age categories<a class="anchor" aria-label="anchor" href="#quantify-mean-expression-of-individual-gene-age-categories"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot mean expression of each age category seperated by old (PS1-3)</span></span>
<span><span class="co"># versus young (PS4-12) genes</span></span>
<span><span class="fu"><a href="../reference/PlotMeans.html">PlotMeans</a></span><span class="op">(</span><span class="va">ExprExample</span>, Groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="reference/figures/unnamed-chunk-12-1.png" alt="plotMeans"><div class="figcaption">plotMeans</div>
</div>
</div>
<div class="section level4">
<h4 id="quantify-relative-mean-expression-of-each-age-category-seperated-by-old-versus-young-genes">Quantify relative mean expression of each age category seperated by
old versus young genes<a class="anchor" aria-label="anchor" href="#quantify-relative-mean-expression-of-each-age-category-seperated-by-old-versus-young-genes"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot relative mean expression of each age category seperated by old (PS1-3)</span></span>
<span><span class="co"># versus young (PS4-12) genes</span></span>
<span><span class="fu"><a href="../reference/PlotRE.html">PlotRE</a></span><span class="op">(</span><span class="va">ExprExample</span>, Groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="reference/figures/unnamed-chunk-13-1.png" alt="plotRE"><div class="figcaption">plotRE</div>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the significant differences between gene expression distributions </span></span>
<span><span class="co"># of old (=group1) versus young (=group2) genes</span></span>
<span><span class="fu"><a href="../reference/PlotGroupDiffs.html">PlotGroupDiffs</a></span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">ExprExample</span>,</span>
<span>               Groups        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>group_1 <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, group_2 <span class="op">=</span> <span class="fl">4</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>               legendName    <span class="op">=</span> <span class="st">"PS"</span>,</span>
<span>               plot.type     <span class="op">=</span> <span class="st">"boxplot"</span><span class="op">)</span></span></code></pre></div>
<p>If you successfully ran these myTAI functions, congrats! Please check
out the more detailed tutorials and discussions under the
<code>Articles</code> tab. For the list of functions, click on the
<code>Reference</code> tab.</p>
<p>Hope you enjoy your (non-alcoholic) <code>myTAI</code>!</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich, Stefan Manolache.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
