<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Other evolutionary and expression indices • myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Other evolutionary and expression indices">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.2.0.9006</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-stats.html">Statistical testing with myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-transform.html">Transforming dataset for myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/index.html">... and view more!</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Other evolutionary and expression indices</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/vignettes/other-strata.Rmd" class="external-link"><code>vignettes/other-strata.Rmd</code></a></small>
      <div class="d-none name"><code>other-strata.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="motivation-for-non-tai-approaches">Motivation for non-TAI approaches<a class="anchor" aria-label="anchor" href="#motivation-for-non-tai-approaches"></a>
</h2>
<p>So far, most of the vignettes have focused on workflows and
discussions centred around the transcriptome age index (TAI). However,
we can also assign other information for each gene (analogous to gene
age in TAI). Examples include <a href="https://www.nature.com/articles/nature11394" class="external-link"><code>TDI</code></a>,
<a href="https://www.nature.com/articles/s41586-024-08059-8" class="external-link"><code>TSI</code></a>,
<a href="https://academic.oup.com/mbe/article/33/7/1669/2578492" class="external-link"><code>TPI</code></a>
and so on (though the names of these indices can differ between studies
and are thus not standardised).</p>
<p>In the sections below, we describe the workflow for the transcriptome
divergence index (TDI) and transcriptome specificity index (TSI). These
and other index-based methods use the same framework as TAI, aside from
the method to give a factor or numeric value to each gene (e.g. gene age
for TAI).</p>
<div class="note">
<p>The good news is that <code>myTAIv2</code> is rather flexible; both
<code>BulkPhyloExpressionSet</code> and
<code>ScPhyloExpressionSet</code> objects can accommodate other indices,
along with most of the <code>myTAIv2</code> functions.</p>
<p>Users can simply follow the workflow outlined in <a href="tai-stats.html">statistical tests</a>, <a href="tai-transform.html">transformations</a> along with most of the <a href="tai-gallery.html">plotting functionalities</a>, using non-TAI
data.</p>
</div>
</div>
<div class="section level2">
<h2 id="transcriptome-divergence-index-tdi">Transcriptome divergence index (TDI)<a class="anchor" aria-label="anchor" href="#transcriptome-divergence-index-tdi"></a>
</h2>
<p>The <code>TDI</code> (<a href="https://www.nature.com/articles/nature11394" class="external-link">Quint et al.,
2012</a>) uses deciled dNdS values for each gene and summarises this
information across all genes at a given stage (analogous to TAI) to
quantify the overall transcriptome ‘divergence’.</p>
<div class="note">
<p>For the vast majority of genes, the dNdS values are below 1
(i.e. purifying selection). Therefore, while TDI is called transcriptome
<em>divergence</em> index, TDI can be thought to capture the degree of
purifying selection (<a href="https://www.nature.com/articles/s41586-024-08059-8" class="external-link">Lotharukpong
et al. 2024</a>).</p>
</div>
<p>R packages such as <a href="https://github.com/drostlab/orthologr" class="external-link"><code>orthologr</code></a>
can facilitate dNdS estimation and <a href="https://drostlab.github.io/orthologr/articles/divergence_stratigraphy.html" class="external-link">divergence-stratigraphy</a>
computation used in the TDI approach.</p>
<details><summary>
Example TDI workflow using mock data
</summary><p>Assuming that we have the <a href="https://drostlab.github.io/orthologr/articles/divergence_stratigraphy.html" class="external-link">divergence
strata</a>, similar to <code>TAI</code>, <code>TDI</code> is calculated
as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mrow><mi>T</mi><mi>D</mi><mi>I</mi></mrow><mi>s</mi></msub><mo>=</mo><mfrac><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msub><mrow><mi>d</mi><mi>s</mi></mrow><mi>i</mi></msub><mo>⋅</mo><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">
{TDI}_s = \frac{\sum_{i=1}^{n} {ds}_i \cdot e_{is}}{\sum_{i=1}^{n} e_{is}}
</annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub><annotation encoding="application/x-tex">e_{is}</annotation></semantics></math>
is the expression level of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
at a given sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>
(e.g. a biological replicate for a developmental stage), and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>d</mi><mi>s</mi></mrow><mi>i</mi></msub><annotation encoding="application/x-tex">{ds}_i</annotation></semantics></math>
is the evolutionary age of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>.</p>
<p>Like TAI, for developmental time-course data,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>D</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TDI}_s</annotation></semantics></math>
can be grouped for each stage (if replicate data exists) and compared
across developmental stages to obtain the overall TDI profile.<br>
For pairwise comparisons (see <a href="tai-stats.html">📈</a>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>D</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TDI}_s</annotation></semantics></math>
can be grouped as one of the conditions to be compared.<br>
For single cell data, (see <a href="phylo-expression-object.html">📊</a>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>D</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TDI}_s</annotation></semantics></math>
is the TDI for a given cell or cell type (if pseudo-bulked).</p>
<div class="section level3">
<h3 id="mock-bulk-rna-seq-dataset-for-tdi">Mock bulk RNA-seq dataset for TDI<a class="anchor" aria-label="anchor" href="#mock-bulk-rna-seq-dataset-for-tdi"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drostlab.github.io/myTAI/">myTAI</a></span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rconsortium.github.io/S7/" class="external-link">S7</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_phyex_set_old"</span><span class="op">)</span></span></code></pre></div>
<p>Now we create the mock dataset, where we assign a value between 1 and
10 for all genes. This is the same distribution of values as the
<code>divergence strata</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># this is a mock divergence strata dataset</span></span>
<span><span class="co"># we assign a value between 1 and 10 for all genes</span></span>
<span><span class="va">gene_names</span> <span class="op">&lt;-</span> <span class="va">example_phyex_set_old</span><span class="op">@</span><span class="va">gene_ids</span></span>
<span><span class="va">example_divergence_strata</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gene_names</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="va">gene_names</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># mock expression data taken from example_phyex_set_old@.expression</span></span>
<span><span class="va">example_expression</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">example_phyex_set_old</span><span class="op">@</span><span class="va">.expression</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span></code></pre></div>
<p>Now we construct the input data.frame for
<code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set_tdi.df</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>phylorank <span class="op">=</span> <span class="va">example_divergence_strata</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">phylorank</span>, <span class="va">GeneID</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">example_expression</span>, by <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span></code></pre></div>
<p>Create the <code>BulkPhyloExpressionSet</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set_tdi</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet</a></span><span class="op">(</span></span>
<span>        <span class="va">example_phyex_set_tdi.df</span>,</span>
<span>        name <span class="op">=</span> <span class="st">"Example TDI data"</span>,</span>
<span>        index_type <span class="op">=</span> <span class="st">"TDI"</span><span class="op">)</span></span></code></pre></div>
<p>plot away!</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set_tdi</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [=======================                 ] 56%  (~0s remaining)       Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="other-strata_files/figure-html/unnamed-chunk-6-1.png" alt="plot signature function output for mock TDI data" width="576" style="display: block; margin: auto;"></p>
<p>Looks interesting right?</p>
<p>… although as you can expect from the
<code><a href="../reference/stat_flatline_test.html">myTAI::stat_flatline_test()</a></code>, because we assigned random
values, the overall pattern doesn’t deviate from a flat line.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_flatline_test.html">stat_flatline_test</a></span><span class="op">(</span><span class="va">example_phyex_set_tdi</span><span class="op">)</span></span></code></pre></div>
<p><img src="other-strata_files/figure-html/unnamed-chunk-7-1.png" alt="stat_flatline_test function output" width="480" style="display: block; margin: auto;"></p>

<div class="note">
<p>The dNdS values are not without issues. For example, it is important
to choose an appropriate evolutionary distance to pairwise align and
compare nucleotide sequences. Too close results in no meaningful
substitution and too distant results in saturation. To overcome this,
similar measures to the dNdS can be computed from MSAs.</p>
</div>
</div>
</details>
</div>
<div class="section level2">
<h2 id="transcriptome-specificity-index-tsi">Transcriptome specificity index (TSI)<a class="anchor" aria-label="anchor" href="#transcriptome-specificity-index-tsi"></a>
</h2>
<p>The <code>TSI</code> (<a href="https://www.nature.com/articles/s41586-024-08059-8" class="external-link">Lotharukpong
et al., 2024</a>) uses expression breadth/specificity information that
can be obtained for each gene and summarises this information across all
genes at a given stage (analogous to TAI) to quantify the overall
transcriptome ‘specificity’.</p>
<p>Similar to <code>TAI</code>, <code>TSI</code> is calculated as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mrow><mi>T</mi><mi>S</mi><mi>I</mi></mrow><mi>s</mi></msub><mo>=</mo><mfrac><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msub><mrow><mi>t</mi><mi>s</mi></mrow><mi>i</mi></msub><mo>⋅</mo><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">
{TSI}_s = \frac{\sum_{i=1}^{n} {ts}_i \cdot e_{is}}{\sum_{i=1}^{n} e_{is}}
</annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub><annotation encoding="application/x-tex">e_{is}</annotation></semantics></math>
is the expression level of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
at a given sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>
(e.g. a biological replicate for a developmental stage), and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>t</mi><mi>s</mi></mrow><mi>i</mi></msub><annotation encoding="application/x-tex">{ts}_i</annotation></semantics></math>
is the evolutionary age of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>.</p>
<p>Like TAI, for developmental time-course data,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>S</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TSI}_s</annotation></semantics></math>
can be grouped for each stage (if replicate data exists) and compared
across developmental stages to obtain the overall TSI profile.<br>
For pairwise comparisons (see <a href="tai-stats.html">📈</a>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>S</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TSI}_s</annotation></semantics></math>
can be grouped as one of the conditions to be compared.<br>
For single cell data, (see <a href="phylo-expression-object.html">📊</a>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>S</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TSI}_s</annotation></semantics></math>
is the TSI for a given cell or cell type (if pseudo-bulked).</p>
<p>Thus, TSI represents the expression-weighted mean gene expression
specificity.</p>
<div class="tip">
<p>You can re-use the same workflow as for TDI that we have shown above.
The only differences are that (1) you need to give each gene the
expression specificity score and not divergence strata, and (2) you will
be using real data not the mock dataset!</p>
</div>
<p>For TSI, accurate inference of expression specificity is key.</p>
<p>According to <a href="https://academic.oup.com/bib/article/18/2/205/2562739" class="external-link">Kryuchkova-Mostacci
&amp; Robinson-Rechavi, 2017</a>:</p>
<blockquote>
<p>Tau appears consistently to be the most robust method in our
analyses.</p>
</blockquote>
<p>Thus, we recommend the <em>tau</em> (or ) score as introduced by <a href="https://academic.oup.com/bioinformatics/article/21/5/650/220059" class="external-link">Yanai
et al. 2005</a> as a measure of expression specificity, though other
measures exist (e.g. those based on information entropy). You can also
decile the <em>tau</em> values (similar to deciled dNdS for
divergence-strata) to enable analogous comparisons to TAI.</p>
<div class="important">
<p>Regardless of the measures chosen for expression specificity/breadth,
the estimates improve with data quality (i.e. number of RNA-seq samples)
and quantity (i.e. diversity of the sampling such as multiple tissues
and stages, and low or no technical noise).</p>
</div>
</div>
<div class="section level2">
<h2 id="transcriptome-insert-your-idea-index">Transcriptome {insert your idea} index<a class="anchor" aria-label="anchor" href="#transcriptome-insert-your-idea-index"></a>
</h2>
<p>Using two examples (TDI and TSI), we went through different (non-TAI)
evolutionary and expression indices that can be used for evolutionary
transcriptomics studies. As long as the computation follows the form
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mrow><mi>T</mi><mi>x</mi><mi>I</mi></mrow><mi>s</mi></msub><mo>=</mo><mfrac><mrow><msubsup><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></msubsup><msub><mi>x</mi><mi>i</mi></msub><mo>⋅</mo><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></msubsup><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">{TxI}_s = \frac{\sum_{i=1}^{n} {x}_i \cdot e_{is}}{\sum_{i=1}^{n} e_{is}}</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>
is a value (factor or numeric) assigned to each gene, you can even
invent new ‘indices’.</p>
<p>Looking forward to seeing what can be done!</p>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>You might wonder - why so many transcriptome indices? TAI, TDI, TSI,
TPI and other such indices provide <strong>complementary</strong> views
on the transcriptome evolution patterns. For more details on their
biological interpretation, please check these studies for <a href="https://www.nature.com/articles/nature09632" class="external-link"><code>TAI</code></a>,
<a href="https://www.nature.com/articles/nature11394" class="external-link"><code>TDI</code></a>,
<a href="https://www.nature.com/articles/s41586-024-08059-8" class="external-link"><code>TSI</code></a>,
<a href="https://academic.oup.com/mbe/article/33/7/1669/2578492" class="external-link"><code>TPI</code></a>,
etc.</p>
<div class="tip">
<p>If you are using different indices, it is important to check that the
gene-level values assigned (e.g. age, divergence-stratum) are not
strongly correlated. High correlation means that the resulting indices
do <strong>not provide complementary</strong> information (i.e. not
reflect different biological aspects).</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Ștefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
