<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Bring your datasets into myTAI â€¢ myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="â€image/svg+xmlâ€" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Bring your datasets into myTAI">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.2.0.9006</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-stats.html">Statistical testing with myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-transform.html">Transforming dataset for myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/index.html">... and view more!</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Bring your datasets into myTAI</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/vignettes/phylo-expression-object.Rmd" class="external-link"><code>vignettes/phylo-expression-object.Rmd</code></a></small>
      <div class="d-none name"><code>phylo-expression-object.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="s7-bulkphyloexpressionset-and-scphyloexpressionset-objects">S7 BulkPhyloExpressionSet and ScPhyloExpressionSet objects<a class="anchor" aria-label="anchor" href="#s7-bulkphyloexpressionset-and-scphyloexpressionset-objects"></a>
</h2>
<p>Starting from <code>myTAIv2</code>, we moved on from the traditional
<code>data.frame</code> to the new <a href="https://rconsortium.github.io/S7/" class="external-link"><code>S7 OOP system</code></a>
to facilitate fast and flexible computation for evolutionary
transcriptomics analyses.</p>
<p>To illustrate this, we will use the example data
<code>example_phyex_set</code>. The <code>BulkPhyloExpressionSet</code>
and <code>ScPhyloExpressionSet</code> objects are designed to store gene
expression data along with associated metadata such as gene age
information (or <a href="phylostratigraphy.html"><code>phylorank</code></a>), experiment
type and other relevant properties.</p>
<p><img src="Figures/myTAI_phyex.png" alt="PhyEx overview" align="center" style="max-height: 500px; max-width: 800px;"></p>
<p>We will first focus on the <code>BulkPhyloExpressionSet</code>
object, which is for bulk RNA-seq data.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drostlab.github.io/myTAI/">myTAI</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_phyex_set"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># inspect the object</span></span>
<span><span class="va">example_phyex_set</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">## PhyloExpressionSet object</span></span>
<span><span class="co">## Class: myTAI::BulkPhyloExpressionSet </span></span>
<span><span class="co">## Name: Embryogenesis 2019 </span></span>
<span><span class="co">## Species: Arabidopsis thaliana </span></span>
<span><span class="co">## Index type: TAI </span></span>
<span><span class="co">## Stages : Preglobular, Globular, Early Heart, Late Heart, Early Torpedo, Late Torpedo, Bent Cotyledon, Mature Green </span></span>
<span><span class="co">## Number of genes: 27520 </span></span>
<span><span class="co">## Number of stages : 8 </span></span>
<span><span class="co">## Number of phylostrata: 19 </span></span>
<span><span class="co">## Number of samples: 24 </span></span>
<span><span class="co">## Samples per condition: 3 3 3 3 3 3 3 3</span></span></code></pre>
</details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">S7</span><span class="fu">::</span><span class="fu"><a href="https://rconsortium.github.io/S7/reference/prop_names.html" class="external-link">prop_names</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">##  [1] "strata"                             "strata_values"                     </span></span>
<span><span class="co">##  [3] "gene_ids"                           "name"                              </span></span>
<span><span class="co">##  [5] "species"                            "index_type"                        </span></span>
<span><span class="co">##  [7] "identities_label"                   "expression"                        </span></span>
<span><span class="co">##  [9] "expression_collapsed"               "groups"                            </span></span>
<span><span class="co">## [11] "identities"                         "sample_names"                      </span></span>
<span><span class="co">## [13] "num_identities"                     "num_samples"                       </span></span>
<span><span class="co">## [15] "num_genes"                          "num_strata"                        </span></span>
<span><span class="co">## [17] "index_full_name"                    "group_map"                         </span></span>
<span><span class="co">## [19] "TXI"                                "TXI_sample"                        </span></span>
<span><span class="co">## [21] "null_conservation_sample_size"      "precomputed_null_conservation_txis"</span></span>
<span><span class="co">## [23] "null_conservation_txis"             ".expression"                       </span></span>
<span><span class="co">## [25] ".groups"                            "precomputed_bootstrapped_txis"     </span></span>
<span><span class="co">## [27] "bootstrapped_txis"</span></span></code></pre>
</details><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># let's explore the properties using @</span></span>
<span><span class="co"># for example:</span></span>
<span><span class="va">example_phyex_set</span><span class="op">@</span><span class="va">strata</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">example_phyex_set</span><span class="op">@</span><span class="va">gene_ids</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">example_phyex_set</span><span class="op">@</span><span class="va">.expression</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<p>See how rich this <code>BulkPhyloExpressionSet</code> object is!</p>
<p>The <code>BulkPhyloExpressionSet</code> is designed to ensure
efficient interaction with <code>myTAIv2</code> functions. For example,
if you try the <code><a href="../reference/stat_flatline_test.html">myTAI::stat_flatline_test()</a></code> function twice,
i.e.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_flatline_test.html">stat_flatline_test</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_flatline_test.html">stat_flatline_test</a></span><span class="op">(</span><span class="va">example_phyex_set</span><span class="op">)</span></span></code></pre></div>
<p>you will notice that the permutations are already pre-computed for
the second run so that you donâ€™t need to wait a couple of seconds again.
Note, we will get to the <code>stat_flatline_test</code> function in the
<a href="tai-stats.html">statistical testing vignette</a>.</p>
<p>But how do you get your <code>.tsv</code> and <code>.csv</code> files
and convert your <code>data.frame</code> and <code>tibble</code> to a S7
<code>BulkPhyloExpressionSet</code> object?</p>
<div class="note">
<p>We will mention the transcriptome age index (TAI) a lot. See <a href="phylostratigraphy.html#gene-age-information-in-tai">ðŸ“š</a> for
more details on the <code>TAI</code> and its formula:
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi><mi>A</mi><msub><mi>I</mi><mi>s</mi></msub><mo>=</mo><mfrac><mrow><msubsup><mo>âˆ‘</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></msubsup><mi>p</mi><msub><mi>s</mi><mi>i</mi></msub><mo>â‹…</mo><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow><mrow><msubsup><mo>âˆ‘</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></msubsup><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">TAI_s = \frac{\sum_{i=1}^{n} ps_i \cdot e_{is}}{\sum_{i=1}^{n} e_{is}}</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub><annotation encoding="application/x-tex">e_{is}</annotation></semantics></math>
is the expression level of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
at a given sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>
(e.g.Â a biological replicate for a developmental stage), and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><msub><mi>s</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">ps_i</annotation></semantics></math>
is the evolutionary age of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>.</p>
</div>
</div>
<div class="section level2">
<h2 id="constructing-bulkphyloexpressionset-and-scphyloexpressionset">Constructing BulkPhyloExpressionSet and ScPhyloExpressionSet<a class="anchor" aria-label="anchor" href="#constructing-bulkphyloexpressionset-and-scphyloexpressionset"></a>
</h2>
<p>The necessary and sufficient information needed for creating a
<code>BulkPhyloExpressionSet</code> (as well as a
<code>ScPhyloExpressionSet</code>) object is:<br>
(1) A matrix or data frame of gene expression values (genes x
samples).<br>
(2) A numerical value or factored string associated with each gene,
typically gene age ranks or <a href="phylostratigraphy.html"><code>phylorank</code></a> for
<code>TAI</code> (but also <a href="other_strata.html"><code>deciled dNdS values</code></a> for
<code>TDI</code> or <a href="other_strata.html"><code>deciled tau values</code></a> for
<code>TSI</code> etc.).</p>
<div class="section level3">
<h3 id="loading-raw-data">Loading raw data<a class="anchor" aria-label="anchor" href="#loading-raw-data"></a>
</h3>
<p>The raw outputs of (1) gene expression quantification and (2) gene
age inference can typically be found in the form of a tab-separated
<code>.tsv</code> or comma-separated <code>.csv</code> file format. One
useful package to do this is <code>readr</code>, using the functions
<code><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">readr::read_csv()</a></code> or <code><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">readr::read_tsv()</a></code>. For
<code>.csv</code> files, if <code><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">readr::read_csv()</a></code> doesnâ€™t work,
try <code><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">readr::read_csv2()</a></code> in European locales.</p>
</div>
<div class="section level3">
<h3 id="mock-bulk-dataset-for-bulkphyloexpressionset">Mock (bulk) dataset for BulkPhyloExpressionSet<a class="anchor" aria-label="anchor" href="#mock-bulk-dataset-for-bulkphyloexpressionset"></a>
</h3>
<div class="section level4">
<h4 id="bulk-rna-seq-data-without-replicates">Bulk RNA-seq data without replicates<a class="anchor" aria-label="anchor" href="#bulk-rna-seq-data-without-replicates"></a>
</h4>
<p>In this example, we are using a dataset with only one entry per stage
(no replicates).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_phyex_set_old"</span><span class="op">)</span></span>
<span><span class="va">example_expression</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">example_phyex_set_old</span><span class="op">@</span><span class="va">.expression</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span>
<span><span class="va">example_phylorank</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">example_phyex_set_old</span><span class="op">@</span><span class="va">strata</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># let's explore the example expression dataset</span></span>
<span><span class="co"># for example:</span></span>
<span><span class="va">example_expression</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">##      GeneID     Zygote   Quadrant   Globular      Heart    Torpedo       Bent</span></span>
<span><span class="co">## 1 AT1G01040  2173.6352  1911.2001  1152.5553  1291.4224  1000.2529   962.9772</span></span>
<span><span class="co">## 2 AT1G01050  1501.0141  1817.3086  1665.3089  1564.7612  1496.3207  1114.6435</span></span>
<span><span class="co">## 3 AT1G01070  1212.7927  1233.0023   939.2000   929.6195   864.2180   877.2060</span></span>
<span><span class="co">## 4 AT1G01080  1016.9203   936.3837  1181.3381  1329.4734  1392.6429  1287.9746</span></span>
<span><span class="co">## 5 AT1G01090 11424.5667 16778.1685 34366.6494 39775.6405 56231.5689 66980.3673</span></span>
<span><span class="co">## 6 AT1G01120   844.0414   787.5929   859.6267   931.6180   942.8453   870.2625</span></span>
<span><span class="co">##      Mature</span></span>
<span><span class="co">## 1 1696.4274</span></span>
<span><span class="co">## 2 1071.6555</span></span>
<span><span class="co">## 3  894.8189</span></span>
<span><span class="co">## 4  861.2605</span></span>
<span><span class="co">## 5 7772.5617</span></span>
<span><span class="co">## 6  792.7542</span></span></code></pre>
As you can see, there is one column as the gene identifier and the rest
of the columns are the expression values for each developmental stage.
Instead of developmental stage, one can also use experimental
conditions, e.g.Â <code>control</code>, <code>treatment</code>, etc.
</details><p>Now we construct the input data.frame for
<code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set.df</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>phylorank <span class="op">=</span> <span class="va">example_phylorank</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">phylorank</span>, <span class="va">GeneID</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">example_expression</span>, by <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set.df</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">example_phyex_set.df</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">##            phylorank    GeneID     Zygote   Quadrant   Globular      Heart</span></span>
<span><span class="co">## 1 Cellular Organisms AT1G01040  2173.6352  1911.2001  1152.5553  1291.4224</span></span>
<span><span class="co">## 2 Cellular Organisms AT1G01050  1501.0141  1817.3086  1665.3089  1564.7612</span></span>
<span><span class="co">## 3          Eukaryota AT1G01070  1212.7927  1233.0023   939.2000   929.6195</span></span>
<span><span class="co">## 4 Cellular Organisms AT1G01080  1016.9203   936.3837  1181.3381  1329.4734</span></span>
<span><span class="co">## 5 Cellular Organisms AT1G01090 11424.5667 16778.1685 34366.6494 39775.6405</span></span>
<span><span class="co">## 6 Cellular Organisms AT1G01120   844.0414   787.5929   859.6267   931.6180</span></span>
<span><span class="co">##      Torpedo       Bent    Mature</span></span>
<span><span class="co">## 1  1000.2529   962.9772 1696.4274</span></span>
<span><span class="co">## 2  1496.3207  1114.6435 1071.6555</span></span>
<span><span class="co">## 3   864.2180   877.2060  894.8189</span></span>
<span><span class="co">## 4  1392.6429  1287.9746  861.2605</span></span>
<span><span class="co">## 5 56231.5689 66980.3673 7772.5617</span></span>
<span><span class="co">## 6   942.8453   870.2625  792.7542</span></span></code></pre>
<pre><code><span><span class="co">## 'data.frame':    25096 obs. of  9 variables:</span></span>
<span><span class="co">##  $ phylorank: Factor w/ 19 levels "Cellular Organisms",..: 1 1 2 1 1 1 1 1 1 1 ...</span></span>
<span><span class="co">##  $ GeneID   : chr  "AT1G01040" "AT1G01050" "AT1G01070" "AT1G01080" ...</span></span>
<span><span class="co">##  $ Zygote   : num  2174 1501 1213 1017 11425 ...</span></span>
<span><span class="co">##  $ Quadrant : num  1911 1817 1233 936 16778 ...</span></span>
<span><span class="co">##  $ Globular : num  1153 1665 939 1181 34367 ...</span></span>
<span><span class="co">##  $ Heart    : num  1291 1565 930 1329 39776 ...</span></span>
<span><span class="co">##  $ Torpedo  : num  1000 1496 864 1393 56232 ...</span></span>
<span><span class="co">##  $ Bent     : num  963 1115 877 1288 66980 ...</span></span>
<span><span class="co">##  $ Mature   : num  1696 1072 895 861 7773 ...</span></span></code></pre>
</details><div class="important">
<p><strong>Requirements for the input <code>data.frame</code> to the
<code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code> function:</strong><br>
column 1 contains the phylorank information, which can be either numeric
or factors.<br>
column 2 contains the gene identifier (here called
<code>GeneID</code>).<br>
column 3 onwards contain the expression data with the column titles
being the developmental stages or the experimental conditions.</p>
</div>
<p>The example <code>example_phyex_set.df</code> is now formatted to
meet the requirements for the <code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code>
function.</p>
<div class="tip">
<p>The <code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code> function will
automatically detect the first column as the phylorank and the second
column as the gene identifier. If your data is structured differently,
you can reassign the columns accordingly using
<code><a href="https://dplyr.tidyverse.org/reference/relocate.html" class="external-link">dplyr::relocate()</a></code>.</p>
</div>
<p>Now, we can use the <code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code> function
to convert the correctly formatted input <code>data.frame</code> to a
<code>BulkPhyloExpressionSet</code> object.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set.remake</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet</a></span><span class="op">(</span></span>
<span>        <span class="va">example_phyex_set.df</span>,</span>
<span>        name <span class="op">=</span> <span class="st">"reconstituted phyex_set_old"</span><span class="op">)</span></span></code></pre></div>
<p>There you have it, a <code>BulkPhyloExpressionSet</code> object!</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set.remake</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">## PhyloExpressionSet object</span></span>
<span><span class="co">## Class: myTAI::BulkPhyloExpressionSet </span></span>
<span><span class="co">## Name: reconstituted phyex_set_old </span></span>
<span><span class="co">## Species: NA </span></span>
<span><span class="co">## Index type: TXI </span></span>
<span><span class="co">## Identities : Zygote, Quadrant, Globular, Heart, Torpedo, Bent, Mature </span></span>
<span><span class="co">## Number of genes: 25096 </span></span>
<span><span class="co">## Number of identities : 7 </span></span>
<span><span class="co">## Number of phylostrata: 19 </span></span>
<span><span class="co">## Number of samples: 7 </span></span>
<span><span class="co">## Samples per condition: 1 1 1 1 1 1 1</span></span></code></pre>
</details><p>You can further explore the properties of the
<code>BulkPhyloExpressionSet</code> object using the <code>@</code>
operator, for example:
<code>example_phyex_set.remake@counts |&gt; head()</code>.</p>
<p>â€¦ and plot your transcriptome age index!</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set.remake</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [====================                    ] 49%  (~1s remaining)       Computing: [========================================] 99%  (~0s remaining)       Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="phylo-expression-object_files/figure-html/unnamed-chunk-18-1.png" alt="plot signature function output for bulk RNAseq" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="what-if-the-dataset-contains-replicates">What if the dataset contains replicates?<a class="anchor" aria-label="anchor" href="#what-if-the-dataset-contains-replicates"></a>
</h4>
<p>This is exactly the case for the new <code>example_phyex_set</code>
dataset (not like <code>example_phyex_set_old</code>), which contains
replicates for each developmental stage. The
<code><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code> function can handle this as
well, as long as we use the <code>groups</code> parameter to specify the
grouping of replicates, i.e.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_phyex_set"</span><span class="op">)</span> <span class="co"># not data("example_phyex_set_old")</span></span>
<span><span class="va">example_expression</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">example_phyex_set</span><span class="op">@</span><span class="va">.expression</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span>
<span><span class="va">example_phylorank</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">example_phyex_set</span><span class="op">@</span><span class="va">strata</span></span>
<span></span>
<span><span class="va">example_phyex_set.df</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>phylorank <span class="op">=</span> <span class="va">example_phylorank</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">phylorank</span>, <span class="va">GeneID</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">example_expression</span>, by <span class="op">=</span> <span class="st">"GeneID"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">example_phyex_set.df</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">##  [1] "phylorank" "GeneID"    "pg_1"      "pg_2"      "pg_3"      "gl_1"     </span></span>
<span><span class="co">##  [7] "gl_2"      "gl_3"      "eh_1"      "eh_2"      "eh_3"      "lh_1"     </span></span>
<span><span class="co">## [13] "lh_2"      "lh_3"      "et_1"      "et_2"      "et_3"      "lt_1"     </span></span>
<span><span class="co">## [19] "lt_2"      "lt_3"      "bc_1"      "bc_2"      "bc_3"      "mg_1"     </span></span>
<span><span class="co">## [25] "mg_2"      "mg_3"</span></span></code></pre>
</details><p>As you can see, this dataset has three replicates per stage.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups_example_phyex_set.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Preglobular"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Globular"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Early Heart"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Late Heart"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Early Torpedo"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Late Torpedo"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Bent Cotyledon"</span>,<span class="fl">3</span><span class="op">)</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Mature Green"</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># we can add the group information using groups</span></span>
<span><span class="va">example_phyex_set.remake</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet</a></span><span class="op">(</span></span>
<span>        <span class="va">example_phyex_set.df</span>,</span>
<span>        groups <span class="op">=</span> <span class="va">groups_example_phyex_set.df</span>, <span class="co"># adding group information here</span></span>
<span>        name <span class="op">=</span> <span class="st">"reconstituted phyex_set"</span><span class="op">)</span></span></code></pre></div>
<p>Done!</p>
<div class="tip">
<p>Along with the <code>groups</code> parameter, you can also specify
the <code>name</code>, <code>strata_labels</code> and other parameters
to provide more descriptions for your
<code>BulkPhyloExpressionSet</code> object. This is useful for plotting
and visualisation purposes. Check this using <code>?</code> before the
function (i.e.Â <code>?myTAI::as_BulkPhyloExpressionSet()</code>.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="mock-single-cell-dataset-for-scphyloexpressionset">Mock single-cell dataset for ScPhyloExpressionSet<a class="anchor" aria-label="anchor" href="#mock-single-cell-dataset-for-scphyloexpressionset"></a>
</h3>
<p>If you are interested in single cell RNA-seq data, the process is
similar, but you will need to ensure that your data is structured
correctly for single cell analysis, i.e.Â as a <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html" class="external-link"><code>seurat object</code></a>.</p>
<p>You can then use the <code><a href="../reference/as_ScPhyloExpressionSet.html">as_ScPhyloExpressionSet()</a></code> function
with adjustments to the relevant metadata.</p>
<p><img src="Figures/myTAI_scphyex.png" alt="scPhyEx overview" align="center" style="max-height: 500px; max-width: 800px;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the relevant packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">pbmc_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'pbmc_raw.txt'</span>, package <span class="op">=</span> <span class="st">'Seurat'</span><span class="op">)</span>,</span>
<span>  as.is <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The input file for <code><a href="https://satijalab.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">Seurat::CreateSeuratObject()</a></code> is a
tab-separated file with the first column as gene identifiers and the
rest of the columns as expression values for each cell. The
<code>pbmc_raw</code> object is a data frame with gene expression
data.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pbmc_raw</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">##          ATGCCAGAACGACT CATGGCCTGTGCAT GAACCTGATGAACC TGACTGGATTCTCA</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 1              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               0              1              0              0</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              1              0              0              0</span></span>
<span><span class="co">##          AGTCAGACTGCACA TCTGATACACGTGT TGGTATCTAAACAG GCAGCTCTGTTTCT</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               1              1              0              1</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              0</span></span>
<span><span class="co">##          GATATAACACGCAT AATGTTGACAGTCA AGGTCATGAGTGTC AGAGATGATCTCGC</span></span>
<span><span class="co">## MS4A1                 0              0              2              2</span></span>
<span><span class="co">## CD79B                 0              1              2              4</span></span>
<span><span class="co">## CD79A                 0              0              0              5</span></span>
<span><span class="co">## HLA-DRA               0              0             14             28</span></span>
<span><span class="co">## TCL1A                 0              0              3              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              1              6</span></span>
<span><span class="co">##          GGGTAACTCTAGTG CATGAGACACGGGA TACGCCACTCCGAA CTAAACCTGTGCAT</span></span>
<span><span class="co">## MS4A1                 4              4              2              3</span></span>
<span><span class="co">## CD79B                 3              3              2              3</span></span>
<span><span class="co">## CD79A                 2              2              5              8</span></span>
<span><span class="co">## HLA-DRA              18              7             15             28</span></span>
<span><span class="co">## TCL1A                 2              4              0              0</span></span>
<span><span class="co">## HLA-DQB1              2              2              2              8</span></span>
<span><span class="co">##          GTAAGCACTCATTC TTGGTACTGAATCC CATCATACGGAGCA TACATCACGCTAAC</span></span>
<span><span class="co">## MS4A1                 3              4              2              3</span></span>
<span><span class="co">## CD79B                 1              2              2              5</span></span>
<span><span class="co">## CD79A                 1              5              5             12</span></span>
<span><span class="co">## HLA-DRA               7             26             10             16</span></span>
<span><span class="co">## TCL1A                 3              3              3              2</span></span>
<span><span class="co">## HLA-DQB1              2              2              1              2</span></span>
<span><span class="co">##          TTACCATGAATCGC ATAGGAGAAACAGA GCGCACGACTTTAC ACTCGCACGAAAGT</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              1              0</span></span>
<span><span class="co">## HLA-DRA               7             22              0             10</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              3              0              0</span></span>
<span><span class="co">##          ATTACCTGCCTTAT CCCAACTGCAATCG AAATTCGAATCACG CCATCCGATTCGCC</span></span>
<span><span class="co">## MS4A1                 1              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              1</span></span>
<span><span class="co">## HLA-DRA               6              0              4              3</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              1              0</span></span>
<span><span class="co">##          TCCACTCTGAGCTT CATCAGGATGCACA CTAAACCTCTGACA GATAGAGAAGGGTG</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              1              1              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               7             13              0              1</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              1              0              0              0</span></span>
<span><span class="co">##          CTAACGGAACCGAT AGATATACCCGTAA TACTCTGAATCGAC GCGCATCTTGCTCC</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 2              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               0              0              1              0</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              0</span></span>
<span><span class="co">##          GTTGACGATATCGG ACAGGTACTGGTGT GGCATATGCTTATC CATTACACCAACTG</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               1              1              0              0</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              0</span></span>
<span><span class="co">##          TAGGGACTGAACTC GCTCCATGAGAAGT TACAATGATGCTAG CTTCATGACCGAAT</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               0              0              0              0</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              0</span></span>
<span><span class="co">##          CTGCCAACAGGAGC TTGCATTGAGCTAC AAGCAAGAGCTTAG CGGCACGAACTCAG</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               0              1              1              1</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              0</span></span>
<span><span class="co">##          GGTGGAGATTACTC GGCCGATGTACTCT CGTAGCCTGTATGC TGAGCTGAATGCTG</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              1              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               0              0             10             10</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              1</span></span>
<span><span class="co">##          CCTATAACGAGACG ATAAGTTGGTACGT AAGCGACTTTGACG ACCAGTGAATACCG</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 1              1              2              2</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               4              1              6             28</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              1              0              2              0</span></span>
<span><span class="co">##          ATTGCACTTGCTTT CTAGGTGATGGTTG GCACTAGACCTTTA CATGCGCTAGTCAC</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              3              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA              10             13              5              8</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              1              1              0</span></span>
<span><span class="co">##          TTGAGGACTACGCA ATACCACTCTAAGC CATATAGACTAAGC TTTAGCTGTACTCT</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              4</span></span>
<span><span class="co">## CD79A                 0              0              0              8</span></span>
<span><span class="co">## HLA-DRA             108             93             41             42</span></span>
<span><span class="co">## TCL1A                 0              0              0              4</span></span>
<span><span class="co">## HLA-DQB1             21             21              3              5</span></span>
<span><span class="co">##          GACATTCTCCACCT ACGTGATGCCATGA ATTGTAGATTCCCG GATAGAGATCACGA</span></span>
<span><span class="co">## MS4A1                 0              0              0              1</span></span>
<span><span class="co">## CD79B                 1              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA             138             77             76             15</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1             11             11             10              1</span></span>
<span><span class="co">##          AATGCGTGGACGGA GCGTAAACACGGTT ATTCAGCTCATTGG GGCATATGGGGAGT</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 1              0              0              0</span></span>
<span><span class="co">## HLA-DRA              19            104              1              0</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              2             11              0              0</span></span>
<span><span class="co">##          ATCATCTGACACCA GTCATACTTCGCCT TTACGTACGTTCAG GAGTTGTGGTAGCT</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               0              0              2              1</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              0</span></span>
<span><span class="co">##          GACGCTCTCTCTCG AGTCTTACTTCGGA GGAACACTTCAGAC CTTGATTGATCTTC</span></span>
<span><span class="co">## MS4A1                 0              0              0              0</span></span>
<span><span class="co">## CD79B                 0              0              0              0</span></span>
<span><span class="co">## CD79A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DRA               1              0              2              7</span></span>
<span><span class="co">## TCL1A                 0              0              0              0</span></span>
<span><span class="co">## HLA-DQB1              0              0              0              1</span></span></code></pre>
</details><p>Now, we create a Seurat object. i.e.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pbmc_raw</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/is.html" class="external-link">is</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Seurat"</span></span></code></pre>
<p>Now, we can convert this Seurat object to a
<code>ScPhyloExpressionSet</code> object using the
<code><a href="../reference/as_ScPhyloExpressionSet.html">as_ScPhyloExpressionSet()</a></code> function.</p>
<p>The next part is a bit tricky, as you need to provide the phylorank
information in a specific format with the same gene identifiers as that
of <code>pbmc_small</code>.</p>
<p>We will generate <strong>random</strong> phyloranks for each gene as
integers.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_names</span> <span class="op">&lt;-</span> <span class="va">pbmc_small</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">example_phylorank_sc</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gene_names</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="va">gene_names</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set_sc</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/as_ScPhyloExpressionSet.html">as_ScPhyloExpressionSet</a></span><span class="op">(</span></span>
<span>        <span class="va">pbmc_small</span>,</span>
<span>        strata <span class="op">=</span> <span class="va">example_phylorank_sc</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Example of a downstream analysis
</summary><p>Here, we will go a bit deeper and cluster the <code>pbmc_small</code>
object before integrating this to the <code>myTAIv2</code> workflow.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example workflow to cluster the single cell data</span></span>
<span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span></span>
<span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">pbmc_small</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span></span>
<span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">pbmc_small</span>, features <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc_small</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pbmc_small</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">pbmc_small</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">pbmc_small.cluster</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">pbmc_small</span>, resolution <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Number of nodes: 80</span></span>
<span><span class="co">## Number of edges: 2352</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Running Louvain algorithm...</span></span>
<span><span class="co">## Maximum modularity in 10 random starts: 0.4014</span></span>
<span><span class="co">## Number of communities: 2</span></span>
<span><span class="co">## Elapsed time: 0 seconds</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set_sc.cluster</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/as_ScPhyloExpressionSet.html">as_ScPhyloExpressionSet</a></span><span class="op">(</span></span>
<span>        <span class="va">pbmc_small.cluster</span>,</span>
<span>        strata <span class="op">=</span> <span class="va">example_phylorank_sc</span><span class="op">)</span></span></code></pre></div>
<p>Now we have the <code>ScPhyloExpressionSet</code> object,</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_phyex_set_sc.cluster</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">## PhyloExpressionSet object</span></span>
<span><span class="co">## Class: myTAI::ScPhyloExpressionSet </span></span>
<span><span class="co">## Name: Single-Cell Phylo Expression Set </span></span>
<span><span class="co">## Species: NA </span></span>
<span><span class="co">## Index type: TXI </span></span>
<span><span class="co">## Identities : 0, 1 </span></span>
<span><span class="co">## Number of genes: 230 </span></span>
<span><span class="co">## Number of identities : 2 </span></span>
<span><span class="co">## Number of phylostrata: 10 </span></span>
<span><span class="co">## Expression layer used: counts </span></span>
<span><span class="co">## Total cells: 80 </span></span>
<span><span class="co">## Valid cells: 80 </span></span>
<span><span class="co">## Cells per type:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  0  1 </span></span>
<span><span class="co">## 44 36</span></span></code></pre>
</details><p>â€¦ which you can now plot!</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set_sc.cluster</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing: [========================================] 100% (done)                         </span></span>
<span><span class="co">## Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="phylo-expression-object_files/figure-html/unnamed-chunk-35-1.png" alt="plot signature function output for single cell" width="576" style="display: block; margin: auto;"></p>
<div class="important">
<p>This example used an example Seurat object with a mock phylorank
dataset (random gene age assignment) and an example workflow. In
practice, you will have to follow the best standards for single cell
RNA-seq data analysis, including quality control, normalisation, and
clustering steps. And assign real phylorank information to the genes in
your dataset (to do this, see the <a href="phylostratigraphy.html">phylostratigraphy vignette</a>).</p>
</div>
</details>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>In this section, we have learned how to create
<code>BulkPhyloExpressionSet</code> and
<code>ScPhyloExpressionSet</code> objects from raw gene expression data
and phylorank information. This object is essential for performing
evolutionary transcriptomics analyses using the <code>myTAI</code>
package, and provides a structured way to store and manipulate
evolutionary transcriptomics data, making it easier to perform various
analyses and visualisations.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, È˜tefan Manolache, Nikola KalÃ¡bovÃ¡, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
