<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Gene age inference • myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Gene age inference">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.3.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-stats.html">Statistical testing with myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-transform.html">Transforming dataset for myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/index.html">... and view more!</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Gene age inference</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/vignettes/phylostratigraphy.Rmd" class="external-link"><code>vignettes/phylostratigraphy.Rmd</code></a></small>
      <div class="d-none name"><code>phylostratigraphy.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="what-is-genomic-phylostratigraphy">What is genomic phylostratigraphy?<a class="anchor" aria-label="anchor" href="#what-is-genomic-phylostratigraphy"></a>
</h2>
<p>Genomic phylostratigraphy is a method to date the evolutionary
emergence of a given gene (i.e. <code>gene age inference</code>),
typically done for all genes in a genome. In practice, one performs
homology searches against a taxonomically-informed database such as the
NCBI NR, and then parse the output to obtain an evolutionary-informed
readout.</p>
<p><img src="Figures/Basics_of_phylostratigraphy.png" alt="phylostratigraphy overview" align="center" style="max-height: 500px; max-width: 500px;"></p>
<p style="font-size:10px;">
Overview of gene age inference. Here, we are searching two genes from
the genome of interest against a taxonomically-informed database. The
first gene is assigned as <em>relative age</em> 1 since homologues can
be found as far as node 1. The second gene is assigned as <em>relative
age</em> 3 since homologues can be found only up to node 3. Adapted from
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02895-z" class="external-link">Barrera-Redondo
et al., 2023</a>.
</p>
<p>The output can be fed into <code>myTAIv2</code> and, combined with
expression data, is used to compute the transcriptome age index
(<code>TAI</code>). More specifically, gene age inference generates a
table storing the gene age in the first column and the corresponding
gene id of the organism of interest in the second column. This table is
named <em>phylostratigraphic map</em> (or <em>phylomap</em>).</p>
<div class="note">
<p>For quick users, there are several recent software developed for gene
age inference such as <a href="https://github.com/josuebarrera/GenEra" class="external-link"><code>GenEra</code></a>
and <a href="https://github.com/kullrich/oggmap" class="external-link"><code>oggmap</code></a>.</p>
<details><summary>
Example <code>GenEra</code> workflow
</summary><p>Assuming that the user has installed GenEra.</p>
<ol style="list-style-type: decimal">
<li><p>Download a taxonomically-resolved protein sequence database such
as the NCBI nr, as well as prot.accession2taxid and taxonomy dump from
the NCBI.</p></li>
<li><p>Run <a href="https://github.com/bbuchfink/diamond" class="external-link"><code>diamond makedb</code></a>.</p></li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">diamond</span> makedb <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a> <span class="at">--in</span> nr <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a> <span class="at">--db</span> nr <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a> <span class="at">--taxonmap</span> prot.accession2taxid <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a> <span class="at">--taxonnodes</span> taxdump/nodes.dmp <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a> <span class="at">--taxonnames</span> taxdump/names.dmp</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Run <a href="https://github.com/josuebarrera/GenEra" class="external-link"><code>GenEra</code></a>.<br>
</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">genEra</span> <span class="at">-q</span> <span class="pp">[</span><span class="ss">query_sequences.fasta</span><span class="pp">]</span> <span class="at">-t</span> <span class="pp">[</span><span class="ss">query_taxid</span><span class="pp">]</span> <span class="at">-b</span> <span class="pp">[</span><span class="ss">path/to/nr</span><span class="pp">]</span> <span class="at">-d</span> <span class="pp">[</span><span class="ss">path/to/taxdump</span><span class="pp">]</span></span></code></pre></div>
</details>
</div>
<p>But we hope you stay for the debate/discussion!</p>
</div>
<div class="section level2">
<h2 id="gene-age-information-in-tai">Gene age information in TAI<a class="anchor" aria-label="anchor" href="#gene-age-information-in-tai"></a>
</h2>
<p><code>TAI</code> (<a href="https://www.nature.com/articles/nature09632" class="external-link">Domazet-Lošo &amp;
Tautz, 2010</a>) is calculated as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mrow><mi>T</mi><mi>A</mi><mi>I</mi></mrow><mi>s</mi></msub><mo>=</mo><mfrac><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msub><mrow><mi>p</mi><mi>s</mi></mrow><mi>i</mi></msub><mo>⋅</mo><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow><mrow><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>n</mi></munderover><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">
{TAI}_s = \frac{\sum_{i=1}^{n} {ps}_i \cdot e_{is}}{\sum_{i=1}^{n} e_{is}}
</annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>e</mi><mrow><mi>i</mi><mi>s</mi></mrow></msub><annotation encoding="application/x-tex">e_{is}</annotation></semantics></math>
is the expression level of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
at a given sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>
(e.g. a biological replicate for a developmental stage), and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>p</mi><mi>s</mi></mrow><mi>i</mi></msub><annotation encoding="application/x-tex">{ps}_i</annotation></semantics></math>
is the evolutionary age of gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>.</p>
<p>For developmental time-course data,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>A</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TAI}_s</annotation></semantics></math>
can be grouped for each stage (if replicate data exists) and compared
across developmental stages to obtain the overall TAI profile.<br>
For pairwise comparisons (see <a href="tai-stats.html">📈</a>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>A</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TAI}_s</annotation></semantics></math>
can be grouped as one of the conditions to be compared.<br>
For single cell data, (see <a href="phylo-expression-object.html">📊</a>),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mrow><mi>T</mi><mi>A</mi><mi>I</mi></mrow><mi>s</mi></msub><annotation encoding="application/x-tex">{TAI}_s</annotation></semantics></math>
is the TAI for a given cell or cell type (if pseudo-bulked).</p>
<p>Thus, TAI represents the expression-weighted mean gene age. Accurate
inference of gene age that scales to all genes in the genome is thus
key.</p>
</div>
<div class="section level2">
<h2 id="the-debate">The debate<a class="anchor" aria-label="anchor" href="#the-debate"></a>
</h2>
<p>Despite its simplicity, since the first studies (<a href="https://doi.org/10.1016/j.tig.2007.08.014" class="external-link">Domazet-Loso et al.,
2007</a>), the method and concept underlying genomic phylostratigraphy
has been subject to intense discussions (<a href="https://doi.org/10.1016/j.tig.2013.07.001" class="external-link">Capra et al., 2013</a>;
<a href="https://www.nature.com/articles/nmeth.3830.pdf" class="external-link">Altenhoff et
al., 2016</a>; <a href="https://doi.org/10.1093/gbe/evw113" class="external-link">Liebeskind
et al., 2016</a>; <a href="https://doi.org/10.1093/molbev/msw284" class="external-link">Domazet-Loso et al.,
2017</a>; <a href="https://doi.org/10.1093/bib/bby074" class="external-link">Yin et al.,
2018</a>; <a href="https://doi.org/10.1093/gbe/evy231" class="external-link">Casola 2018</a>;
<a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000862" class="external-link">Weisman
et al., 2020</a>; <a href="https://doi.org/10.1093/molbev/msaa217" class="external-link">Futo
et al., 2021</a>; <a href="https://doi.org/10.1016/j.cub.2022.04.085" class="external-link">Weisman et al.,
2022</a>; <a href="https://link.springer.com/article/10.1186/s13059-023-02895-z" class="external-link">Barrera-Redondo
et al., 2023</a>; etc.).</p>
<p>The crux of the discussion focuses on the homology search bias,
though several studies have also noted database bias (e.g., species
representation and contamination) and evolutionary assumptions (e.g.,
Dollo parsimony).</p>
<div class="section level3">
<h3 id="homology-search-bias">Homology search bias<a class="anchor" aria-label="anchor" href="#homology-search-bias"></a>
</h3>
<p>The first phylostratigraphy studies used <code>BLASTp</code>, which
performs pairwise protein sequence alignment, for the homology search
step. At the time of these study (late-2000s/early-2010s),
<code>BLASTp</code> was chosen as it balanced the requirements of
sensitivity and speed when performing sequence homology searches against
(at that time) a large databases.</p>
<p>The following discussions pertain to the potential biases with
<code>BLASTp</code> in particular and pairwise sequence aligners in
general.</p>
<details><summary><strong><code>BLASTp</code> underestimates of gene age (for certain
genes)</strong>
</summary><p>Moyers &amp; Zhang argue that the use of <code>BLASTp</code> genomic
phylostratigraphy</p>
<ol style="list-style-type: decimal">
<li><p>underestimates gene age for a considerable fraction of
genes,</p></li>
<li><p>is biased for rapidly evolving proteins which are short, and/or
their most conserved block of sites is small, and</p></li>
<li><p>these biases create spurious nonuniform distributions of various
gene properties among age groups, many of which cannot be predicted a
priori (<a href="https://doi.org/10.1093/molbev/msu286" class="external-link">Moyers &amp;
Zhang, 2015</a>; <a href="https://doi.org/10.1093/molbev/msw008" class="external-link">Moyers
&amp; Zhang, 2016</a>; <a href="https://doi.org/10.1093/gbe/evw113" class="external-link">Liebeskind et al.,
2016</a>).</p></li>
</ol>
<p>However, these arguments were based on simulated data and were
inconclusive due to <a href="https://doi.org/10.1093/molbev/msw202" class="external-link">errors in their
analyses</a>. Furthermore, <a href="https://doi.org/10.1093/molbev/msw284" class="external-link">Domazet-Loso et al.,
2017</a> provide convincing evidence that there is <strong>no</strong>
phylostratigraphic bias. As a response, <a href="https://doi.org/10.1093/gbe/evx109" class="external-link">Moyers &amp; Zhang, 2017</a>
published a counter-study stating that a phylostratigraphic trend
claimed by <a href="https://doi.org/10.1093/molbev/msw284" class="external-link">Domazet-Loso
et al., 2017</a> to be robust to error disappears when genes likely to
be error-resistant are analyzed. <a href="https://doi.org/10.1093/gbe/evx109" class="external-link">Moyers &amp; Zhang, 2017</a>
further suggest a more robust methodology for controlling for the
effects of error by first restricting to those genes which can be
simulated and then removing those genes which, through simulation, have
been shown to be error-prone (see also <a href="https://doi.org/10.1093/gbe/evy161" class="external-link">Moyers &amp; Zhang,
2018</a>).</p>
<p>In general, an objective benchmarking set representing the tree of
life is still missing and therefore any procedure aiming to quantify
gene ages will be biased to some degree. Based on this discussion <a href="https://doi.org/10.1093/gbe/evw113" class="external-link">Liebeskind et al., 2016</a>
suggest inferring gene age by combining several common orthology
inference algorithms to create gene age datasets and then characterize
the error around each age-call on a per-gene and per-algorithm basis.
Using this approach systematic error was found to be a large factor in
estimating gene age, suggesting that simple consensus algorithms are not
enough to give a reliable point estimate (<a href="https://doi.org/10.1093/gbe/evw113" class="external-link">Liebeskind et al., 2016</a>).
This was also observed by <a href="https://doi.org/10.1093/gbe/evy161" class="external-link">Moyers &amp; Zhang, 2018</a>
when running alternative tools such as <code>PSIBLAST</code>,
<code>HMMER</code>, <code>OMA</code>, etc. However, by generating a
consensus gene age and quantifying the possible error in each workflow
step, <a href="https://doi.org/10.1093/gbe/evw113" class="external-link">Liebeskind et al.,
2016</a> provide a very useful <a href="https://github.com/marcottelab/Gene-Ages" class="external-link">database</a> of
consensus gene ages for a variety of genomes.</p>
<p>Alternatively, <a href="https://doi.org/10.1093/bib/bbw034" class="external-link">Smith
&amp; Pease, 2016</a> argues that <em>de novo</em> gene birth/death and
gene family expansion/contraction studies should avoid drawing direct
inferences of evolutionary relatedness from measures of sequence
similarity alone, and should instead, where possible, use more rigorous
phylogeny-based methods. For this purpose, we recommend researchers to
consult the <a href="https://orthology.phylomedb.org/" class="external-link">phylomedb
database</a> to retrieve phylogeny-based gene orthology relationships
and use these age estimates in combination with <code>myTAI</code>.
Similarly, users can also retrieve gene ages from <a href="http://eggnog5.embl.de/#/app/home" class="external-link">EggNOG database</a> using <a href="https://github.com/kullrich/orthomap" class="external-link"><code>orthomap</code></a>.
Alternatively, users might find the simulation based removal approach
proposed by <a href="https://doi.org/10.1093/gbe/evy161" class="external-link">Moyers &amp;
Zhang, 2018</a> more suitable.</p>
</details><details><summary><strong><code>BLASTp</code> parameters are complicated</strong>
</summary><p>In these extensive debates, one aspect that is often missed and can
be directly corrected by an interested user is the
bioinformatics/technical aspect of using <code>BLASTp</code> or any
other BLAST-like tool for gene age inference. Namely,
<code>BLASTp</code> hits are biased by the use of the default argument
<code>max_target_seqs</code> (<a href="https://doi.org/10.1093/bioinformatics/bty833" class="external-link">Shah et al.,
2018</a>). The main issue of how this <code>max_target_seqs</code> is
set is that:</p>
<blockquote>
<p>According to the BLAST documentation itself (2008), this parameter
represents the ‘number of aligned sequences to keep’. This statement is
commonly interpreted as meaning that BLAST will return the top N
database hits for a sequence query if the value of max_target_seqs is
set to N. For example, in a recent article (Wang et al., 2016) the
authors explicitly state ‘Setting ’max target seqs’ as ‘1’ only the best
match result was considered’. <strong>To our surprise, we have recently
discovered that this intuition is incorrect.</strong> Instead, BLAST
returns the first N hits that exceed the specified E-value threshold,
which may or may not be the highest scoring N hits. The invocation using
the parameter ‘-max_target_seqs 1’ simply returns the first good hit
found in the database, not the best hit as one would assume. Worse yet,
the output produced depends on the order in which the sequences occur in
the database. For the same query, different results will be returned by
BLAST when using different versions of the database even if all versions
contain the same best hit for this database sequence. Even ordering the
database in a different way would cause BLAST to return a different ‘top
hit’ when setting the max_target_seqs parameter to 1. - Shah et al.,
2018</p>
</blockquote>
<p>The solution to this issue seems to be that any <code>BLASTp</code>
search must be performed with a significantly high
<code>-max_target_seqs</code>, e.g. <code>-max_target_seqs 10000</code>
(see <a href="https://gist.github.com/sujaikumar/504b3b7024eaf3a04ef5" class="external-link uri">https://gist.github.com/sujaikumar/504b3b7024eaf3a04ef5</a>
for details) and best hits must be filtered subsequently. It is not
clear from any of the studies referenced above how the best
<code>BLASTp</code> hit was retrieved and which
<code>-max_target_seqs</code> values were used to perform
<code>BLASTp</code> searches in the respective study. Thus, the
comparability of the results between studies is impossible and any
individual claim made in these studies might be biased.</p>
<p>In addition, the <code>-max_target_seqs</code> argument issue seems
not to be the only issue that might influence technical differences in
<code>BLASTp</code> hit results. <a href="https://doi.org/10.1093/bioinformatics/bty1018" class="external-link">Gonzalez-Pech et
al., 2018</a> discuss another problem of retrieving the best
<code>BLASTp</code> hits based on <code>E-value</code> thresholds.</p>
<blockquote>
<p>Many users assume that BLAST alignment hits with E-values less than
or equal to the predefined threshold (e.g. 10<sup>-5</sup> via the
specification of evalue 1e-5) are identified after the search is
completed, in a final step to rank all alignments by E-value, from the
smallest (on the top of the list of results) to the largest E-value (at
the bottom of the list). <strong>However, the E-value filtering step
does not occur at the final stage of BLAST; it occurs earlier during the
scanning phase (Altschul et al., 1997; Camacho et al., 2009). During
this phase, a gapped alignment is generated using less-sensitive
heuristic parameters (Camacho et al., 2009); alignments with an E-value
that satisfies the defined cut-off are included in the subsequent phase
of the BLAST algorithm (and eventually reported). During the final
(trace-back) phase, these gapped alignments are further adjusted using
moresensitive heuristic parameters (Camacho et al., 2009), and the
E-value for each of these refined alignments is then
recalculated.</strong> - Gonzalez-Pech et al., 2018</p>
</blockquote>
<p>This means that if one study mentioned above ran a
<code>BLASTp</code> search with a BLAST parameter configuration of lets
say <code>-max_target_seqs 250</code> (default value in BLAST) and
<code>evalue 10</code> (default value in BLAST) and then subsequently
selected the best hit which returned the smallest <code>E-value</code>
and another study used the parameter configuration
<code>-max_target_seqs 1</code> and <code>evalue 0.0001</code> then the
results of both studies would not be comparable and the proposed gene
age inference bias might simply result from a technical difference in
running <code>BLASTp</code> searches.</p>
<p>In more detail, even if one study for example ran <code>BLASTp</code>
with <code>evalue 10</code> (default value in BLAST) and then
subsequently filtered for hits that resulted in
<code>evalue &lt; 0.0001</code> whereas another study ran
<code>BLASTp</code> directly with <code>evalue 0.0001</code>, according
to <a href="https://doi.org/10.1093/bioinformatics/bty1018" class="external-link">Gonzalez-Pech et
al., 2018</a> these studies although referring to the same
<code>E-value</code> threshold for filtering hits will result in
different sets of filtered <code>BLASTp</code> hits.</p>
</details>
</div>
<div class="section level3">
<h3 id="alternatives-to-blastp">Alternatives to <code>BLASTp</code><a class="anchor" aria-label="anchor" href="#alternatives-to-blastp"></a>
</h3>
<p>While BLASTP is considered the gold-standard tool for pairwise
protein sequence alignment, newer approaches (such as <a href="https://github.com/bbuchfink/diamond" class="external-link">DIAMOND v2</a> and <a href="https://github.com/soedinglab/mmseqs2/" class="external-link">MMSeqs2</a>) have been
developed, which can be used for gene age inference with the same
sensitivity - just much faster. Redesigned parameters in these software
can also overcome overlooked technical aspects of <code>BLASTp</code>
such as the <code>-max_target_seqs</code>.</p>
<p>Beyond pairwise protein sequence aligners, gene ages can also be
inferred via structural alignment, using synteny information and using
orthogroups.</p>
<details><summary><strong>Accelerated and sensitive sequence alignment</strong>
</summary><p>They retain the sensitivity of <code>BLASTp</code> while being
thousands to hundred-folds faster. This speed-up is crucial as sequence
databases increase in size. This is important for gene age inference and
<a href="https://github.com/bbuchfink/diamond" class="external-link">DIAMOND v2</a> and <a href="https://github.com/soedinglab/mmseqs2/" class="external-link">MMSeqs2</a> have been
employed in various phylostratigraphic and phylotranscriptomic analyses
(either explicitly or under-the-hood in tools such as <a href="https://github.com/davidemms/OrthoFinder" class="external-link">OrthoFinder</a>, which
uses DIAMOND by default) (see <a href="https://www.nature.com/articles/s41467-024-47017-w" class="external-link">Domazet-Loso
et al., 2024</a>; <a href="https://www.pnas.org/doi/full/10.1073/pnas.2216351120" class="external-link">Ma &amp;
Zheng 2023</a>; <a href="https://doi.org/10.1093/g3journal/jkad077" class="external-link">Manley et al.,
2023</a>).</p>
<p>With <code>DIAMOND</code>, users can overcome the overlooked
technical aspects of <code>BLASTp</code> such as the
<code>max_target_seqs</code> issue by setting
<code>--max-target-seqs</code> to <code>0</code> (or simply
<code>-k0</code>). Furthermore, the <code>--ultra-sensitive</code> mode
can be employed, which is as sensitive as <code>BLASTp</code> while
being 80x faster (see <a href="https://www.nature.com/articles/s41592-021-01101-x/figures/1" class="external-link">Fig
1. in the DIAMOND paper</a>).</p>
<p><a href="https://github.com/josuebarrera/GenEra" class="external-link">GenEra</a>, a
recently developed gene age inference tool, utilises <a href="https://github.com/bbuchfink/diamond" class="external-link">DIAMOND v2</a> with
appropriate parameter settings by default. Users can also specify the
use of <a href="https://github.com/soedinglab/mmseqs2/" class="external-link">MMSeqs2</a> (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02895-z" class="external-link">Barrera-Redondo
et al., 2023</a>). Potential issues such as <em>homology detection
failure</em> can be overcome by providing a pair-wise evolutionary
distance file as an input. Furthermore, users can set thresholds such as
for <code>taxonomic representativeness</code> to filter evolutionarily
inconsistent matches to the database, i.e. matches stemming from
database contamination or horizontal gene transfer.</p>
</details><details><summary><strong>Protein structure-based gene age inference</strong>
</summary><p>Advances in protein structure prediction have enabled the inference
of protein structure across the tree of life in a speedy manner. With
this new data, protein structure aligners such as <a href="https://github.com/steineggerlab/foldseek" class="external-link"><code>FoldSeek</code></a>
can be used instead of sequence aligners for gene age inference.</p>
<p>In the context of gene age inference, sequence-based alignments have
not been extensively studied compared to sequence-based alignments.
However, interested users can check out the <code>FoldSeek</code> option
for <a href="https://github.com/josuebarrera/GenEra" class="external-link">GenEra</a>.</p>
</details><details><summary><strong>Synteny-based gene age inference</strong>
</summary><p>A recently introduced approach is called
<code>synteny-based phylostratigraphy</code> (<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3023-y" class="external-link">Arendsee
et al., 2019</a>). Here, the authors provide a comparative analysis of
genes across evolutionary clades, augmenting standard phylostratigraphy
with a detailed, synteny-based analysis.</p>
<p>Whereas standard phylostratigraphy searches the proteomes of related
species for similarities to focal genes, their <code>fagin</code>
pipeline first finds syntenic genomic intervals and then searches within
these intervals for any trace of similarity.</p>
<p><code>fagin</code> searches the (<code>in silico</code> translated)
amino acid sequence of all unannotated ORFs as well as all known CDS
within the syntenic search space of the target genomes. If no amino acid
similarity is found within the syntenic search space, their
<code>fagin</code> pipeline will search for nucleotide similarity.
Finding nucleotide sequence similarity, but not amino acid similarity,
is consistent with a <code>de novo</code> origin of the focal gene. If
no similarity of any sort is found, their <code>fagin</code> pipeline
will use the syntenic data to infer a possible reason. Thus, they detect
indels, scrambled synteny, assembly issues, and regions of uncertain
synteny (<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3023-y" class="external-link">Arendsee
et al., 2019</a>).</p>
</details><details><summary><strong>Orthogroup-based gene age inference</strong>
</summary><p>The first studies in gene age inference used the framework of genomic
phylostratigraphy, where each gene of the focal species is compared
against a sequence database to find the most distantly related homologue
(<a href="https://doi.org/10.1016/j.tig.2007.08.014" class="external-link">Domazet-Loso et
al., 2007</a>). Some more recent studies have instead inferred gene age
using the node in the phylogeny that contains all members of an
<code>orthogroup</code>.</p>
<p>An orthogroup is a <em>group of genes descended from a single gene in
the last common ancestor of a clade of species</em>. This framework has
been very useful for phylogenetic inference and comparative
transcriptomics. Since an orthogroup is a subset of all sequence
homologues in the given dataset, a gene can hit genes outside of its
orthogroup (using the same parameters on a pairwise sequence
aligner).</p>
<p><a href="https://github.com/davidemms/OrthoFinder" class="external-link">OrthoFinder</a> (a
tool for phylogenetic orthology inference for comparative genomics) is
commonly used in <em>orthogroup</em>-based gene age inference. The
result of OrthoFinder includes the files <code>Orthogroups.tsv</code>
(now <a href="https://github.com/davidemms/OrthoFinder#orthogroups-directory-deprecated" class="external-link">depreciated</a>)
and <code>Phylogenetic_Hierarchical_Orthogroups/N0.tsv</code>, which
have been used for <em>orthogroup</em>-based TAI studies (e.g. <a href="https://www.pnas.org/doi/full/10.1073/pnas.2216351120" class="external-link">Ma &amp;
Zheng 2023</a>; <a href="https://doi.org/10.1101/2023.07.04.546637" class="external-link">Nishimiya-Fujisawa et
al., 2023</a>). Incorporating these results into <code>myTAIv2</code>
can be done using <a href="https://github.com/kullrich/orthomap" class="external-link"><code>orthomap</code></a>
(see <a href="https://orthomap.readthedocs.io/en/latest/" class="external-link">documentation</a>).</p>
<p>Methodologically, it should be stressed that while
<code>Orthogroups.tsv</code> and
<code>Phylogenetic_Hierarchical_Orthogroups/N0.tsv</code> contain
orthogroups with more than one member (i.e. orthogroups with at least
one orthologue or paralogue), users should be aware that these output
files of OrthoFinder do not include singletons, which is a signature of
a novel or species-specific gene. To incorporate singletons, users
should include the genes in <code>Orthogroups_UnassignedGenes.tsv</code>
output or, alternatively, singletons in the <code>Orthogroups.txt</code>
file, and assign the singletons to the species-specific phylostrata in
the study.</p>
<p>The <em>orthogroup</em>-based approach seems to be motivated by an
assumption that the more restricted set of genes that consists of
orthologues and in-paralogues (paralogues that arose after the species
split) tends to have more similar attributes (e.g. conservation in
function and/or expression pattern) than the larger set of genes that
contains all sequence homologues. It follows that this framework should
better reflect the ‘gene age’. This assumption draws parallels to the
<code>orthologue conjecture</code> (i.e. orthologues tend to have more
conserved attributes than paralogues), which is heavily debated (<a href="https://www.nature.com/articles/nrg3456#Sec8" class="external-link">Gabaldon &amp;
Koonin 2013</a>; <a href="https://www.pnas.org/doi/abs/10.1073/pnas.1707515115" class="external-link">Dunn et al.,
2018</a>; <a href="https://pubmed.ncbi.nlm.nih.gov/32657391/" class="external-link">Stamboulian et al.,
2020</a>; etc.). In several cases, orthologues do not have more
conserved attributes than paralogues. Indeed, ‘shifts’ or
‘substitutions’ in the expression profile of paralogues that arose
before the species split are observed in single-cell studies (<a href="https://elifesciences.org/articles/66747" class="external-link">Tarashansky et al.,
2021</a>; <a href="https://www.nature.com/articles/s41559-021-01580-3" class="external-link">Shafer et al.,
2022</a>).</p>
<p>Overall, it remains unclear whether inferring gene age using the
<em>orthogroup</em>-based approach better captures evolutionary
information than the classic genomic phylostratigraphy approach
(all/unrestricted sequence homologues), given that several issues
pertaining to genomic phylostratigraphy have been addressed (see <a href="https://link.springer.com/article/10.1186/s13059-023-02895-z" class="external-link">Barrera-Redondo
et al., 2023</a>).</p>
</details>
</div>
<div class="section level3">
<h3 id="open-questions">Open questions<a class="anchor" aria-label="anchor" href="#open-questions"></a>
</h3>
<p>Regardless of the homology search algorithm choice, there remains
some open issues with gene age inference.</p>
<details><summary><strong>Homology detection failure</strong>
</summary><p>Small and fast-evolving genes are often wrongly annotated as young
genes due to <code>homology detection failure</code>. This issue and
potential solution in the context of gene age inference have been
discussed by <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000862" class="external-link">Weisman
et al., 2020</a>.</p>
<blockquote>
<p>An alternative explanation for a lineage-specific gene is that
nothing particularly special has happened in the gene’s evolutionary
history. The gene has homologs outside of the lineage (no de novo
origination), and no novel function has emerged (no
neofunctionalization), but despite this lack of novelty, computational
similarity searches (e.g., BLAST) have failed to detect the
out-of-lineage homologs. We refer to such unsuccessful searches as
<em>homology detection failure</em>. As homologs diverge in sequence
from one another, the statistical significance of their similarity
declines. <em>Over evolutionary time, with a constant rate of sequence
evolution, the degree of similarity may fall below the chosen
significance threshold, resulting in a failure to detect the
homolog</em>. - Weisman et al., 2020</p>
</blockquote>
<p>Users can run their test, <a href="https://github.com/caraweisman/abSENSE" class="external-link"><code>abSENSE</code></a>
to detect the <em>probability that a homolog of a given gene would fail
to be detected by a homology search</em>. We note that a recently
developed tool <a href="https://github.com/josuebarrera/GenEra" class="external-link"><code>GenEra</code></a>
can automate <code>abSENSE</code> analysis for every gene in a given
genome and provide an output compatible with <code>myTAIv2</code>.</p>
<div class="important">
<p>One drawback of the <code>homology detection failure test</code>
approach is that a large percentage of genes are removed as <em>not</em>
passing the test. This <em>may</em> be due to the conservative nature of
the test and its assumptin regarding protein sequence evolution as
detected via pairwise sequence alignment.</p>
<p>Downstream, the removal of genes can affects on the TAI computation
since less genes are used to compute the TAI. Regardless, users can run
<code><a href="../reference/select_genes.html">myTAI::select_genes()</a></code> with the gene passing from the
<code>homology detection failure test</code> to filter the
<code>BulkPhyloExpressionSet</code> or <code>ScPhyloExpressionSet</code>
object before running <code><a href="../reference/plot_signature.html">myTAI::plot_signature()</a></code>.</p>
</div>
</details><details><summary><strong>Dollo parsimony</strong>
</summary><p><em>Dollo parsimony</em>, which precludes the possibility that an
identical character (gene) can be gained more than once, is a major
conceptual assumption for gene age inference (<a href="https://doi.org/10.1093/gbe/evae062" class="external-link">Galvez-Morante et al.,
2024</a>).</p>
<p>This assumption has been found to overestimate ancestral gene content
(i.e., pushing more genes to be evolutionarily older) on simulated data,
when compared to a maximum likelihood-based approach that allows a gene
family to be gained more than once within a tree (via convergent
evolution and horizontal gene transfer).</p>
<p>A suitable method to overcome this limitation is yet to be developed
for user-friendly gene age inference.</p>
</details><details><summary><strong>Sequence contamination</strong>
</summary><p>As we amass more and more genomic data across the tree of life, we
inevitably accumulate sequences that are taxonomically mislabeled (<a href="https://link.springer.com/article/10.1186/s13059-020-02023-1" class="external-link">Steinegger
&amp; Salzberg 2020</a>). These mislabeled sequences are thought to
typically arise from sequencing contamination and biases gene age
inference by overestimating the real age of the gene (i.e. as
evolutionarily older).</p>
<p>The gene age inference software, <code>GenEra</code>, accounts for
potential contamination to some extent. Upstream software can be
employed to further address potential contamination but these are
typically applied to query genomes and not the database (see <a href="https://www.nature.com/articles/s41467-024-45024-5" class="external-link">Balint et al.,
2024</a> and <a href="https://www.nature.com/articles/s41587-024-02147-w" class="external-link">Nevers et al.,
2025</a>).</p>
<p>To the best of our knowledge, while several bioinformatic solutions
exists for query genome contamination, database sequence contamination
remains a challenge.</p>
</details>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>Despite the ongoing debate about how to correctly infer gene age,
users of <code>myTAIv2</code> can perform any gene age inference method
they find most appropriate for their biological question and pass this
gene age inference table as input to <code>myTAIv2</code>.</p>
<p>To do so, users need to follow the data format specification vignette
to use their gene age inference information with
<code>myTAIv2</code>:<br>
→ <a href="phylo-expression-object.html">📊</a></p>
<p>As these discussions show, even when users rely on established
procedures such as phylostratigraphy, the gene age inference bias will
be present as ‘systematic error’ in all developmental stages for which
TAI computations are performed. Thus, <em>stages of constraint (or vice
versa) will be detectable in any case.</em></p>
<p>Since TAI computation is intended to enable screening for conserved
or constrained stages in developmental or biological processes for
further downstream experimental studies, even simple approaches such as
phylostratigraphy can give first evidence for the existence of
transcriptomic constraints within a biological process.</p>
<p>Indeed, it has been observed that differences in gene age inference
methods (e.g. <a href="https://www.nature.com/articles/nature11394" class="external-link">blastp e-value
parameter</a>, or <a href="https://www.nature.com/articles/s41467-024-47017-w" class="external-link">mmseqs2
clustering coverage parameter</a>) does not influence the overall TAI or
other macro-evolutionary trends. We nevertheless encourage users to test
different methods (and parameters) as described in this vignette and
consider their strengths/limitation.</p>
<p>For other methods that can assign an evolutionarily-informed or
expression-informed index for each gene (analogous to TAI) such as TDI,
TSI, etc., see here:<br>
→ <a href="other-strata.html">🧬</a></p>
<div class="important">
<p>Interested instead in studying a small set of genes and date their
evolutionary age? We recommend more manual workflows that take multiple
contexts (e.g. domain information, 3D protein structure and genomic
synteny) into account. This vignette describes high throughput
bioinformatic methods that scales to all genes in a given organism.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Stefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
