<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Statistical testing with myTAI • myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Statistical testing with myTAI">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.3.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-stats.html">Statistical testing with myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-transform.html">Transforming dataset for myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/index.html">... and view more!</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Statistical testing with myTAI</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/vignettes/tai-stats.Rmd" class="external-link"><code>vignettes/tai-stats.Rmd</code></a></small>
      <div class="d-none name"><code>tai-stats.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="the-motivation-for-statistically-informed-analysis-of-tai-patterns">The motivation for statistically-informed analysis of TAI
patterns<a class="anchor" aria-label="anchor" href="#the-motivation-for-statistically-informed-analysis-of-tai-patterns"></a>
</h2>
<p>Let’s say you have your transcriptome age index (<a href="phylostratigraphy.html#gene-age-information-in-tai">TAI</a>)
pattern, e.g.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drostlab.github.io/myTAI/">myTAI</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_phyex_set_old"</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_signature.html">plot_signature</a></span><span class="op">(</span><span class="va">example_phyex_set_old</span>, show_p_val <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-stats_files/figure-html/unnamed-chunk-2-1.png" alt="plot_signature function output" width="576" style="display: block; margin: auto;"></p>
<p>But how can you test if this pattern is significant?</p>
</div>
<div class="section level2">
<h2 id="statistical-analysis-of-tai-patterns">Statistical analysis of TAI patterns<a class="anchor" aria-label="anchor" href="#statistical-analysis-of-tai-patterns"></a>
</h2>
<p>With <code>myTAIv2</code>, we have introduced a suite of
permutation-based statistical testing framework to test the significance
of your TAI (as well as TDI, TSI and analogous) profiles.</p>
<p>In this section, we will go through:<br>
(1) <code><a href="../reference/stat_flatline_test.html">myTAI::stat_flatline_test()</a></code> which evaluates whether the
observed TAI profile is significantly different from a flatline (i.e. no
evolutionary trend).<br>
(2) <code><a href="../reference/stat_reductive_hourglass_test.html">myTAI::stat_reductive_hourglass_test()</a></code> which evaluates
whether the observed TAI profile is significantly different from an
hourglass pattern (i.e. consistent with the molecular hourglass
hypothesis), as well as the reverse
<code><a href="../reference/stat_reverse_hourglass_test.html">myTAI::stat_reverse_hourglass_test()</a></code> and tests of potential
other evolutionary patterns such as
<code><a href="../reference/stat_early_conservation_test.html">myTAI::stat_early_conservation_test()</a></code> and
<code><a href="../reference/stat_late_conservation_test.html">myTAI::stat_late_conservation_test()</a></code>.<br>
(3) <code><a href="../reference/stat_pairwise_test.html">myTAI::stat_pairwise_test()</a></code> which evaluates whether the
TAI profiles of two groups of samples are significantly different from
each other.</p>
<div class="note">
<p>Unlike <code><a href="../reference/stat_flatline_test.html">stat_flatline_test()</a></code>, the other tests
(<code><a href="../reference/stat_reductive_hourglass_test.html">stat_reductive_hourglass_test()</a></code>,
<code><a href="../reference/stat_reverse_hourglass_test.html">stat_reverse_hourglass_test()</a></code>,
<code><a href="../reference/stat_early_conservation_test.html">stat_early_conservation_test()</a></code>,
<code><a href="../reference/stat_late_conservation_test.html">stat_late_conservation_test()</a></code>, and
<code><a href="../reference/stat_pairwise_test.html">stat_pairwise_test()</a></code>) require an <em>a priori</em> grouping
of developmental stage or experimental conditions.</p>
</div>
</div>
<div class="section level2">
<h2 id="flatline-test">Flatline test<a class="anchor" aria-label="anchor" href="#flatline-test"></a>
</h2>
<p>The flat line test tests the variance of the TAI (or equivalent)
profile against a null distribution.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_flatline_test.html">stat_flatline_test</a></span><span class="op">(</span><span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><pre><code><span><span class="co">## Computing: [========================================] 100% (done)</span></span></code></pre>
<p><img src="tai-stats_files/figure-html/unnamed-chunk-4-1.png" alt="stat_flatline_test function output" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Statistical Test Result</span></span>
<span><span class="co">## =======================</span></span>
<span><span class="co">## Method: Flat Line Test </span></span>
<span><span class="co">## Test statistic: 0.02864928 </span></span>
<span><span class="co">## P-value: 7.063146e-07 </span></span>
<span><span class="co">## Alternative hypothesis: greater </span></span>
<span><span class="co">## Data: Embryogenesis 2011</span></span></code></pre>
</details><p>We can see a histogram of the <strong>null</strong> distribution
(along with the fitted null distribution) and the
<strong>observed</strong> variance. In text, we can see the summary test
statistics of the flat line test. The plots are created by default but
you can also turn <code>plot_result = FALSE</code> to not plot the
results of the test.</p>
<p>In this example, we can see that the pattern significantly deviates
from a flat line (i.e. no variance).</p>
<div class="tip">
<p>The <code><a href="../reference/stat_flatline_test.html">myTAI::stat_flatline_test()</a></code> is a good first step to
evaluating the evolutionary signals in your transcriptomic data.</p>
</div>
<div class="section level3">
<h3 id="cullen-and-frey-plot">Cullen and Frey plot<a class="anchor" aria-label="anchor" href="#cullen-and-frey-plot"></a>
</h3>
<p>To further inspect the permutation test, we can use the Cullen and
Frey plot (skewness vs. kurtosis;
<code><a href="../reference/plot_cullen_frey.html">myTAI::plot_cullen_frey()</a></code>) to help identify appropriate
distribution families for the null sample data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_flt</span> <span class="op">&lt;-</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_flatline_test.html">stat_flatline_test</a></span><span class="op">(</span><span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_cullen_frey.html">plot_cullen_frey</a></span><span class="op">(</span><span class="va">res_flt</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_cullen_frey.html">plot_cullen_frey</a></span><span class="op">(</span><span class="va">res_flt</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-stats_files/figure-html/unnamed-chunk-7-1.png" alt="plot_cullen_frey function output for stat_flatline_test" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## summary statistics</span></span>
<span><span class="co">## ------</span></span>
<span><span class="co">## min:  2.492963e-05   max:  0.0311538 </span></span>
<span><span class="co">## median:  0.002239929 </span></span>
<span><span class="co">## mean:  0.002976196 </span></span>
<span><span class="co">## estimated sd:  0.002606176 </span></span>
<span><span class="co">## estimated skewness:  2.376258 </span></span>
<span><span class="co">## estimated kurtosis:  13.62014</span></span></code></pre>
</details><p>We can see that the null sample lies within the gamma distribution
family, which is a good sign because the
<code><a href="../reference/stat_flatline_test.html">myTAI::stat_flatline_test()</a></code> assumes that the null sample is
gamma-distributed.</p>
<div class="note">
<p>The <code><a href="../reference/plot_cullen_frey.html">myTAI::plot_cullen_frey()</a></code> function is also useful
for the other tests as well
(e.g. <code><a href="../reference/stat_reductive_hourglass_test.html">stat_reductive_hourglass_test()</a></code>,
<code><a href="../reference/stat_reverse_hourglass_test.html">stat_reverse_hourglass_test()</a></code>,
<code><a href="../reference/stat_early_conservation_test.html">stat_early_conservation_test()</a></code>,
<code><a href="../reference/stat_late_conservation_test.html">stat_late_conservation_test()</a></code>, and
<code><a href="../reference/stat_pairwise_test.html">stat_pairwise_test()</a></code>).</p>
</div>
</div>
<div class="section level3">
<h3 id="null-distribution-visualisation">Null distribution visualisation<a class="anchor" aria-label="anchor" href="#null-distribution-visualisation"></a>
</h3>
<p>Furthermore, using the results of the flat line test (saved here as
<code>res_flt</code>), we can visualise TAI pattern of each of the
permutations using <code><a href="../reference/plot_null_txi_sample.html">myTAI::plot_null_txi_sample()</a></code>, i.e.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_null_txi_sample.html">plot_null_txi_sample</a></span><span class="op">(</span><span class="va">res_flt</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_null_txi_sample.html">plot_null_txi_sample</a></span><span class="op">(</span><span class="va">res_flt</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-stats_files/figure-html/unnamed-chunk-9-1.png" alt="plot_null_txi_sample function output" width="576" style="display: block; margin: auto;"></p>
<p>The null variance distribution is produced from the grey lines, while
the red line is the observed TAI profile. The distance between the red
and grey lines indicate that older genes are generally higher expressed
than younger genes (as they contribute more to the TAI), and the
observed TAI profile is more variable than the null distribution.</p>
</details>
</div>
</div>
<div class="section level2">
<h2 id="reductive-hourglass-test-and-other-time-course-tests">Reductive hourglass test and other time-course tests<a class="anchor" aria-label="anchor" href="#reductive-hourglass-test-and-other-time-course-tests"></a>
</h2>
<p>The following tests (<code><a href="../reference/stat_reductive_hourglass_test.html">stat_reductive_hourglass_test()</a></code>,
<code><a href="../reference/stat_reverse_hourglass_test.html">stat_reverse_hourglass_test()</a></code>,
<code><a href="../reference/stat_early_conservation_test.html">stat_early_conservation_test()</a></code> and
<code><a href="../reference/stat_late_conservation_test.html">stat_late_conservation_test()</a></code>) evaluates changes in
evolutionary signal across developmental stages. Rather than testing the
<strong>variance</strong> of the TAI profile against a null
distribution, these tests evaluate whether the observed TAI profile is
significantly different from the null to support an hourglass pattern
(i.e. consistent with the molecular hourglass hypothesis), a reverse
hourglass pattern, or an early/late conservation pattern.</p>
<p>We also need to specify <em>a priori</em> grouping of developmental
stages or experimental conditions. For example, we can group the
developmental stages into: early (stages 1-3), mid (stages 4-5) and late
(stages 6-7). This is specified via the parameter
<code>modules</code>.</p>
<div class="section level3">
<h3 id="reductive-hourglass-test">Reductive hourglass test<a class="anchor" aria-label="anchor" href="#reductive-hourglass-test"></a>
</h3>
<p>The reductive hourglass test evaluates whether the observed TAI
profile significantly deviates from the null towards an
<code>hourglass</code> pattern, i.e. the TAI values are lower in the
mid-stage compared to the early and late stages. This can be achieved
with the function
<code><a href="../reference/stat_reductive_hourglass_test.html">myTAI::stat_reductive_hourglass_test()</a></code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, mid <span class="op">=</span> <span class="fl">4</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_reductive_hourglass_test.html">stat_reductive_hourglass_test</a></span><span class="op">(</span></span>
<span>  <span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  modules <span class="op">=</span> <span class="va">modules</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><p><img src="tai-stats_files/figure-html/unnamed-chunk-11-1.png" alt="stat_reductive_hourglass_test function output" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Statistical Test Result</span></span>
<span><span class="co">## =======================</span></span>
<span><span class="co">## Method: Reductive Hourglass Test </span></span>
<span><span class="co">## Test statistic: 0.1218678 </span></span>
<span><span class="co">## P-value: 0.001219147 </span></span>
<span><span class="co">## Alternative hypothesis: greater </span></span>
<span><span class="co">## Data: Embryogenesis 2011</span></span></code></pre>
</details><p>In this example, we can see that the TAI pattern is consistent with
the molecular hourglass model, as the observed TAI profile is
significantly different from the null distribution.</p>
<details><summary>
Cullen and Frey plot
</summary><p>We can also check the Cullen and Frey plot (though I have put this
under a details tag to not clutter the output) to see if the null sample
is normally-distributed, which the assumption for the
<code><a href="../reference/stat_reductive_hourglass_test.html">myTAI::stat_reductive_hourglass_test()</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_flt</span> <span class="op">&lt;-</span> <span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_reductive_hourglass_test.html">stat_reductive_hourglass_test</a></span><span class="op">(</span></span>
<span>  <span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  modules <span class="op">=</span> <span class="va">modules</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_cullen_frey.html">plot_cullen_frey</a></span><span class="op">(</span><span class="va">res_flt</span><span class="op">)</span></span></code></pre></div>
<p><img src="tai-stats_files/figure-html/unnamed-chunk-13-1.png" alt="plot_cullen_frey function output for stat_reductive_hourglass_test" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## summary statistics</span></span>
<span><span class="co">## ------</span></span>
<span><span class="co">## min:  -0.2743518   max:  0.1137399 </span></span>
<span><span class="co">## median:  -0.02584063 </span></span>
<span><span class="co">## mean:  -0.02991773 </span></span>
<span><span class="co">## estimated sd:  0.05008444 </span></span>
<span><span class="co">## estimated skewness:  -0.3800859 </span></span>
<span><span class="co">## estimated kurtosis:  3.536079</span></span></code></pre>
<p>Indeed it is!</p>
</details>
</div>
<div class="section level3">
<h3 id="reverse-hourglass-test">Reverse hourglass test<a class="anchor" aria-label="anchor" href="#reverse-hourglass-test"></a>
</h3>
<p>Analogously, we can also test for whether the observed patterns are
consistent with an inverse or <code>reverse hourglass</code> pattern,
i.e. the TAI (as well as TDI, TSI and analogous) values tend to be
higher in the mid-stage. This can be achieved with the function
<code><a href="../reference/stat_reverse_hourglass_test.html">myTAI::stat_reverse_hourglass_test()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, mid <span class="op">=</span> <span class="fl">4</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_reverse_hourglass_test.html">stat_reverse_hourglass_test</a></span><span class="op">(</span></span>
<span>  <span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  modules <span class="op">=</span> <span class="va">modules</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><p><img src="tai-stats_files/figure-html/unnamed-chunk-15-1.png" alt="stat_reverse_hourglass_test function output" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Statistical Test Result</span></span>
<span><span class="co">## =======================</span></span>
<span><span class="co">## Method: Reverse Hourglass Test </span></span>
<span><span class="co">## Test statistic: -0.2726821 </span></span>
<span><span class="co">## P-value: 0.9999987 </span></span>
<span><span class="co">## Alternative hypothesis: greater </span></span>
<span><span class="co">## Data: Embryogenesis 2011</span></span></code></pre>
</details><p>As we can see from the p-values of the one-tailed test, the TAI
profile does not support the reverse hourglass pattern, i.e. the TAI
values are not higher in the mid-stage. This is not surprisingly given
the results of the
<code><a href="../reference/stat_reductive_hourglass_test.html">myTAI::stat_reductive_hourglass_test()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="early-conservation-test">Early conservation test<a class="anchor" aria-label="anchor" href="#early-conservation-test"></a>
</h3>
<p>Another potential pattern of so-called “ontogeny-phylogeny”
correlation is <code>early conservation</code>, which has a long history
in evolutionary developmental biology. This pattern suggests that early
developmental stages are more conserved across species, while later
stages show more variability.</p>
<p>To test for the early conservation pattern, we can use the function
<code><a href="../reference/stat_early_conservation_test.html">myTAI::stat_early_conservation_test()</a></code>. This test evaluates
whether the TAI profile is significantly lower in the early stages.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, mid <span class="op">=</span> <span class="fl">4</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_early_conservation_test.html">stat_early_conservation_test</a></span><span class="op">(</span></span>
<span>  <span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  modules <span class="op">=</span> <span class="va">modules</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><p><img src="tai-stats_files/figure-html/unnamed-chunk-17-1.png" alt="stat_early_conservation_test function output" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Statistical Test Result</span></span>
<span><span class="co">## =======================</span></span>
<span><span class="co">## Method: Early Conservation Test </span></span>
<span><span class="co">## Test statistic: -0.1218678 </span></span>
<span><span class="co">## P-value: 0.9540384 </span></span>
<span><span class="co">## Alternative hypothesis: greater </span></span>
<span><span class="co">## Data: Embryogenesis 2011</span></span></code></pre>
</details><p>Again, the TAI profile does not support the early conservation
pattern, as the p-value is not significant. TAI values are not
significantly lower in the early stages compared to the mid and late
stages.</p>
</div>
<div class="section level3">
<h3 id="late-conservation-test">Late conservation test<a class="anchor" aria-label="anchor" href="#late-conservation-test"></a>
</h3>
<p>Another potential pattern is <code>late conservation</code>, which
suggests that late developmental stages are more conserved across
species compared to early stages. To test for the late conservation
pattern, we can use the function
<code><a href="../reference/stat_late_conservation_test.html">myTAI::stat_late_conservation_test()</a></code>, i.e.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, mid <span class="op">=</span> <span class="fl">4</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_late_conservation_test.html">stat_late_conservation_test</a></span><span class="op">(</span></span>
<span>  <span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  modules <span class="op">=</span> <span class="va">modules</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><p><img src="tai-stats_files/figure-html/unnamed-chunk-19-1.png" alt="stat_late_conservation_test function output" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Statistical Test Result</span></span>
<span><span class="co">## =======================</span></span>
<span><span class="co">## Method: Late Conservation Test </span></span>
<span><span class="co">## Test statistic: -0.2726821 </span></span>
<span><span class="co">## P-value: 0.999778 </span></span>
<span><span class="co">## Alternative hypothesis: greater </span></span>
<span><span class="co">## Data: Embryogenesis 2011</span></span></code></pre>
</details><p>As expected again, we do not see support for the late conservation
pattern, as the TAI values are not significantly lower in the late
stages compared to the early and mid stages.</p>
</div>
</div>
<div class="section level2">
<h2 id="pairwise-test">Pairwise test<a class="anchor" aria-label="anchor" href="#pairwise-test"></a>
</h2>
<p>Finally, we can also test whether the TAI profiles of two groups of
samples are significantly different from each other. This can be
achieved with the function <code><a href="../reference/stat_pairwise_test.html">myTAI::stat_pairwise_test()</a></code>.</p>
<p>For example, let’s say we have an experiment where we disrupt an
organisms’ development at a given stage using a certain perturbation
(e.g. drug treatment). We can compare the TAI profiles of the control
and treatment groups to see if the perturbation has an effect on the TAI
profile. This could be interesting if the perturbed pathway is thought
to recruit orphan genes, i.e. genes that are not conserved across
species. An example can be found <a href="https://doi.org/10.1101/2024.05.28.596352" class="external-link">here</a>.</p>
<p>With <code>myTAIv2</code>, we can facilitate such test. Since we do
not have the data, we will use the example data set
<code>example_phyex_set_old</code> and create two groups of samples:
<code>control</code> and <code>treatment</code>. We can then compare the
TAI profiles of these two groups using the
<code><a href="../reference/stat_pairwise_test.html">myTAI::stat_pairwise_test()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>contrast1 <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, contrast2 <span class="op">=</span> <span class="fl">5</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></span>
<span><span class="fu">myTAI</span><span class="fu">::</span><span class="fu"><a href="../reference/stat_pairwise_test.html">stat_pairwise_test</a></span><span class="op">(</span></span>
<span>  <span class="va">example_phyex_set_old</span>, plot_result <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  modules <span class="op">=</span> <span class="va">modules</span>,</span>
<span>  alternative <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
Show output
</summary><p><img src="tai-stats_files/figure-html/unnamed-chunk-21-1.png" alt="stat_pairwise_test function output" width="480" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Statistical Test Result</span></span>
<span><span class="co">## =======================</span></span>
<span><span class="co">## Method: Pairwise Test </span></span>
<span><span class="co">## Test statistic: -0.07232483 </span></span>
<span><span class="co">## P-value: 0.8890747 </span></span>
<span><span class="co">## Alternative hypothesis: greater </span></span>
<span><span class="co">## Data: Embryogenesis 2011</span></span></code></pre>
</details><p>In this mock example, we can see that the TAI profile of the control
group (<code>contrast1</code>) is not significantly higher than the
treatment group (<code>contrast2</code>), indicating that the
perturbation has no effect on the TAI profile. The p-value is
significant, and the observed TAI profile is significantly different
from the null distribution.</p>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>In this section, we have introduced the statistical testing framework
of <code>myTAIv2</code> to evaluate the significance of TAI patterns.
With this, we can address the question “is my TAI pattern significant?”
in a statistically-informed manner.</p>
<p>We have also introduced the Cullen and Frey plot to help identify
appropriate distribution families for the null sample data, as well as
the null distribution visualisation to inspect the permutation test
results.</p>
<div class="note">
<p>It should be noted, however, that the results of the tests may be
sensitive to the grouping of developmental stages or experimental
conditions. This <em>a priori</em> information should be handled with
care.</p>
<p>Moreover, RNA-seq data transformation (e.g. log2 transformation) can
also affect the results of the tests and the overall TAI profile. Given
that we do not have a gold standard transformation, it is recommended to
try different transformations (using <code><a href="../reference/tf.html">myTAI::tf()</a></code>) and
evaluate the <code>robustness</code> of the results. For more
information, check out this <a href="tai-transform.html">article</a>.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Stefan Manolache, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
