<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Package index"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.3.0.9006</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/myTAI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/other-strata.html">Other evolutionary and expression indices</a>
    </li>
    <li>
      <a href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a>
    </li>
    <li>
      <a href="../articles/phylostratigraphy.html">Gene age inference</a>
    </li>
    <li>
      <a href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a>
    </li>
    <li>
      <a href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-stats.html">Statistical testing with myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-transform.html">Transforming dataset for myTAI</a>
    </li>
    <li>
      <a href="../articles/taxonomy.html">Obtaining taxonomy information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="BulkPhyloExpressionSet.html">BulkPhyloExpressionSet()</a></code> </p>
        </td>
        <td><p>Bulk PhyloExpressionSet Class</p></td>
      </tr><tr><td>
          <p><code><a href="BulkPhyloExpressionSet_from_df.html">BulkPhyloExpressionSet_from_df()</a></code> </p>
        </td>
        <td><p>Convert Data to BulkPhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="COUNT_TRANSFORMS.html">COUNT_TRANSFORMS</a></code> </p>
        </td>
        <td><p>Count Transformation Functions</p></td>
      </tr><tr><td>
          <p><code><a href="ConservationTestResult.html">ConservationTestResult()</a></code> </p>
        </td>
        <td><p>Conservation Test Result S7 Class</p></td>
      </tr><tr><td>
          <p><code><a href="Distribution.html">Distribution()</a></code> </p>
        </td>
        <td><p>Distribution S7 Class</p></td>
      </tr><tr><td>
          <p><code><a href="PS_colours.html">PS_colours()</a></code> </p>
        </td>
        <td><p>Generate Phylostratum Colors</p></td>
      </tr><tr><td>
          <p><code><a href="PhyloExpressionSetBase.html">PhyloExpressionSetBase()</a></code> </p>
        </td>
        <td><p>PhyloExpressionSet Base Class</p></td>
      </tr><tr><td>
          <p><code><a href="ScPhyloExpressionSet.html">ScPhyloExpressionSet()</a></code> </p>
        </td>
        <td><p>Single-Cell PhyloExpressionSet Class</p></td>
      </tr><tr><td>
          <p><code><a href="ScPhyloExpressionSet_from_matrix.html">ScPhyloExpressionSet_from_matrix()</a></code> </p>
        </td>
        <td><p>Create Single-Cell PhyloExpressionSet from Expression Matrix</p></td>
      </tr><tr><td>
          <p><code><a href="ScPhyloExpressionSet_from_seurat.html">ScPhyloExpressionSet_from_seurat()</a></code> </p>
        </td>
        <td><p>Convert Seurat Object to Single-Cell PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="TAI.html">TAI()</a></code> </p>
        </td>
        <td><p>Calculate Transcriptomic Age Index (TAI)</p></td>
      </tr><tr><td>
          <p><code><a href="TDI.html">TDI()</a></code> </p>
        </td>
        <td><p>Calculate Transcriptomic Divergence Index (TDI)</p></td>
      </tr><tr><td>
          <p><code><a href="TEI.html">TEI()</a></code> </p>
        </td>
        <td><p>Calculate Transcriptomic Evolutionary Index (TEI)</p></td>
      </tr><tr><td>
          <p><code><a href="TI_map.html">TI_map</a></code> </p>
        </td>
        <td><p>Transcriptomic Index Name Mapping</p></td>
      </tr><tr><td>
          <p><code><a href="TPI.html">TPI()</a></code> </p>
        </td>
        <td><p>Calculate Transcriptomic Polymorphism Index (TPI)</p></td>
      </tr><tr><td>
          <p><code><a href="TXI.html">TXI()</a></code> </p>
        </td>
        <td><p>Calculate Transcriptomic Index (TXI)</p></td>
      </tr><tr><td>
          <p><code><a href="TXI_conf_int.html">TXI_conf_int()</a></code> </p>
        </td>
        <td><p>Confidence Intervals for Transcriptomic Index (TXI)</p></td>
      </tr><tr><td>
          <p><code><a href="TXI_std_dev.html">TXI_std_dev()</a></code> </p>
        </td>
        <td><p>Standard Deviation for TXI</p></td>
      </tr><tr><td>
          <p><code><a href="TestResult.html">TestResult()</a></code> </p>
        </td>
        <td><p>Test Result S7 Class</p></td>
      </tr><tr><td>
          <p><code><a href="age.apply.html">age.apply()</a></code> </p>
        </td>
        <td><p>Age Category Specific apply Function</p></td>
      </tr><tr><td>
          <p><code><a href="as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code> </p>
        </td>
        <td><p>as_BulkPhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="as_data_frame.html">as_data_frame()</a></code> </p>
        </td>
        <td><p>Convert BulkPhyloExpressionSet to Data Frame</p></td>
      </tr><tr><td>
          <p><code><a href="check_BulkPhyloExpressionSet.html">check_BulkPhyloExpressionSet()</a></code> </p>
        </td>
        <td><p>Check if object is a BulkPhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="check_PhyloExpressionSet.html">check_PhyloExpressionSet()</a></code> </p>
        </td>
        <td><p>Check if object is a PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="check_ScPhyloExpressionSet.html">check_ScPhyloExpressionSet()</a></code> </p>
        </td>
        <td><p>Check if object is a ScPhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="collapse.html">collapse()</a></code> </p>
        </td>
        <td><p>Collapse PhyloExpressionSet Replicates</p></td>
      </tr><tr><td>
          <p><code><a href="destroy_pattern.html">destroy_pattern()</a></code> </p>
        </td>
        <td><p>Destroy Phylotranscriptomic Pattern Using GATAI</p></td>
      </tr><tr><td>
          <p><code><a href="diagnose_test_robustness.html">diagnose_test_robustness()</a></code> </p>
        </td>
        <td><p>Diagnose Test Robustness</p></td>
      </tr><tr><td>
          <p><code><a href="distributions.html">distributions</a></code> </p>
        </td>
        <td><p>Predefined Distribution Objects</p></td>
      </tr><tr><td>
          <p><code><a href="downsample.html">downsample()</a></code> </p>
        </td>
        <td><p>Downsample ScPhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="downsample_expression.html">downsample_expression()</a></code> </p>
        </td>
        <td><p>Downsample Expression Matrix by Groups</p></td>
      </tr><tr><td>
          <p><code><a href="example_phyex_set.html">example_phyex_set</a></code> </p>
        </td>
        <td><p>Example phyex set</p></td>
      </tr><tr><td>
          <p><code><a href="example_phyex_set_old.html">example_phyex_set_old</a></code> </p>
        </td>
        <td><p>Example phyex set old</p></td>
      </tr><tr><td>
          <p><code><a href="example_phyex_set_sc.html">example_phyex_set_sc</a></code> </p>
        </td>
        <td><p>Load Example Single-Cell PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="exp_p.html">exp_p()</a></code> </p>
        </td>
        <td><p>Format P-Value for Scientific Notation</p></td>
      </tr><tr><td>
          <p><code><a href="genes_lowly_expressed.html">genes_lowly_expressed()</a></code> </p>
        </td>
        <td><p>Select Lowly Expressed Genes</p></td>
      </tr><tr><td>
          <p><code><a href="genes_top_expr.html">genes_top_expr()</a></code> </p>
        </td>
        <td><p>Gene Expression Filtering Functions</p></td>
      </tr><tr><td>
          <p><code><a href="genes_top_mean.html">genes_top_mean()</a></code> </p>
        </td>
        <td><p>Select Top Mean Expressed Genes</p></td>
      </tr><tr><td>
          <p><code><a href="genes_top_variance.html">genes_top_variance()</a></code> </p>
        </td>
        <td><p>Select Top Variable Genes</p></td>
      </tr><tr><td>
          <p><code><a href="match_map.html">match_map()</a></code> </p>
        </td>
        <td><p>Match Gene Expression Data with Phylostratum Map</p></td>
      </tr><tr><td>
          <p><code><a href="match_map_sc_matrix.html">match_map_sc_matrix()</a></code> </p>
        </td>
        <td><p>Match Expression Matrix with Phylostratum Map</p></td>
      </tr><tr><td>
          <p><code><a href="match_map_sc_seurat.html">match_map_sc_seurat()</a></code> </p>
        </td>
        <td><p>Match Single-Cell Expression Data with Phylostratum Map (Seurat)</p></td>
      </tr><tr><td>
          <p><code><a href="normalise_stage_expression.html">normalise_stage_expression()</a></code> </p>
        </td>
        <td><p>Normalise Stage Expression Data</p></td>
      </tr><tr><td>
          <p><code><a href="omit_matrix.html">omit_matrix()</a></code> </p>
        </td>
        <td><p>Compute TXI Profiles Omitting Each Gene</p></td>
      </tr><tr><td>
          <p><code><a href="pTXI.html">pTXI()</a></code> </p>
        </td>
        <td><p>Calculate Phylostratum-Specific Transcriptomic Index</p></td>
      </tr><tr><td>
          <p><code><a href="permute_PS.html">permute_PS()</a></code> </p>
        </td>
        <td><p>Permute Strata in PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="plot_contribution.html">plot_contribution()</a></code> </p>
        </td>
        <td><p>Plot Phylostratum Contribution to Transcriptomic Index</p></td>
      </tr><tr><td>
          <p><code><a href="plot_cullen_frey.html">plot_cullen_frey()</a></code> </p>
        </td>
        <td><p>Plot Cullen-Frey Diagram for Distribution Assessment</p></td>
      </tr><tr><td>
          <p><code><a href="plot_distribution_expression.html">plot_distribution_expression()</a></code> </p>
        </td>
        <td><p>Comparing expression levels distributions across developmental stages</p></td>
      </tr><tr><td>
          <p><code><a href="plot_distribution_pTAI.html">plot_distribution_pTAI()</a></code> </p>
        </td>
        <td><p>Partial TAI Distribution Plotting Functions</p></td>
      </tr><tr><td>
          <p><code><a href="plot_distribution_pTAI_qqplot.html">plot_distribution_pTAI_qqplot()</a></code> </p>
        </td>
        <td><p>QQ plot comparing partial TAI distributions across developmental stages against a reference stage</p></td>
      </tr><tr><td>
          <p><code><a href="plot_distribution_strata.html">plot_distribution_strata()</a></code> </p>
        </td>
        <td><p>Plot Distribution of Genes Across Phylostrata</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gatai_results.html">plot_gatai_results()</a></code> </p>
        </td>
        <td><p>Plot Comprehensive GATAI Results</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gene_heatmap.html">plot_gene_heatmap()</a></code> </p>
        </td>
        <td><p>Plot Gene Expression Heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gene_profiles.html">plot_gene_profiles()</a></code> </p>
        </td>
        <td><p>Plot Individual Gene Expression Profiles</p></td>
      </tr><tr><td>
          <p><code><a href="plot_gene_space.html">plot_gene_space()</a></code> </p>
        </td>
        <td><p>Plot Gene Space Using PCA</p></td>
      </tr><tr><td>
          <p><code><a href="plot_mean_var.html">plot_mean_var()</a></code> </p>
        </td>
        <td><p>Plot Mean-Variance Relationship</p></td>
      </tr><tr><td>
          <p><code><a href="plot_null_txi_sample.html">plot_null_txi_sample()</a></code> </p>
        </td>
        <td><p>Plot Null TXI Sample Distribution</p></td>
      </tr><tr><td>
          <p><code><a href="plot_relative_expression_bar.html">plot_relative_expression_bar()</a></code> </p>
        </td>
        <td><p>Plot Mean Relative Expression Levels as Barplot</p></td>
      </tr><tr><td>
          <p><code><a href="plot_relative_expression_line.html">plot_relative_expression_line()</a></code> </p>
        </td>
        <td><p>Plot Relative Expression Profiles (Line Plot)</p></td>
      </tr><tr><td>
          <p><code><a href="plot_sample_space.html">plot_sample_space()</a></code> </p>
        </td>
        <td><p>Plot Sample Space Visualization</p></td>
      </tr><tr><td>
          <p><code><a href="plot_signature.html">plot_signature()</a></code> </p>
        </td>
        <td><p>Plot Transcriptomic Signature</p></td>
      </tr><tr><td>
          <p><code><a href="plot_signature_gene_quantiles.html">plot_signature_gene_quantiles()</a></code> </p>
        </td>
        <td><p>Plot Signature Across Gene Expression Quantiles</p></td>
      </tr><tr><td>
          <p><code><a href="plot_signature_multiple.html">plot_signature_multiple()</a></code> </p>
        </td>
        <td><p>Plot Multiple Transcriptomic Signatures</p></td>
      </tr><tr><td>
          <p><code><a href="plot_signature_transformed.html">plot_signature_transformed()</a></code> </p>
        </td>
        <td><p>Plot Signature Under Different Transformations</p></td>
      </tr><tr><td>
          <p><code><a href="plot_strata_expression.html">plot_strata_expression()</a></code> </p>
        </td>
        <td><p>Plot Expression Levels by Phylostratum</p></td>
      </tr><tr><td>
          <p><code><a href="quantile_rank.html">quantile_rank()</a></code> </p>
        </td>
        <td><p>Calculate Quantile Ranks</p></td>
      </tr><tr><td>
          <p><code><a href="rel_exp_matrix.html">rel_exp_matrix()</a></code> </p>
        </td>
        <td><p>Compute Relative Expression Matrix for PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="relative_expression.html">relative_expression()</a></code> </p>
        </td>
        <td><p>Relative Expression Functions</p></td>
      </tr><tr><td>
          <p><code><a href="remove_genes.html">remove_genes()</a></code> </p>
        </td>
        <td><p>Remove Genes from PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="rename_phyex_set.html">rename_phyex_set()</a></code> </p>
        </td>
        <td><p>Rename a PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="sTXI.html">sTXI()</a></code> </p>
        </td>
        <td><p>Calculate Stratum-Specific Transcriptomic Index</p></td>
      </tr><tr><td>
          <p><code><a href="save_gatai_results_pdf.html">save_gatai_results_pdf()</a></code> </p>
        </td>
        <td><p>Save GATAI Analysis Results to PDF</p></td>
      </tr><tr><td>
          <p><code><a href="select_genes.html">select_genes()</a></code> </p>
        </td>
        <td><p>Select Genes from PhyloExpressionSet</p></td>
      </tr><tr><td>
          <p><code><a href="set_expression.html">set_expression()</a></code> </p>
        </td>
        <td><p>Gene Expression Transformation Functions</p></td>
      </tr><tr><td>
          <p><code><a href="stat_early_conservation_test.html">stat_early_conservation_test()</a></code> </p>
        </td>
        <td><p>Early Conservation Test</p></td>
      </tr><tr><td>
          <p><code><a href="stat_flatline_test.html">stat_flatline_test()</a></code> </p>
        </td>
        <td><p>Flat Line Test for Conservation Pattern</p></td>
      </tr><tr><td>
          <p><code><a href="stat_generic_conservation_test.html">stat_generic_conservation_test()</a></code> </p>
        </td>
        <td><p>Generic Conservation Test Framework</p></td>
      </tr><tr><td>
          <p><code><a href="stat_late_conservation_test.html">stat_late_conservation_test()</a></code> </p>
        </td>
        <td><p>Late Conservation Test</p></td>
      </tr><tr><td>
          <p><code><a href="stat_pairwise_test.html">stat_pairwise_test()</a></code> </p>
        </td>
        <td><p>Pairwise Conservation Test</p></td>
      </tr><tr><td>
          <p><code><a href="stat_reductive_hourglass_test.html">stat_reductive_hourglass_test()</a></code> </p>
        </td>
        <td><p>Reductive Hourglass Test</p></td>
      </tr><tr><td>
          <p><code><a href="stat_reverse_hourglass_test.html">stat_reverse_hourglass_test()</a></code> </p>
        </td>
        <td><p>Reverse Hourglass Test</p></td>
      </tr><tr><td>
          <p><code><a href="strata_enrichment.html">strata_enrichment()</a></code> </p>
        </td>
        <td><p>Calculate Phylostratum Enrichment</p></td>
      </tr><tr><td>
          <p><code><a href="taxid.html">taxid()</a></code> </p>
        </td>
        <td><p>Retrieve taxonomy categories from NCBI Taxonomy</p></td>
      </tr><tr><td>
          <p><code><a href="tf.html">tf()</a></code> </p>
        </td>
        <td><p>Short Alias for Transform Counts</p></td>
      </tr><tr><td>
          <p><code><a href="tf_PS.html">tf_PS()</a></code> </p>
        </td>
        <td><p>Transform Phylostratum Values</p></td>
      </tr><tr><td>
          <p><code><a href="tf_stability.html">tf_stability()</a></code> </p>
        </td>
        <td><p>Perform Permutation Tests Under Different Transformations</p></td>
      </tr><tr><td>
          <p><code><a href="transform_counts.html">transform_counts()</a></code> </p>
        </td>
        <td><p>Transform Expression Counts in PhyloExpressionSet</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Ștefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

