<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Fira_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet"><link href="../deps/Fira_Sans-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Package index"><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.3.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-stats.html">Statistical testing with myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-transform.html">Transforming dataset for myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a></li>
    <li><a class="dropdown-item" href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a></li>
    <li><a class="dropdown-item" href="../articles/index.html">... and view more!</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="visualisation">Visualisation<a class="anchor" aria-label="anchor" href="#visualisation"></a></h2>

      <div class="section-desc"><p>Visualise and visually inspect evolutionary transcriptomics patterns as well as summary statistics.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="plot_signature.html">plot_signature()</a></code>

        </dt>
        <dd>Plot Transcriptomic Signature</dd>
      </dl><dl><dt>

          <code><a href="plot_signature_gene_quantiles.html">plot_signature_gene_quantiles()</a></code>

        </dt>
        <dd>Plot Signature Across Gene Expression Quantiles</dd>
      </dl><dl><dt>

          <code><a href="plot_signature_multiple.html">plot_signature_multiple()</a></code>

        </dt>
        <dd>Plot Multiple Transcriptomic Signatures</dd>
      </dl><dl><dt>

          <code><a href="plot_signature_transformed.html">plot_signature_transformed()</a></code>

        </dt>
        <dd>Plot Signature Under Different Transformations</dd>
      </dl><dl><dt>

          <code><a href="plot_contribution.html">plot_contribution()</a></code>

        </dt>
        <dd>Plot Phylostratum Contribution to Transcriptomic Index</dd>
      </dl><dl><dt>

          <code><a href="plot_cullen_frey.html">plot_cullen_frey()</a></code>

        </dt>
        <dd>Plot Cullen-Frey Diagram for Distribution Assessment</dd>
      </dl><dl><dt>

          <code><a href="plot_distribution_expression.html">plot_distribution_expression()</a></code>

        </dt>
        <dd>Comparing expression levels distributions across developmental stages</dd>
      </dl><dl><dt>

          <code><a href="plot_distribution_pTAI.html">plot_distribution_pTAI()</a></code>

        </dt>
        <dd>Partial TAI Distribution Plotting Functions</dd>
      </dl><dl><dt>

          <code><a href="plot_distribution_pTAI_qqplot.html">plot_distribution_pTAI_qqplot()</a></code>

        </dt>
        <dd>QQ plot comparing partial TAI distributions across developmental stages against a reference stage</dd>
      </dl><dl><dt>

          <code><a href="plot_distribution_strata.html">plot_distribution_strata()</a></code>

        </dt>
        <dd>Plot Distribution of Genes Across Phylostrata</dd>
      </dl><dl><dt>

          <code><a href="plot_gatai_results.html">plot_gatai_results()</a></code>

        </dt>
        <dd>Plot Comprehensive GATAI Results</dd>
      </dl><dl><dt>

          <code><a href="plot_gene_heatmap.html">plot_gene_heatmap()</a></code>

        </dt>
        <dd>Plot Gene Expression Heatmap</dd>
      </dl><dl><dt>

          <code><a href="plot_gene_profiles.html">plot_gene_profiles()</a></code>

        </dt>
        <dd>Plot Individual Gene Expression Profiles</dd>
      </dl><dl><dt>

          <code><a href="plot_gene_space.html">plot_gene_space()</a></code>

        </dt>
        <dd>Plot Gene Space Using PCA</dd>
      </dl><dl><dt>

          <code><a href="plot_mean_var.html">plot_mean_var()</a></code>

        </dt>
        <dd>Plot Mean-Variance Relationship</dd>
      </dl><dl><dt>

          <code><a href="plot_null_txi_sample.html">plot_null_txi_sample()</a></code>

        </dt>
        <dd>Plot Null TXI Sample Distribution</dd>
      </dl><dl><dt>

          <code><a href="plot_relative_expression_bar.html">plot_relative_expression_bar()</a></code>

        </dt>
        <dd>Plot Mean Relative Expression Levels as Barplot</dd>
      </dl><dl><dt>

          <code><a href="plot_relative_expression_line.html">plot_relative_expression_line()</a></code>

        </dt>
        <dd>Plot Relative Expression Profiles (Line Plot)</dd>
      </dl><dl><dt>

          <code><a href="plot_sample_space.html">plot_sample_space()</a></code>

        </dt>
        <dd>Plot Sample Space Visualization</dd>
      </dl><dl><dt>

          <code><a href="plot_strata_expression.html">plot_strata_expression()</a></code>

        </dt>
        <dd>Plot Expression Levels by Phylostratum</dd>
      </dl></div><div class="section level2">
      <h2 id="statistical-tests">Statistical tests<a class="anchor" aria-label="anchor" href="#statistical-tests"></a></h2>

      <div class="section-desc"><p>Functions for statistical testing of evolutionary transcriptomics patterns (with visualisation).</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="stat_early_conservation_test.html">stat_early_conservation_test()</a></code>

        </dt>
        <dd>Early Conservation Test</dd>
      </dl><dl><dt>

          <code><a href="stat_flatline_test.html">stat_flatline_test()</a></code>

        </dt>
        <dd>Flat Line Test for Conservation Pattern</dd>
      </dl><dl><dt>

          <code><a href="stat_generic_conservation_test.html">stat_generic_conservation_test()</a></code>

        </dt>
        <dd>Generic Conservation Test Framework</dd>
      </dl><dl><dt>

          <code><a href="stat_late_conservation_test.html">stat_late_conservation_test()</a></code>

        </dt>
        <dd>Late Conservation Test</dd>
      </dl><dl><dt>

          <code><a href="stat_pairwise_test.html">stat_pairwise_test()</a></code>

        </dt>
        <dd>Pairwise Conservation Test</dd>
      </dl><dl><dt>

          <code><a href="stat_reductive_hourglass_test.html">stat_reductive_hourglass_test()</a></code>

        </dt>
        <dd>Reductive Hourglass Test</dd>
      </dl><dl><dt>

          <code><a href="stat_reverse_hourglass_test.html">stat_reverse_hourglass_test()</a></code>

        </dt>
        <dd>Reverse Hourglass Test</dd>
      </dl></div><div class="section level2">
      <h2 id="data-structures">Data structures<a class="anchor" aria-label="anchor" href="#data-structures"></a></h2>

      <div class="section-desc"><p>Core S7 data classes used for evolutionary transcriptomics analysis.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="BulkPhyloExpressionSet.html">BulkPhyloExpressionSet()</a></code>

        </dt>
        <dd>Bulk PhyloExpressionSet Class</dd>
      </dl><dl><dt>

          <code><a href="ConservationTestResult.html">ConservationTestResult()</a></code>

        </dt>
        <dd>Conservation Test Result S7 Class</dd>
      </dl><dl><dt>

          <code><a href="PhyloExpressionSetBase.html">PhyloExpressionSetBase()</a></code>

        </dt>
        <dd>PhyloExpressionSet Base Class</dd>
      </dl><dl><dt>

          <code><a href="ScPhyloExpressionSet.html">ScPhyloExpressionSet()</a></code>

        </dt>
        <dd>Single-Cell PhyloExpressionSet Class</dd>
      </dl><dl><dt>

          <code><a href="TestResult.html">TestResult()</a></code>

        </dt>
        <dd>Test Result S7 Class</dd>
      </dl><dl><dt>

          <code><a href="Distribution.html">Distribution()</a></code>

        </dt>
        <dd>Distribution S7 Class</dd>
      </dl></div><div class="section level2">
      <h2 id="data-structures-related">Data structures-related<a class="anchor" aria-label="anchor" href="#data-structures-related"></a></h2>

      <div class="section-desc"><p>Functions to convert between dataframe and <code>BulkPhyloExpressionSet</code> or <code>ScPhyloExpressionSet</code>.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="as_BulkPhyloExpressionSet.html">as_BulkPhyloExpressionSet()</a></code>

        </dt>
        <dd>as_BulkPhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="BulkPhyloExpressionSet_from_df.html">BulkPhyloExpressionSet_from_df()</a></code>

        </dt>
        <dd>Convert Data to BulkPhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="ScPhyloExpressionSet_from_matrix.html">ScPhyloExpressionSet_from_matrix()</a></code>

        </dt>
        <dd>Create Single-Cell PhyloExpressionSet from Expression Matrix</dd>
      </dl><dl><dt>

          <code><a href="ScPhyloExpressionSet_from_seurat.html">ScPhyloExpressionSet_from_seurat()</a></code>

        </dt>
        <dd>Convert Seurat Object to Single-Cell PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="as_data_frame.html">as_data_frame()</a></code>

        </dt>
        <dd>Convert BulkPhyloExpressionSet to Data Frame</dd>
      </dl><dl><dt>

          <code><a href="check_PhyloExpressionSet.html">check_PhyloExpressionSet()</a></code>

        </dt>
        <dd>Check if object is a PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="check_BulkPhyloExpressionSet.html">check_BulkPhyloExpressionSet()</a></code>

        </dt>
        <dd>Check if object is a BulkPhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="check_ScPhyloExpressionSet.html">check_ScPhyloExpressionSet()</a></code>

        </dt>
        <dd>Check if object is a ScPhyloExpressionSet</dd>
      </dl></div><div class="section level2">
      <h2 id="single-cell-specific-functions">Single-cell specific functions<a class="anchor" aria-label="anchor" href="#single-cell-specific-functions"></a></h2>

      <div class="section-desc"><p>Functions specifically designed for single-cell RNA-seq data analysis.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="downsample_expression.html">downsample_expression()</a></code>

        </dt>
        <dd>Downsample Expression Matrix by Groups</dd>
      </dl><dl><dt>

          <code><a href="downsample.html">downsample()</a></code>

        </dt>
        <dd>Downsample ScPhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="match_map_sc_matrix.html">match_map_sc_matrix()</a></code>

        </dt>
        <dd>Match Expression Matrix with Phylostratum Map</dd>
      </dl><dl><dt>

          <code><a href="match_map_sc_seurat.html">match_map_sc_seurat()</a></code>

        </dt>
        <dd>Match Single-Cell Expression Data with Phylostratum Map (Seurat)</dd>
      </dl></div><div class="section level2">
      <h2 id="taxonomic-information-retrieval">Taxonomic information retrieval<a class="anchor" aria-label="anchor" href="#taxonomic-information-retrieval"></a></h2>

      <div class="section-desc"><p>Functions to compute transcriptome indices.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="taxid.html">taxid()</a></code>

        </dt>
        <dd>Retrieve taxonomy categories from NCBI Taxonomy</dd>
      </dl></div><div class="section level2">
      <h2 id="transcriptomic-index-calculation">Transcriptomic index calculation<a class="anchor" aria-label="anchor" href="#transcriptomic-index-calculation"></a></h2>

      <div class="section-desc"><p>Compute full and partial evolutionary transcriptomics measures.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="TAI.html">TAI()</a></code>

        </dt>
        <dd>Calculate Transcriptomic Age Index (TAI)</dd>
      </dl><dl><dt>

          <code><a href="TDI.html">TDI()</a></code>

        </dt>
        <dd>Calculate Transcriptomic Divergence Index (TDI)</dd>
      </dl><dl><dt>

          <code><a href="TEI.html">TEI()</a></code>

        </dt>
        <dd>Calculate Transcriptomic Evolutionary Index (TEI)</dd>
      </dl><dl><dt>

          <code><a href="TPI.html">TPI()</a></code>

        </dt>
        <dd>Calculate Transcriptomic Polymorphism Index (TPI)</dd>
      </dl><dl><dt>

          <code><a href="TXI.html">TXI()</a></code>

        </dt>
        <dd>Calculate Transcriptomic Index (TXI)</dd>
      </dl><dl><dt>

          <code><a href="sTXI.html">sTXI()</a></code>

        </dt>
        <dd>Calculate Stratum-Specific Transcriptomic Index</dd>
      </dl><dl><dt>

          <code><a href="pTXI.html">pTXI()</a></code>

        </dt>
        <dd>Calculate Phylostratum-Specific Transcriptomic Index</dd>
      </dl></div><div class="section level2">
      <h2 id="expression-or-index-data-related">Expression or index data-related<a class="anchor" aria-label="anchor" href="#expression-or-index-data-related"></a></h2>

      <div class="section-desc"><p>Functions to transform, normalise, and manipulate expression datasets.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="age.apply.html">age.apply()</a></code>

        </dt>
        <dd>Age Category Specific apply Function</dd>
      </dl><dl><dt>

          <code><a href="collapse.html">collapse()</a></code>

        </dt>
        <dd>Collapse PhyloExpressionSet Replicates</dd>
      </dl><dl><dt>

          <code><a href="destroy_pattern.html">destroy_pattern()</a></code>

        </dt>
        <dd>Destroy Phylotranscriptomic Pattern Using GATAI</dd>
      </dl><dl><dt>

          <code><a href="diagnose_test_robustness.html">diagnose_test_robustness()</a></code>

        </dt>
        <dd>Diagnose Test Robustness</dd>
      </dl><dl><dt>

          <code><a href="distributions.html">distributions</a></code>

        </dt>
        <dd>Predefined Distribution Objects</dd>
      </dl><dl><dt>

          <code><a href="exp_p.html">exp_p()</a></code>

        </dt>
        <dd>Format P-Value for Scientific Notation</dd>
      </dl><dl><dt>

          <code><a href="genes_lowly_expressed.html">genes_lowly_expressed()</a></code>

        </dt>
        <dd>Select Lowly Expressed Genes</dd>
      </dl><dl><dt>

          <code><a href="genes_top_expr.html">genes_top_expr()</a></code>

        </dt>
        <dd>Gene Expression Filtering Functions</dd>
      </dl><dl><dt>

          <code><a href="genes_top_mean.html">genes_top_mean()</a></code>

        </dt>
        <dd>Select Top Mean Expressed Genes</dd>
      </dl><dl><dt>

          <code><a href="genes_top_variance.html">genes_top_variance()</a></code>

        </dt>
        <dd>Select Top Variable Genes</dd>
      </dl><dl><dt>

          <code><a href="match_map.html">match_map()</a></code>

        </dt>
        <dd>Match Gene Expression Data with Phylostratum Map</dd>
      </dl><dl><dt>

          <code><a href="normalise_stage_expression.html">normalise_stage_expression()</a></code>

        </dt>
        <dd>Normalise Stage Expression Data</dd>
      </dl><dl><dt>

          <code><a href="omit_matrix.html">omit_matrix()</a></code>

        </dt>
        <dd>Compute TXI Profiles Omitting Each Gene</dd>
      </dl><dl><dt>

          <code><a href="permute_PS.html">permute_PS()</a></code>

        </dt>
        <dd>Permute Strata in PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="quantile_rank.html">quantile_rank()</a></code>

        </dt>
        <dd>Calculate Quantile Ranks</dd>
      </dl><dl><dt>

          <code><a href="rel_exp_matrix.html">rel_exp_matrix()</a></code>

        </dt>
        <dd>Compute Relative Expression Matrix for PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="relative_expression.html">relative_expression()</a></code>

        </dt>
        <dd>Relative Expression Functions</dd>
      </dl><dl><dt>

          <code><a href="remove_genes.html">remove_genes()</a></code>

        </dt>
        <dd>Remove Genes from PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="select_genes.html">select_genes()</a></code>

        </dt>
        <dd>Select Genes from PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="set_expression.html">set_expression()</a></code>

        </dt>
        <dd>Gene Expression Transformation Functions</dd>
      </dl><dl><dt>

          <code><a href="strata_enrichment.html">strata_enrichment()</a></code>

        </dt>
        <dd>Calculate Phylostratum Enrichment</dd>
      </dl><dl><dt>

          <code><a href="transform_counts.html">transform_counts()</a></code>

        </dt>
        <dd>Transform Expression Counts in PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="tf.html">tf()</a></code>

        </dt>
        <dd>Short Alias for Transform Counts</dd>
      </dl><dl><dt>

          <code><a href="tf_PS.html">tf_PS()</a></code>

        </dt>
        <dd>Transform Phylostratum Values</dd>
      </dl><dl><dt>

          <code><a href="tf_stability.html">tf_stability()</a></code>

        </dt>
        <dd>Perform Permutation Tests Under Different Transformations</dd>
      </dl></div><div class="section level2">
      <h2 id="utilities-metadata">Utilities/Metadata<a class="anchor" aria-label="anchor" href="#utilities-metadata"></a></h2>

      <div class="section-desc"><p>Helper functions and metadata constants.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="COUNT_TRANSFORMS.html">COUNT_TRANSFORMS</a></code>

        </dt>
        <dd>Count Transformation Functions</dd>
      </dl><dl><dt>

          <code><a href="PS_colours.html">PS_colours()</a></code>

        </dt>
        <dd>Generate Phylostratum Colors</dd>
      </dl><dl><dt>

          <code><a href="save_gatai_results_pdf.html">save_gatai_results_pdf()</a></code>

        </dt>
        <dd>Save GATAI Analysis Results to PDF</dd>
      </dl><dl><dt>

          <code><a href="print.TestResult.html">print.TestResult</a></code>

        </dt>
        <dd>Print Method for TestResult</dd>
      </dl><dl><dt>

          <code><a href="rename_phyex_set.html">rename_phyex_set()</a></code>

        </dt>
        <dd>Rename a PhyloExpressionSet</dd>
      </dl><dl><dt>

          <code><a href="TI_map.html">TI_map</a></code>

        </dt>
        <dd>Transcriptomic Index Name Mapping</dd>
      </dl><dl><dt>

          <code><a href="TXI_conf_int.html">TXI_conf_int()</a></code>

        </dt>
        <dd>Confidence Intervals for Transcriptomic Index (TXI)</dd>
      </dl><dl><dt>

          <code><a href="TXI_std_dev.html">TXI_std_dev()</a></code>

        </dt>
        <dd>Standard Deviation for TXI</dd>
      </dl></div><div class="section level2">
      <h2 id="example-datasets">Example datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a></h2>

      <div class="section-desc"><p>Example datasets for demonstrating and testing package functions.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="example_phyex_set.html">example_phyex_set</a></code>

        </dt>
        <dd>Example phyex set</dd>
      </dl><dl><dt>

          <code><a href="example_phyex_set_old.html">example_phyex_set_old</a></code>

        </dt>
        <dd>Example phyex set old</dd>
      </dl><dl><dt>

          <code><a href="example_phyex_set_sc.html">example_phyex_set_sc</a></code>

        </dt>
        <dd>Load Example Single-Cell PhyloExpressionSet</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Stefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

