<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>PhyloExpressionSet Base Class — PhyloExpressionSetBase • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="PhyloExpressionSet Base Class — PhyloExpressionSetBase"><meta property="og:description" content="Abstract S7 base class for storing and manipulating phylotranscriptomic expression data.
This class provides the common interface for both bulk and single-cell phylotranscriptomic data."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.3.0.9006</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/myTAI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/other-strata.html">Other evolutionary and expression indices</a>
    </li>
    <li>
      <a href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a>
    </li>
    <li>
      <a href="../articles/phylostratigraphy.html">Gene age inference</a>
    </li>
    <li>
      <a href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a>
    </li>
    <li>
      <a href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-stats.html">Statistical testing with myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-transform.html">Transforming dataset for myTAI</a>
    </li>
    <li>
      <a href="../articles/taxonomy.html">Obtaining taxonomy information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>PhyloExpressionSet Base Class</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/phyloset_base.R" class="external-link"><code>R/phyloset_base.R</code></a></small>
    <div class="hidden name"><code>PhyloExpressionSetBase.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Abstract S7 base class for storing and manipulating phylotranscriptomic expression data.
This class provides the common interface for both bulk and single-cell phylotranscriptomic data.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PhyloExpressionSetBase</span><span class="op">(</span></span>
<span>  strata <span class="op">=</span> <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"@strata is required"</span><span class="op">)</span>,</span>
<span>  strata_values <span class="op">=</span> <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"@strata_values is required"</span><span class="op">)</span>,</span>
<span>  expression <span class="op">=</span> <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"@expression is required"</span><span class="op">)</span>,</span>
<span>  groups <span class="op">=</span> <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"@groups is required"</span><span class="op">)</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"Phylo Expression Set"</span>,</span>
<span>  species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>,</span>
<span>  index_type <span class="op">=</span> <span class="st">"TXI"</span>,</span>
<span>  identities_label <span class="op">=</span> <span class="st">"Identities"</span>,</span>
<span>  null_conservation_sample_size <span class="op">=</span> <span class="fl">5000L</span>,</span>
<span>  .null_conservation_txis <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-strata">strata<a class="anchor" aria-label="anchor" href="#arg-strata"></a></dt>
<dd><p>Factor vector of phylostratum assignments for each gene</p></dd>


<dt id="arg-strata-values">strata_values<a class="anchor" aria-label="anchor" href="#arg-strata-values"></a></dt>
<dd><p>Numeric vector of phylostratum values used in TXI calculations</p></dd>


<dt id="arg-expression">expression<a class="anchor" aria-label="anchor" href="#arg-expression"></a></dt>
<dd><p>Matrix of expression counts with genes as rows and samples as columns</p></dd>


<dt id="arg-groups">groups<a class="anchor" aria-label="anchor" href="#arg-groups"></a></dt>
<dd><p>Factor vector indicating which identity each sample belongs to</p></dd>


<dt id="arg-name">name<a class="anchor" aria-label="anchor" href="#arg-name"></a></dt>
<dd><p>Character string naming the dataset (default: "Phylo Expression Set")</p></dd>


<dt id="arg-species">species<a class="anchor" aria-label="anchor" href="#arg-species"></a></dt>
<dd><p>Character string specifying the species (default: NULL)</p></dd>


<dt id="arg-index-type">index_type<a class="anchor" aria-label="anchor" href="#arg-index-type"></a></dt>
<dd><p>Character string specifying the transcriptomic index type (default: "TXI")</p></dd>


<dt id="arg-identities-label">identities_label<a class="anchor" aria-label="anchor" href="#arg-identities-label"></a></dt>
<dd><p>Character string labeling the identities (default: "Identities")</p></dd>


<dt id="arg-null-conservation-sample-size">null_conservation_sample_size<a class="anchor" aria-label="anchor" href="#arg-null-conservation-sample-size"></a></dt>
<dd><p>Numeric value for null conservation sample size (default: 5000)</p></dd>


<dt id="arg--null-conservation-txis">.null_conservation_txis<a class="anchor" aria-label="anchor" href="#arg--null-conservation-txis"></a></dt>
<dd><p>Precomputed null conservation TXI values (default: NULL)</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The PhyloExpressionSetBase class serves as the foundation for phylotranscriptomic analysis,
providing shared functionality for both bulk and single-cell data types.</p>
<p><strong>Abstract Properties:</strong>
Subclasses must implement the <code>expression_collapsed</code> property to define how expression
data should be collapsed across replicates or cells.</p>
<p><strong>Computed Properties:</strong>
Several properties are computed automatically when accessed:</p><ul><li><p><code>gene_ids</code> - Character vector of gene identifiers (rownames of expression matrix)</p></li>
<li><p><code>identities</code> - Character vector of identity labels (colnames of collapsed expression)</p></li>
<li><p><code>sample_names</code> - Character vector of sample names (colnames of expression matrix)</p></li>
<li><p><code>num_identities</code> - Integer count of unique identities</p></li>
<li><p><code>num_samples</code> - Integer count of total samples</p></li>
<li><p><code>num_genes</code> - Integer count of genes</p></li>
<li><p><code>num_strata</code> - Integer count of phylostrata</p></li>
<li><p><code>index_full_name</code> - Full name of the transcriptomic index type</p></li>
<li><p><code>group_map</code> - List mapping identity names to sample names</p></li>
<li><p><code>TXI</code> - Numeric vector of TXI values for each identity (computed from collapsed expression)</p></li>
<li><p><code>TXI_sample</code> - Numeric vector of TXI values for each sample (computed from raw expression)</p></li>
<li><p><code>null_conservation_txis</code> - Matrix of null conservation TXI values for statistical testing</p></li>
</ul><p><strong>Validation:</strong>
The class ensures consistency between expression data, phylostratum assignments, and groupings.
All gene-level vectors must have matching lengths, and sample groupings must be consistent.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Ștefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

