<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot Multiple Transcriptomic Signatures — plot_signature_multiple • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Plot Multiple Transcriptomic Signatures — plot_signature_multiple"><meta property="og:description" content="Create a plot comparing multiple transcriptomic signatures on the same axes,
with options for statistical testing and transformations."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.3.0.9006</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/myTAI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/other-strata.html">Other evolutionary and expression indices</a>
    </li>
    <li>
      <a href="../articles/phylo-expression-object.html">Bring your datasets into myTAI</a>
    </li>
    <li>
      <a href="../articles/phylostratigraphy.html">Gene age inference</a>
    </li>
    <li>
      <a href="../articles/tai-breaker.html">Break TAI patterns using gaTAI</a>
    </li>
    <li>
      <a href="../articles/tai-gallery.html">Beautiful plots made via myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-stats.html">Statistical testing with myTAI</a>
    </li>
    <li>
      <a href="../articles/tai-transform.html">Transforming dataset for myTAI</a>
    </li>
    <li>
      <a href="../articles/taxonomy.html">Obtaining taxonomy information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot Multiple Transcriptomic Signatures</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/plot_signature_multiple.R" class="external-link"><code>R/plot_signature_multiple.R</code></a></small>
    <div class="hidden name"><code>plot_signature_multiple.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Create a plot comparing multiple transcriptomic signatures on the same axes,
with options for statistical testing and transformations.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_signature_multiple</span><span class="op">(</span></span>
<span>  <span class="va">phyex_sets</span>,</span>
<span>  legend_title <span class="op">=</span> <span class="st">"Phylo Expression Set"</span>,</span>
<span>  show_p_val <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  conservation_test <span class="op">=</span> <span class="va">stat_flatline_test</span>,</span>
<span>  transformation <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  colours <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-phyex-sets">phyex_sets<a class="anchor" aria-label="anchor" href="#arg-phyex-sets"></a></dt>
<dd><p>A vector of PhyloExpressionSet objects to compare (BulkPhyloExpressionSet or ScPhyloExpressionSet)</p></dd>


<dt id="arg-legend-title">legend_title<a class="anchor" aria-label="anchor" href="#arg-legend-title"></a></dt>
<dd><p>Title for the legend (default: "Phylo Expression Set")</p></dd>


<dt id="arg-show-p-val">show_p_val<a class="anchor" aria-label="anchor" href="#arg-show-p-val"></a></dt>
<dd><p>Logical indicating whether to show p-values (default: TRUE)</p></dd>


<dt id="arg-conservation-test">conservation_test<a class="anchor" aria-label="anchor" href="#arg-conservation-test"></a></dt>
<dd><p>Function to use for conservation testing (default: stat_flatline_test)</p></dd>


<dt id="arg-transformation">transformation<a class="anchor" aria-label="anchor" href="#arg-transformation"></a></dt>
<dd><p>Optional transformation function to apply to all datasets (default: NULL)</p></dd>


<dt id="arg-colours">colours<a class="anchor" aria-label="anchor" href="#arg-colours"></a></dt>
<dd><p>Optional vector of colors for each dataset (default: NULL)</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments passed to plot_signature</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A ggplot2 object showing multiple transcriptomic signatures</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function allows comparison of multiple transcriptomic signatures by overlaying
them on the same plot. Each signature is colored differently and can be tested
for conservation patterns (bulk data only). If a transformation is provided, it's applied to all
datasets before plotting.</p>
<p>The function automatically adapts to the data type:
- **Bulk data**: Line plots with optional statistical testing
- **Single-cell data**: Violin plots showing distributions</p>
<p>All datasets must use the same axis labels (developmental stages or cell types).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Compare multiple bulk datasets</span></span></span>
<span class="r-in"><span><span class="co"># bulk_list &lt;- c(bulk_phyex_set1, bulk_phyex_set2)</span></span></span>
<span class="r-in"><span><span class="co"># p &lt;- plot_signature_multiple(bulk_list, legend_title = "Dataset")</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Compare single-cell datasets</span></span></span>
<span class="r-in"><span><span class="co"># sc_list &lt;- c(sc_phyex_set1, sc_phyex_set2)</span></span></span>
<span class="r-in"><span><span class="co"># p2 &lt;- plot_signature_multiple(sc_list, legend_title = "Cell Type")</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># With transformation</span></span></span>
<span class="r-in"><span><span class="co"># p3 &lt;- plot_signature_multiple(bulk_list, transformation = log1p)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Ștefan Manolache, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

