<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot the Expression Profiles of a Gene Set — PlotGeneSet • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot the Expression Profiles of a Gene Set — PlotGeneSet"><meta name="description" content="This function simply visualizes the gene expression profiles of
a defined subset of genes stored in the input ExpressionSet."><meta property="og:description" content="This function simply visualizes the gene expression profiles of
a defined subset of genes stored in the input ExpressionSet."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a></li>
    <li><a class="dropdown-item" href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Introduction.html">Introduction to the myTAI package</a></li>
    <li><a class="dropdown-item" href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Plot the Expression Profiles of a Gene Set</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/PlotGeneSet.R" class="external-link"><code>R/PlotGeneSet.R</code></a></small>
      <div class="d-none name"><code>PlotGeneSet.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function simply visualizes the gene expression profiles of
a defined subset of genes stored in the input <code>ExpressionSet</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PlotGeneSet</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  <span class="va">gene.set</span>,</span>
<span>  get.subset <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use.only.map <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot.legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  y.ticks <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  digits.ylab <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-expressionset">ExpressionSet<a class="anchor" aria-label="anchor" href="#arg-expressionset"></a></dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt id="arg-gene-set">gene.set<a class="anchor" aria-label="anchor" href="#arg-gene-set"></a></dt>
<dd><p>a character vector storing the gene ids for which gene expression profiles shall be visualized.</p></dd>


<dt id="arg-get-subset">get.subset<a class="anchor" aria-label="anchor" href="#arg-get-subset"></a></dt>
<dd><p>a logical value indicating whether or not an <code>ExpressionSet</code> subset of the selected <code>gene.set</code> should be retuned.</p></dd>


<dt id="arg-use-only-map">use.only.map<a class="anchor" aria-label="anchor" href="#arg-use-only-map"></a></dt>
<dd><p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to the function.</p></dd>


<dt id="arg-colors">colors<a class="anchor" aria-label="anchor" href="#arg-colors"></a></dt>
<dd><p>colors for gene expression profiles. Default: <code>colors = NULL</code>, hence default colours are used.</p></dd>


<dt id="arg-plot-legend">plot.legend<a class="anchor" aria-label="anchor" href="#arg-plot-legend"></a></dt>
<dd><p>a logical value indicating whether gene ids should be printed as legend next to the plot.</p></dd>


<dt id="arg-y-ticks">y.ticks<a class="anchor" aria-label="anchor" href="#arg-y-ticks"></a></dt>
<dd><p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p></dd>


<dt id="arg-digits-ylab">digits.ylab<a class="anchor" aria-label="anchor" href="#arg-digits-ylab"></a></dt>
<dd><p>a numeric value specifying the number of digits shown for the expression levels on the y-axis.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>additional parameters passed to <code><a href="https://rdrr.io/r/graphics/matplot.html" class="external-link">matplot</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function simply visualizes or subsets the gene expression levels of a set of genes
that are stored in the input <code>ExpressionSet</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="SelectGeneSet.html">SelectGeneSet</a></code>, <code><a href="PlotEnrichment.html">PlotEnrichment</a></code>, <code><a href="DiffGenes.html">DiffGenes</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the best parameter setting to visualize this plot:</span></span></span>
<span class="r-in"><span><span class="co"># png("test_png.png",700,400)</span></span></span>
<span class="r-in"><span><span class="fu">PlotGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>            gene.set      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>            type          <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            lty           <span class="op">=</span> <span class="fl">1</span>, </span></span>
<span class="r-in"><span>            lwd           <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>            xlab          <span class="op">=</span> <span class="st">"Ontogeny"</span>,</span></span>
<span class="r-in"><span>            ylab          <span class="op">=</span> <span class="st">"Expression Level"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="PlotGeneSet-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># dev.off()</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># In case you would like to work with the expression levels</span></span></span>
<span class="r-in"><span><span class="co"># of selected genes you can specify the 'get.subset' argument:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>            gene.set      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>            get.subset    <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID    Zygote   Quadrant  Globular      Heart   Torpedo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  2173.635  1911.2001  1152.555  1291.4224  1000.253</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  1501.014  1817.3086  1665.309  1564.7612  1496.321</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  1212.793  1233.0023   939.200   929.6195   864.218</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2  1016.920   936.3837  1181.338  1329.4734  1392.643</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 11424.567 16778.1685 34366.649 39775.6405 56231.569</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Bent    Mature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   962.9772 1696.4274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  1114.6435 1071.6555</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   877.2060  894.8189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  1287.9746  861.2605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 66980.3673 7772.5617</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get a gene subset using only a phylostratihraphic map</span></span></span>
<span class="r-in"><span><span class="va">ExamplePSMap</span> <span class="op">&lt;-</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span> , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">ExamplePSMap</span>, </span></span>
<span class="r-in"><span>            gene.set      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>            get.subset    <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>            use.only.map  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1</span>
<span class="r-in"><span>            </span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

